<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>gyq&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta property="og:type" content="website">
<meta property="og:title" content="gyq&#39;s blog">
<meta property="og:url" content="http://example.com/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="gongyuqiqi">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="gyq's blog" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 5.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">gyq&#39;s blog</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main">
  
    <article id="post-å¯å˜å‰ªåˆ‡ä¹‹SUPPA2" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/" class="article-date">
  <time class="dt-published" datetime="2021-01-23T08:30:27.000Z" itemprop="datePublished">2021-01-23</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="å¯å˜å‰ªåˆ‡ä¹‹SUPPA2"><a href="#å¯å˜å‰ªåˆ‡ä¹‹SUPPA2" class="headerlink" title="å¯å˜å‰ªåˆ‡ä¹‹SUPPA2"></a>å¯å˜å‰ªåˆ‡ä¹‹SUPPA2</h1><p>SUPPA2å‚è€ƒèµ„æ–™ï¼š<br><a target="_blank" rel="noopener" href="https://github.com/comprna/SUPPA/wiki/SUPPA2-tutorial#differential-splicing-with-local-events">https://github.com/comprna/SUPPA/wiki/SUPPA2-tutorial#differential-splicing-with-local-events</a><br>åˆ†ææµç¨‹ä¸­æ‰€éœ€è¦å¾—<strong>å‚è€ƒåŸºå› ç»„</strong>ã€<strong>æ³¨é‡Šæ–‡ä»¶</strong>ã€<strong>Rè„šæœ¬</strong>éƒ½å¯ä»¥åœ¨è¯¥é“¾æ¥ä¸­æ‰¾åˆ°å¹¶ä¸‹è½½</p>
<p>ä¸‹é¢è¿™æ®µè¯å¼•è‡ªå®˜ç½‘ï¼Œç®€å•äº†è§£ä¸€ä¸‹SUPPA2åŠå…¶åŠŸèƒ½</p>
<blockquote>
<p>SUPPA2 obtains the inclusion levels of AS events exploiting transcript quantification. There are different measures for isoform quantification (counts, TMM, FPKMâ€¦). We strongly recommend the use of TPM (Transcripts Per Million), because is normalized for gene length first, and then for sequencing depth. This allows direct comparison between samples (check the following video if you want to know more).</p>
</blockquote>
<blockquote>
<p>There are several tools for transcript quantification (this is an interesting reviewÂ² on this matter). Each one has different pros and cons and the user is free to use the one that fits better to his requirements. We recommend to use SalmonÂ³, because is one of the fastest methods for quantification without compromising accuracy and it also offers some other interesting options like GC bias correction or bootstrapping among others. The commands indicated here are oriented for using this tools.</p>
</blockquote>
<h2 id="ä¸€-ç¯å¢ƒæ­å»º"><a href="#ä¸€-ç¯å¢ƒæ­å»º" class="headerlink" title="(ä¸€)ç¯å¢ƒæ­å»º"></a>(ä¸€)ç¯å¢ƒæ­å»º</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda creat SUPPA2_3.9 python=3.9.1</span><br><span class="line">conda activate SUPPA2_3.9</span><br><span class="line">conda intall -c bioconda salmon suppa matplotlib</span><br></pre></td></tr></table></figure>

<h2 id="äºŒ-è½¬å½•æœ¬å®šé‡"><a href="#äºŒ-è½¬å½•æœ¬å®šé‡" class="headerlink" title="(äºŒ)è½¬å½•æœ¬å®šé‡"></a>(äºŒ)è½¬å½•æœ¬å®šé‡</h2><h3 id="1ã€å»ºç«‹ç´¢å¼•"><a href="#1ã€å»ºç«‹ç´¢å¼•" class="headerlink" title="1ã€å»ºç«‹ç´¢å¼•"></a>1ã€å»ºç«‹ç´¢å¼•</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">salmon index -t hg19_EnsenmblGenes_sequence_ensenmbl.fasta -i Ensembl_hg19_salmon_index</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/hg19_salmon_index.JPG"/>

<h3 id="2ã€å®šé‡è·å¾—TPMå€¼"><a href="#2ã€å®šé‡è·å¾—TPMå€¼" class="headerlink" title="2ã€å®šé‡è·å¾—TPMå€¼"></a>2ã€å®šé‡è·å¾—TPMå€¼</h3><p><strong>(1)ã€è¿è¡Œsalmonè½¯ä»¶å®šé‡</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">x&#x3D;_1</span><br><span class="line">y&#x3D;_2</span><br><span class="line">for id in &#123;SRR1513329,SRR1513330,SRR1513331,SRR1513332,SRR1513333,SRR1513334&#125;</span><br><span class="line">do</span><br><span class="line">nohup salmon quant -i &#x2F;home&#x2F;gongyuqi&#x2F;ref&#x2F;hg38&#x2F;ensembl&#x2F;SUPPA_ref&#x2F;Ensembl_hg19_salmon_index -l ISF --gcBias -1 $id$x.fastq.gz -2 $id$y.fastq.gz -p 20 -o &#x2F;home&#x2F;gongyuqi&#x2F;project&#x2F;AS&#x2F;SUPPA2&#x2F;$id &amp;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>
<p>å±•ç¤ºä¸€ä¸ªæ ·æœ¬çš„è¾“å‡ºç»“æœç»“æ„<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tree_SRR1513329.JPG"/><br>å…¶ä¸­quant.sfæ–‡ä»¶å¾ˆé‡è¦ï¼Œè¦ç”¨äºåç»­çš„åˆ†æï¼Œå±•ç¤ºè¯¥æ–‡ä»¶éƒ¨åˆ†å†…å®¹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/quant_df.JPG"/></p>
<p><strong>(2)ã€æå–æ‰€æœ‰æ ·æœ¬çš„TPMå€¼å¹¶åˆå¹¶ä¸ºä¸€ä¸ªæ–‡ä»¶</strong></p>
<p><code>-k</code> indicates the row used as index<br><code>-f</code> TPMå€¼åœ¨ç¬¬å››åˆ—ï¼Œåœ¨è¿™é‡Œè¦æå–æ¯ä¸ªæ ·æœ¬çš„TPMå€¼</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">suppa_dir=/home/gongyuqi/miniconda3/envs/SUPPA2_3.9/bin/</span><br><span class="line">python <span class="variable">$suppa_dir</span>/multipleFieldSelection.py -i /home/gongyuqi/project/AS/SUPPA2/*/quant.sf -k 1 -f 4 -o /home/gongyuqi/project/AS/SUPPA2/iso_tpm.txt</span><br></pre></td></tr></table></figure>
<p>æŸ¥çœ‹iso_tpm.txtçš„å†…å®¹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/iso_tmp_txt.JPG"/></p>
<p><strong>(3)ã€ä½¿iso_tpm.txtæ–‡ä»¶ä¸­çš„è½¬å½•æœ¬idåŒä¸‹è½½çš„gtfæ–‡ä»¶idä¸€è‡´</strong>  </p>
<p>è¿è¡Œæ­¤Rè„šæœ¬ï¼Œä¼šç”Ÿæˆiso_tpm_formatted.txtæ–‡ä»¶</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Rscript format_Ensembl_ids.R iso_tpm.txt</span><br></pre></td></tr></table></figure>
<p>æŸ¥çœ‹iso_tpm_formatted.txtçš„å†…å®¹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/iso_tmp_formatted_txt.JPG"/></p>
<h2 id="ä¸‰ã€PSIè®¡ç®—"><a href="#ä¸‰ã€PSIè®¡ç®—" class="headerlink" title="ä¸‰ã€PSIè®¡ç®—"></a>ä¸‰ã€PSIè®¡ç®—</h2><p><strong>1ã€æ ¹æ®å‚è€ƒåŸºå› ç»„æ³¨é‡Šæ–‡ä»¶ç”Ÿæˆå¯å˜å‰ªåˆ‡äº‹ä»¶æ–‡ä»¶</strong></p>
<p><code>-i</code> GTFæ–‡ä»¶<br><code>-o</code> è¾“å‡ºæ–‡ä»¶å‰ç¼€<br><code>-e</code> è¾“å‡ºæ–‡ä»¶ä¸­åŒ…å«çš„å¯å˜å‰ªåˆ‡ç±»å‹<br><code>-f</code> è¾“å‡ºçš„æ ¼å¼  </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ç”Ÿæˆioeæ–‡ä»¶</span></span><br><span class="line">gtf_dir=/home/gongyuqi/ref/hg38/ensembl/SUPPA_ref</span><br><span class="line">python <span class="variable">$suppa_dir</span>/suppa.py generateEvents -i <span class="variable">$gtf_dir</span>/Homo_sapiens.GRCh37.75.formatted.gtf -o ensembl_hg19.events -e SE SS MX RI FL -f ioe</span><br><span class="line"><span class="comment">#åˆå¹¶æ‰€æœ‰çš„ioeæ–‡ä»¶</span></span><br><span class="line"><span class="built_in">cd</span> <span class="variable">$gtf_dir</span></span><br><span class="line">awk <span class="string">&#x27;</span></span><br><span class="line"><span class="string">    FNR==1 &amp;&amp; NR!=1 &#123; while (/^&lt;header&gt;/) getline; &#125;</span></span><br><span class="line"><span class="string">    1 &#123;print&#125;</span></span><br><span class="line"><span class="string">&#x27;</span> *.ioe &gt; ensembl_hg19.events.ioe</span><br></pre></td></tr></table></figure>
<p>ç»“æœå¦‚ä¸‹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ensembl_hg19.events.ioe.JPG"/></p>
<p><strong>2ã€è®¡ç®—æ ·æœ¬çš„PSIå€¼</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/SUPPA2</span><br><span class="line">ioe_merge_file=~/ref/hg38/ensembl/SUPPA_ref/hg19_event/ensembl_hg19.events.ioe</span><br><span class="line">nohup python <span class="variable">$suppa_dir</span>/suppa.py psiPerEvent -i <span class="variable">$ioe_merge_file</span> -e iso_tpm_formatted.txt -o TRA2_events &amp;</span><br></pre></td></tr></table></figure>
<p>æŸ¥çœ‹ç»“æœTRA2_events.psiæ–‡ä»¶<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/TRA2_events.psi.JPG"/></p>
<p><strong>3ã€boxplotå›¾å¯è§†åŒ–</strong></p>
<p><code>-i</code> è¾“å…¥çš„PSIçŸ©é˜µ<br><code>-e</code> éœ€è¦å¯è§†åŒ–çš„æŸåŸºå› çš„æŸç§å¯å˜å‰ªåˆ‡ç±»å‹<br><code>-g</code> æ ·å“æŒ‰é¡ºåºåˆ†ç»„<br><code>-c</code> åˆ†ç»„åç§°ï¼Œ1-3å¥½æ ·æœ¬ä¸ºNCç»„ï¼Œ4-6å·æ ·æœ¬ä¸ºKDç»„<br><code>-o</code> ç»“æœè¾“å‡ºåœ°å€  </p>
<p>ä»¥ENSG00000149554ä¸ºä¾‹è¿›è¡Œå¯è§†åŒ–</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">scripts_dir=/home/gongyuqi/project/AS/SUPPA2/scripts</span><br><span class="line">python <span class="variable">$scripts_dir</span>/generate_boxplot_event.py -i TRA2_events.psi -e <span class="string">&quot;ENSG00000149554;SE:chr11:125496728-125497502:125497725-125499127:+&quot;</span> -g 1-3,4-6 -c NC,KD -o ./</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/boxplot_PSI.JPG" width="50%">


<h2 id="å››ã€å·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ"><a href="#å››ã€å·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ" class="headerlink" title="å››ã€å·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ"></a>å››ã€å·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ</h2><p>åˆ†ç»„æƒ…å†µ<br>negative control siRNA ï¼šSRR1513329,SRR1513330,SRR1513331<br>TRA2A/B siRNA ï¼šSRR1513332,SRR1513333,SRR1513334  </p>
<p><strong>1ã€åˆ†åˆ«æ„å»ºä¸¤ç»„çš„TPMå’ŒPSIæ–‡ä»¶</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="variable">$scripts_dir</span>/split_file.R iso_tpm_formatted.txt SRR1513329,SRR1513330,SRR1513331 SRR1513332,SRR1513333,SRR1513334 TRA2_NC_iso.tpm TRA2_KD_iso.tpm -i</span><br><span class="line"><span class="variable">$scripts_dir</span>/split_file.R TRA2_events.psi SRR1513329,SRR1513330,SRR1513331 SRR1513332,SRR1513333,SRR1513334 TRA2_NC_events.psi TRA2_KD_events.psi -e</span><br></pre></td></tr></table></figure>
<p><strong>2ã€å·®å¼‚åˆ†æ</strong> </p>
<p><code>-m empirical</code> é€‰æ‹©empiricalæ–¹æ³•<br><code>-gc</code> åŸºå› ä¿®æ­£<br><code>--save_tpm_events</code> å‚æ•°çš„æ·»åŠ ï¼Œä¼šå¤šç”Ÿæˆä¸€ä¸ª<br>TRA2_diffSplice_avglogtpm.tabæ–‡ä»¶ï¼Œæ˜¯åç»­ç«å±±å›¾å¯è§†åŒ–çš„è¾“å…¥æ–‡ä»¶ä¹‹ä¸€</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python <span class="variable">$suppa_dir</span>/suppa.py diffSplice -m empirical -gc -i <span class="variable">$ioe_merge_file</span> --save_tpm_events -p TRA2_KD_events.psi TRA2_NC_events.psi -e TRA2_KD_iso.tpm TRA2_NC_iso.tpm -o ./<span class="built_in">test</span>/TRA2_diffSplice</span><br></pre></td></tr></table></figure>
<p>å·®å¼‚åˆ†æç”Ÿæˆä¸‹åˆ—3ä¸ªæ–‡ä»¶<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/diffsplice.JPG"/></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat TRA2_diffSplice.dpsi|grep <span class="string">&quot;ENSG00000149554&quot;</span>|less -N</span><br></pre></td></tr></table></figure>
<p>æŸ¥çœ‹dpsiæ–‡ä»¶ï¼Œä»¥ENSG00000149554ä¸ºä¾‹ï¼Œåˆ†æç»“æœä¸­ä¼šæ˜¾ç¤ºè¿™ä¸ªåŸºå› ä¸åŒçš„å¯å˜å‰ªåˆ‡ç±»å‹åœ¨ä¸¤ç§å¤„ç†æƒ…å†µä¸‹çš„psiå€¼åŠå…¶æ˜¾è‘—æ€§ï¼Œçº¢è‰²ç®­å¤´æ ‡è®°çš„å¯å˜å‰ªåˆ‡äº‹ä»¶åœ¨ä¸¤ç»„ä¸­çš„å·®å¼‚æ˜¯æ˜¾è‘—çš„(p_value&lt;0.05)<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/diffsplice_ENSG00000149554_mark.JPG"/></p>
<p><strong>å½“ç„¶,ä¸Šè¿°å¯å˜å‰ªåˆ‡å·®å¼‚åˆ†æçš„ç»“æœè¿˜å¯ä»¥å¯¼å…¥Rä¸­è¿›è¡Œvolcanoplotå¯è§†åŒ–åˆ†æï¼Œå®˜ç½‘æä¾›äº†å…·ä½“çš„Rè„šæœ¬</strong><br>å¯è§†åŒ–ç»“æœå¦‚ä¸‹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/volcano_plot.JPG" width="50%"></p>
<p><strong>æ€»ç»“ï¼š</strong><br>1ã€æŠ€æœ¯è·¯çº¿ï¼šsalmonå®šé‡è·å¾—å„ä¸ªæ ·æœ¬çš„TPMå€¼â€”â€”&gt; åˆå¹¶æ ·æœ¬â€”â€”&gt; è®¡ç®—PSIå€¼â€”â€”&gt; å¯å˜å‰ªåˆ‡å·®å¼‚åˆ†æ<br>2ã€æ‹¿åˆ°æœ€ç»ˆçš„å·®å¼‚åˆ†æç»“æœï¼Œæå–å‡ºæ˜¾è‘—çš„å·®å¼‚å¯å˜å‰ªåˆ‡äº‹ä»¶ï¼Œå†™è„šæœ¬å°†å¯å˜å‰ªåˆ‡äº‹ä»¶çš„idè½¬æ¢æˆåŸºå› åï¼Œå°±å¾—è§†è§‰ä¸Šæ›´å‹å¥½çš„æ–‡ä»¶äº†ã€‚æ‹¿åˆ°è¿™ä¸ªæ–‡ä»¶å¼€å±•åç»­çš„ä¸ªæ€§åŒ–åˆ†æã€‚</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/" data-id="ckl4ra1mh0013agvf4qiychia" data-title="SUPPA2" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-å¯å˜å‰ªåˆ‡ä¹‹MART" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/" class="article-date">
  <time class="dt-published" datetime="2021-01-14T03:05:01.000Z" itemprop="datePublished">2021-01-14</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="å¯å˜å‰ªåˆ‡ä¹‹rMART"><a href="#å¯å˜å‰ªåˆ‡ä¹‹rMART" class="headerlink" title="å¯å˜å‰ªåˆ‡ä¹‹rMART"></a>å¯å˜å‰ªåˆ‡ä¹‹rMART</h1><p><strong>å®˜æ–¹å‚è€ƒèµ„æ–™</strong><br><a target="_blank" rel="noopener" href="http://rnaseq-mats.sourceforge.net/">http://rnaseq-mats.sourceforge.net/</a><br><strong>rMATS</strong>:<a target="_blank" rel="noopener" href="https://github.com/Xinglab/rmats-turbo/blob/v4.1.0/README.md">https://github.com/Xinglab/rmats-turbo/blob/v4.1.0/README.md</a><br><strong>rmats2sashimiplot</strong>:<a target="_blank" rel="noopener" href="https://github.com/Xinglab/rmats2sashimiplot">https://github.com/Xinglab/rmats2sashimiplot</a></p>
<h2 id="ä¸€-rMATSè½¯ä»¶åŠç›¸å…³ä¾èµ–åŒ…çš„ä¸‹è½½"><a href="#ä¸€-rMATSè½¯ä»¶åŠç›¸å…³ä¾èµ–åŒ…çš„ä¸‹è½½" class="headerlink" title="(ä¸€)rMATSè½¯ä»¶åŠç›¸å…³ä¾èµ–åŒ…çš„ä¸‹è½½"></a>(ä¸€)rMATSè½¯ä»¶åŠç›¸å…³ä¾èµ–åŒ…çš„ä¸‹è½½</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å…ˆç”¨condaä¸‹è½½rMATsè½¯ä»¶éœ€è¦çš„ä¾èµ–åŒ…</span></span><br><span class="line"><span class="comment">#æœ‰æ—¶å€™pythonç‰ˆæœ¬é—®é¢˜æå¾—æˆ‘å¾ˆçƒ¦ï¼Œæœ‰äº›ç”Ÿä¿¡è½¯ä»¶ä¾èµ–ç‰¹å®šçš„pythonç‰ˆæœ¬ï¼Œæ‰€ä»¥åœ¨è½¯ä»¶å®‰è£…è¿‡ç¨‹ä¸­å¤šåŠ æ³¨æ„è¿™ä¸ªé—®é¢˜</span></span><br><span class="line">conda create rMATs2.7 python=2.7</span><br><span class="line">conda activate rMATs2.7</span><br><span class="line">conda install -y Cython=0.29.14 numpy=1.16.5 blas lapack gsl=2.6 gcc=5.4.0 CMake=3.15.4</span><br><span class="line">conda install libgfortran==1</span><br><span class="line">deactivate rMATs2.7</span><br><span class="line"></span><br><span class="line"><span class="comment">#ä¸‹è½½è½¯ä»¶å¹¶è§£å‹rMATsè½¯ä»¶</span></span><br><span class="line"><span class="comment">#rMATsä¸åŒç‰ˆæœ¬ä¸‹è½½ç½‘å€ï¼šhttps://github.com/Xinglab/rmats-turbo/releases/tag/v4.1.0</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/rMATS</span><br><span class="line">wget https://github.com/Xinglab/rmats-turbo/releases/download/v4.1.0/rmats_turbo_v4_1_0_python_2_7.tar.gz</span><br><span class="line">tar -zxvf rmats_turbo_v4_1_0_python_2_7.tar.gz</span><br><span class="line"><span class="comment">#æ°¸ä¹…æ·»åŠ è·¯å¾„åˆ°ç¯å¢ƒå˜é‡ä¸­</span></span><br><span class="line"><span class="built_in">export</span> PATH=/home/gongyuqi/project/AS/rMATS/rmats-turbo/:<span class="variable">$PATH</span></span><br><span class="line"><span class="built_in">source</span> ~/.bashrc</span><br></pre></td></tr></table></figure>

<h2 id="äºŒ-STARæ¯”å¯¹è½¯ä»¶çš„ä½¿ç”¨"><a href="#äºŒ-STARæ¯”å¯¹è½¯ä»¶çš„ä½¿ç”¨" class="headerlink" title="(äºŒ)STARæ¯”å¯¹è½¯ä»¶çš„ä½¿ç”¨"></a>(äºŒ)STARæ¯”å¯¹è½¯ä»¶çš„ä½¿ç”¨</h2><p>è¿™ä¹ˆå¤šæ¯”å¯¹è½¯ä»¶ï¼Œä¸ºä»€ä¹ˆé€‰STARï¼Ÿ<br>å› ä¸ºå®˜ç½‘ç”¨çš„STARï¼æ®è¯´å…¶ä»–å¯ä»¥åšå¯å˜å‰ªåˆ‡çš„æ¯”å¯¹è½¯ä»¶ç”Ÿæˆçš„bamæ–‡ä»¶ï¼ŒrMATsè¿è¡Œä¸äº†~~~</p>
<p><strong>1ã€æ„å»ºå‚è€ƒåŸºå› ç»„ç´¢å¼•</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#æ¿€æ´»æ¯”å¯¹çš„ç¯å¢ƒ,è¿™ä¸ªç¯å¢ƒé‡Œé¢æˆ‘ä¸‹è½½äº†å„ç§æ¯”å¯¹è½¯ä»¶ï¼Œpythonç‰ˆæœ¬ä¸º2.7.15.</span></span><br><span class="line">conda activate alignment2.7 </span><br><span class="line">nohup STAR --runThreadN 30 --runMode genomeGenerate \</span><br><span class="line">--genomeDir /home/gongyuqi/ref/hg38/index/star \</span><br><span class="line">--genomeFastaFiles /home/gongyuqi/ref/hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa \</span><br><span class="line">--sjdbGTFfile /home/gongyuqi/ref/hg38/Homo_sapiens.GRCh38.102.gtf &amp;</span><br></pre></td></tr></table></figure>
<p><strong>2ã€ç”¨STARè½¯ä»¶è¿›è¡Œæ¯”å¯¹</strong></p>
<p>STARçš„ä½¿ç”¨ä»¥åŠè¾“å‡ºæ–‡ä»¶è§£è¯»å‚è€ƒèµ„æ–™ï¼š<br><a target="_blank" rel="noopener" href="https://blog.csdn.net/yssxswl/article/details/105703869">https://blog.csdn.net/yssxswl/article/details/105703869</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/eca16bf2824e">https://www.jianshu.com/p/eca16bf2824e</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#è¿›å…¥fastqæ–‡ä»¶ååœ¨ç›®å½•ï¼Œè¿™é‡Œæ˜¯åŸå§‹çš„fastqæ•°æ®ï¼Œè´¨æ§ç»“æœæŒºå¥½çš„ï¼Œä¸ç”¨å†è¿‡æ»¤äº†ï¼Œæ‰€ä»¥æˆ‘ç›´æ¥æ¯”å¯¹äº†</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/rawdata</span><br><span class="line"><span class="comment">#å†™å¾ªç¯è¿›è¡Œæ¯”å¯¹</span></span><br><span class="line">ls *.gz|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup STAR --runThreadN 20 \</span><br><span class="line">--genomeDir /home/gongyuqi/ref/hg38/index/star \</span><br><span class="line">--readFilesCommand gunzip -c \</span><br><span class="line">--readFilesIn ./<span class="variable">$id</span> \</span><br><span class="line">--outSAMtype BAM SortedByCoordinate \</span><br><span class="line">--outFileNamePrefix /home/gongyuqi/project/AS/aligndata_star/<span class="variable">$&#123;id%%.*&#125;</span>. &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p><strong>3ã€æ¯”å¯¹æƒ…å†µç»Ÿè®¡å¹¶æŸ¥çœ‹</strong></p>
<p>(1)ç›´æ¥æŸ¥çœ‹</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.Log.progress.out|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> sed -n <span class="string">&#x27;1,3p&#x27;</span> <span class="variable">$id</span>;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>ç»“æœå¦‚ä¸‹ï¼Œæ¯”å¯¹æƒ…å†µè¿˜æ˜¯ä¸é”™çš„ï¼Œå¯ä»¥è¿›è¡Œåç»­çš„åˆ†æã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/star_align_Log.progress.out.JPG"/></p>
<p>(2)flagstatç»Ÿè®¡æ¯”å¯¹æƒ…å†µæŸ¥çœ‹æ¯”å¯¹æƒ…å†µ</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup samtools flagstat -@ 20 <span class="variable">$id</span> &gt; <span class="variable">$&#123;id%%.*&#125;</span>.flagstat.txt &amp;</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line">ls *.txt|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> cat <span class="variable">$id</span> | grep <span class="string">&quot;mapped (&quot;</span>;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>è¿™ä¸ªæ¯”å¯¹ç»“æœå®åœ¨æ˜¯éš¾ä»¥ç½®ä¿¡ï¼ï¼ä½†æ˜¯ç»“åˆä¸Šé¢çš„æ¯”å¯¹ç»“æœï¼Œè¿˜æ˜¯å¯ä»¥ç›¸ä¿¡æ¯”å¯¹è¿™ä¸€æ­¥æ˜¯æ²¡æœ‰é—®é¢˜çš„ã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/star_align_flagstat.JPG"/></p>
<h2 id="ä¸‰-è¿è¡ŒrMATs"><a href="#ä¸‰-è¿è¡ŒrMATs" class="headerlink" title="(ä¸‰)è¿è¡ŒrMATs"></a>(ä¸‰)è¿è¡ŒrMATs</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#é¦–å…ˆæˆ‘ä»¬è¦å¤±æ´»æ‰æ¯”å¯¹çš„ç¯å¢ƒï¼ˆalignment2.7ï¼‰ï¼Œæ¿€æ´»rMATsç¯å¢ƒ</span></span><br><span class="line">conda deactivate alignment2.7</span><br><span class="line">conda activate rMATs2.7</span><br><span class="line"><span class="comment">#è™½ç„¶rMATSè·¯å¾„è¢«æ°¸ä¹…æ·»åŠ åˆ°ç¯å¢ƒå˜é‡ä¸­ï¼Œä½†æ˜¯PATHåªè®¤å¯æ‰§è¡Œæ–‡ä»¶ï¼Œæ‰€ä»¥è¿™ä¸€æ­¥è¿˜æ˜¯è¦åˆ‡æ¢åˆ°rMATsè·¯å¾„ä¸‹</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/rMATS/rmats-turbo</span><br><span class="line">input_dir=/home/gongyuqi/project/AS/aligndata_star</span><br><span class="line">python rmats.py \</span><br><span class="line">--b1 <span class="variable">$input_dir</span>/b1.txt \</span><br><span class="line">--b2 <span class="variable">$input_dir</span>/b2.txt \</span><br><span class="line">--gtf /home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gtf \</span><br><span class="line">-t single --readLength 36 --nthread 20 \</span><br><span class="line">--od /home/gongyuqi/project/AS/rMATS/rmats_output \</span><br><span class="line">--tmp /home/gongyuqi/project/AS/rMATS/rmats_tmp</span><br><span class="line"><span class="comment">#è¿è¡Œå®Œè®°å¾—é€€å‡ºå½“å‰ç¯å¢ƒï¼Œå› ä¸ºåé¢æˆ‘ä»¬è¦ç”¨å¦å¤–ä¸€ä¸ªç¯å¢ƒ</span></span><br><span class="line">conda deactivate rMATs2.7</span><br></pre></td></tr></table></figure>
<p>è¿è¡Œä¸Šé¢çš„ä»£ç ï¼Œä¼šæŠ¥å¦‚ä¸‹é”™è¯¯ï¼š<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/erro_libgsl.so.25.JPG"/></p>
<p>æˆ‘æŸ¥çœ‹äº†ä¸€ä¸‹ï¼Œ/home/gongyuqi/miniconda3/envs/rMATs2.7/libè·¯å¾„ä¸‹æœ‰è¿™ä¸ªæ–‡ä»¶å‘¢ï¼ä¸ºä»€ä¹ˆç”µè„‘æ‰¾ä¸åˆ°ï¼Ÿè§¦åŠåˆ°äº†æˆ‘çš„çŸ¥è¯†ç›²åŒºäº†~~~<br>æŸ¥äº†ä¸€ä¸‹åˆçš„èµ„æ–™ï¼Œå‘ç°è¿™ä¸ªæŠ¥é”™æ¶‰åŠåˆ°æŸ¥æ‰¾å…±äº«åº“ï¼ˆåŠ¨æ€é“¾æ¥åº“ï¼‰é—®é¢˜ï¼Œæ ¹æ®æŸ¥åˆ°çš„ä¸œè¥¿æ“ä½œäº†ä¸€æ³¢å±…ç„¶æ²¡æœ‰è§£å†³ï¼Œæ—è¾¹çš„é›¶é£Ÿéƒ½ä¸é¦™äº†ï¼ğŸ˜«<br>ç®—äº†~~~æˆ‘è¿˜æ˜¯å…ˆå»åƒæ™šé¥­å§ï¼Œå·²ç»é¥¿å¾—å¾ˆç˜¦äº†ï¼ğŸ˜«<br>ã€‚ã€‚ã€‚ã€‚ã€‚ã€‚<br>å¥½å•¦ï¼Œæˆ‘åƒå®Œé¥­å›æ¥äº†ã€‚æˆ‘ç»§ç»­å€’è…¾ï¼Œä¸æ”¾å¼ƒã€‚<br>ã€‚ã€‚ã€‚ã€‚ã€‚ã€‚<br>é¢ï¼Œæˆ‘çŸ¥é“äº†ï¼Œè¿™æ®µä»£ç è¿è¡Œè¿‡ç¨‹ä¸­è¿˜æ¶‰åŠåˆ°gccç¼–è¯‘ï¼Œå› ä¸ºå®ƒä¾èµ–åŒ…ä¸­æœ‰GCCã€‚<br>è¿™ä¸¤ä¸ªé—®é¢˜å®åœ¨æ˜¯å¤ªä¸“ä¸šäº†ï¼Œä»Šå¤©åˆå­¦ä¹ åˆ°äº†ã€‚ç°åœ¨æœ‰ä¸€ç‚¹å°å°çš„å¼€å¿ƒï¼ğŸ˜Š<br>è§£å†³æ–¹æ³•å‚è€ƒèµ„æ–™ï¼š<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/a62e1d327023">https://www.jianshu.com/p/a62e1d327023</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Linuxé»˜è®¤åªä¼šåœ¨æŒ‡å®šçš„å‡ ä¸ªç›®å½•æ‰¾å…±äº«åº“ï¼ˆåŠ¨æ€é“¾æ¥åº“ï¼‰ï¼Œå…¶ä»–ç›®å½•çš„è¦è‡ªå·±åŠ åœ¨LD_LIBRARY_PATH</span></span><br><span class="line"><span class="built_in">export</span> LD_LIBRARY_PATH=/home/gongyuqi/miniconda3/envs/rMATs2.7/lib</span><br><span class="line"><span class="comment">#å› ä¸ºä¸Šè¿°å‘½ä»¤æ˜¯ä¾èµ–gccçš„ï¼Œéœ€è¦gccè¿›è¡Œç¼–è¯‘ï¼Œæ‰€ä»¥è¿˜éœ€è¦ä¸‹é¢çš„æ“ä½œ</span></span><br><span class="line"><span class="built_in">export</span> LIBRARY_PATH=/home/gongyuqi/miniconda3/envs/rMATs2.7/lib</span><br><span class="line"><span class="comment">#æ³¨æ„ï¼Œä¸Šè¿°éƒ½æ˜¯ä¸´æ—¶çš„ï¼Œæ²»æ ‡ä¸æ²»æœ¬ï¼Œä¸‹æ¬¡æ‰§è¡Œè¿è¡ŒrMATsæ—¶ï¼Œä¾ç„¶è¦é‡å¤ä¸Šè¿°æ“ä½œã€‚å½“ç„¶ä½ ä¹Ÿå¯ä»¥å°†å…¶æ°¸ä¹…æ€§æ·»åŠ ã€‚</span></span><br></pre></td></tr></table></figure>
<p>è¿è¡Œç»“æœå¦‚ä¸‹ï¼Œå…³äºè¿™äº›ç»“æœæ–‡ä»¶çš„è§£é‡Šè¯´æ˜ï¼Œå¯ä»¥å‚è€ƒrMATSå®˜ç½‘ï¼Œä¸Šé¢æœ‰å¾ˆè¯¦ç»†çš„é˜è¿°ã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/rMATs_output.JPG"/></p>
<h2 id="å››-rmats2sashimiplotå¯è§†åŒ–"><a href="#å››-rmats2sashimiplotå¯è§†åŒ–" class="headerlink" title="(å››)rmats2sashimiplotå¯è§†åŒ–"></a>(å››)rmats2sashimiplotå¯è§†åŒ–</h2><p>ç»ˆäºåˆ°äº†æ¿€åŠ¨äººå¿ƒçš„å¯è§†åŒ–é˜¶æ®µäº†ï¼Œä¸€é¡¿æ“ä½œä¹‹åï¼Œäº§ç”Ÿä¸€å¼ æ¼‚äº®çš„å›¾æ”¾PPTä¸Šè¿˜æ˜¯å¾ˆæ¬£æ…°çš„ã€‚</p>
<p><strong>1ã€ç¯å¢ƒæ­å»º</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">conda create -n rmats2sashimiplot python=2.7 </span><br><span class="line">conda activate rmats2sashimiplot</span><br><span class="line">conda install -y numpy scipy matplotlib pysam samtools bedtools</span><br></pre></td></tr></table></figure>
<p><strong>2ã€å‡†å¤‡å¯è§†åŒ–çš„txtæ–‡ä»¶</strong>  </p>
<p>ä»¥Exon Skippingä¸ºä¾‹æ¼”ç¤º,è¿™é‡Œæˆ‘éšä¾¿æ‰¾ä¸€ä¸ªåŸºå› è¿›è¡Œå¯è§†åŒ–ï¼Œæ³¨æ„äº†ï¼Œè¿™ä¸ªåŸºå› ä¸ä¸€å®šæ˜¯å·®å¼‚å¯å˜å‰ªåˆ‡çš„åŸºå› å“Ÿï¼Œå› ä¸ºæ˜¯æˆ‘éšä¾¿æ‰¾çš„ä¸€ä¸ªã€‚å˜»å˜»ğŸ¤­</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">sed -n <span class="string">&#x27;2p&#x27;</span> SE.MATS.JC.txt &gt; SE.plot.PRMT2.txt</span><br><span class="line">cat SE.plot.PRMT2.txt</span><br></pre></td></tr></table></figure>
<p>æ–‡ä»¶å†…å®¹å¦‚ä¸‹ï¼š<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PRMT2_rMATs.txt.JPG"/></p>
<p><strong>3ã€å¯è§†åŒ–çš„txtæ–‡ä»¶</strong></p>
<h6 id="ä¸€å¼€å§‹ï¼Œæˆ‘éå¸¸è‡ªä¿¡çš„è¿è¡Œäº†ä¸‹é¢çš„ä»£ç "><a href="#ä¸€å¼€å§‹ï¼Œæˆ‘éå¸¸è‡ªä¿¡çš„è¿è¡Œäº†ä¸‹é¢çš„ä»£ç " class="headerlink" title="ä¸€å¼€å§‹ï¼Œæˆ‘éå¸¸è‡ªä¿¡çš„è¿è¡Œäº†ä¸‹é¢çš„ä»£ç ~"></a>ä¸€å¼€å§‹ï¼Œæˆ‘éå¸¸è‡ªä¿¡çš„è¿è¡Œäº†ä¸‹é¢çš„ä»£ç ~</h6><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">gff3=/home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gff3</span><br><span class="line">output_dir=/home/gongyuqi/project/AS/rMATS/rmats2sashimiplot/</span><br><span class="line">rmats2sashimiplot \</span><br><span class="line">--b1 SRR974968.Aligned.sortedByCoord.out.bam,SRR974972.Aligned.sortedByCoord.out.bam,SRR974978.Aligned.sortedByCoord.out.bam,SRR974984.Aligned.sortedByCoord.out.bam \</span><br><span class="line">--b2 SRR974974.Aligned.sortedByCoord.out.bam,SRR974976.Aligned.sortedByCoord.out.bam,SRR974980.Aligned.sortedByCoord.out.bam,SRR974982.Aligned.sortedByCoord.out.bam \</span><br><span class="line">-c chr21:+:46636438:46636547:<span class="variable">$gff3</span> \</span><br><span class="line">-e /home/gongyuqi/project/AS/rMATS/rmats_output/SE.plot.PRMT2.txt \</span><br><span class="line">--l1 treatment --l2 control \</span><br><span class="line">--exon_s 1 --intron_s 5 -t SE \</span><br><span class="line">-o <span class="variable">$output_dir</span>/<span class="built_in">test</span></span><br></pre></td></tr></table></figure>
<h6 id="ä½†æ˜¯ï¼Œæˆ‘é‡åˆ°äº†å¦‚ä¸‹æŠ¥é”™"><a href="#ä½†æ˜¯ï¼Œæˆ‘é‡åˆ°äº†å¦‚ä¸‹æŠ¥é”™" class="headerlink" title="ä½†æ˜¯ï¼Œæˆ‘é‡åˆ°äº†å¦‚ä¸‹æŠ¥é”™"></a>ä½†æ˜¯ï¼Œæˆ‘é‡åˆ°äº†å¦‚ä¸‹æŠ¥é”™</h6><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/rmats2sashimiplot_chr_error.JPG"/>


<h6 id="æˆ‘å›å¤´çœ‹äº†ä¸€ä¸‹æˆ‘çš„bamæ–‡ä»¶ï¼ŒæŸ“è‰²ä½“æ˜¯ä»¥1-2â€¦X-Yå½¢å¼ä½“ç°è€Œä¸æ˜¯chr1-chr2â€¦chrX-chrYå½¢å¼ä½“ç°ã€‚æˆ‘ç¬¬ä¸€æ¬¡é‡åˆ°æŸ“è‰²ä½“å·çš„é—®é¢˜å‘¢ã€‚æˆ‘è¦æ€ä¹ˆè®©bamæ–‡ä»¶ä¸­æœ‰chrå­˜åœ¨ä¸”è®©å…¶å­˜åœ¨åœ¨æ­£ç¡®çš„ä½ç½®å‘¢ã€‚"><a href="#æˆ‘å›å¤´çœ‹äº†ä¸€ä¸‹æˆ‘çš„bamæ–‡ä»¶ï¼ŒæŸ“è‰²ä½“æ˜¯ä»¥1-2â€¦X-Yå½¢å¼ä½“ç°è€Œä¸æ˜¯chr1-chr2â€¦chrX-chrYå½¢å¼ä½“ç°ã€‚æˆ‘ç¬¬ä¸€æ¬¡é‡åˆ°æŸ“è‰²ä½“å·çš„é—®é¢˜å‘¢ã€‚æˆ‘è¦æ€ä¹ˆè®©bamæ–‡ä»¶ä¸­æœ‰chrå­˜åœ¨ä¸”è®©å…¶å­˜åœ¨åœ¨æ­£ç¡®çš„ä½ç½®å‘¢ã€‚" class="headerlink" title="æˆ‘å›å¤´çœ‹äº†ä¸€ä¸‹æˆ‘çš„bamæ–‡ä»¶ï¼ŒæŸ“è‰²ä½“æ˜¯ä»¥1,2â€¦X,Yå½¢å¼ä½“ç°è€Œä¸æ˜¯chr1,chr2â€¦chrX,chrYå½¢å¼ä½“ç°ã€‚æˆ‘ç¬¬ä¸€æ¬¡é‡åˆ°æŸ“è‰²ä½“å·çš„é—®é¢˜å‘¢ã€‚æˆ‘è¦æ€ä¹ˆè®©bamæ–‡ä»¶ä¸­æœ‰chrå­˜åœ¨ä¸”è®©å…¶å­˜åœ¨åœ¨æ­£ç¡®çš„ä½ç½®å‘¢ã€‚"></a>æˆ‘å›å¤´çœ‹äº†ä¸€ä¸‹æˆ‘çš„bamæ–‡ä»¶ï¼ŒæŸ“è‰²ä½“æ˜¯ä»¥1,2â€¦X,Yå½¢å¼ä½“ç°è€Œä¸æ˜¯chr1,chr2â€¦chrX,chrYå½¢å¼ä½“ç°ã€‚æˆ‘ç¬¬ä¸€æ¬¡é‡åˆ°æŸ“è‰²ä½“å·çš„é—®é¢˜å‘¢ã€‚æˆ‘è¦æ€ä¹ˆè®©bamæ–‡ä»¶ä¸­æœ‰<code>chr</code>å­˜åœ¨ä¸”è®©å…¶å­˜åœ¨åœ¨æ­£ç¡®çš„ä½ç½®å‘¢ã€‚</h6><h6 id="æˆ‘ç»å°½è„‘æ±çš„æƒ³äº†å¾ˆä¹…å¾ˆä¹…ğŸ˜«ï¼Œæˆ‘è¿˜è€ƒè™‘åˆ°æ˜¯ä¸æ˜¯æˆ‘å‚è€ƒåŸºå› ç»„çš„é—®é¢˜ï¼Œä¸åŒæ•°æ®åº“å‚è€ƒåŸºå› ç»„æ˜¯ä¸æ˜¯æœ‰å·®å¼‚ï¼Œæˆ‘è¦ä¸è¦æ¢ä¸€ä¸ªæ•°æ®åº“çš„å‚è€ƒåŸºå› ç»„ã€‚ç„¶åæˆ‘ä¸€é¡¿æ“ä½œï¼Œæ— æœ-ç®—äº†ï¼Œæˆ‘æ‰“ä¸€å±€æ¸¸æˆç¡è§‰å§ï¼Œæ˜å¤©å†æƒ³ã€‚"><a href="#æˆ‘ç»å°½è„‘æ±çš„æƒ³äº†å¾ˆä¹…å¾ˆä¹…ğŸ˜«ï¼Œæˆ‘è¿˜è€ƒè™‘åˆ°æ˜¯ä¸æ˜¯æˆ‘å‚è€ƒåŸºå› ç»„çš„é—®é¢˜ï¼Œä¸åŒæ•°æ®åº“å‚è€ƒåŸºå› ç»„æ˜¯ä¸æ˜¯æœ‰å·®å¼‚ï¼Œæˆ‘è¦ä¸è¦æ¢ä¸€ä¸ªæ•°æ®åº“çš„å‚è€ƒåŸºå› ç»„ã€‚ç„¶åæˆ‘ä¸€é¡¿æ“ä½œï¼Œæ— æœ-ç®—äº†ï¼Œæˆ‘æ‰“ä¸€å±€æ¸¸æˆç¡è§‰å§ï¼Œæ˜å¤©å†æƒ³ã€‚" class="headerlink" title="æˆ‘ç»å°½è„‘æ±çš„æƒ³äº†å¾ˆä¹…å¾ˆä¹…ğŸ˜«ï¼Œæˆ‘è¿˜è€ƒè™‘åˆ°æ˜¯ä¸æ˜¯æˆ‘å‚è€ƒåŸºå› ç»„çš„é—®é¢˜ï¼Œä¸åŒæ•°æ®åº“å‚è€ƒåŸºå› ç»„æ˜¯ä¸æ˜¯æœ‰å·®å¼‚ï¼Œæˆ‘è¦ä¸è¦æ¢ä¸€ä¸ªæ•°æ®åº“çš„å‚è€ƒåŸºå› ç»„ã€‚ç„¶åæˆ‘ä¸€é¡¿æ“ä½œï¼Œæ— æœ~~~ç®—äº†ï¼Œæˆ‘æ‰“ä¸€å±€æ¸¸æˆç¡è§‰å§ï¼Œæ˜å¤©å†æƒ³ã€‚"></a>æˆ‘ç»å°½è„‘æ±çš„æƒ³äº†å¾ˆä¹…å¾ˆä¹…ğŸ˜«ï¼Œæˆ‘è¿˜è€ƒè™‘åˆ°æ˜¯ä¸æ˜¯æˆ‘å‚è€ƒåŸºå› ç»„çš„é—®é¢˜ï¼Œä¸åŒæ•°æ®åº“å‚è€ƒåŸºå› ç»„æ˜¯ä¸æ˜¯æœ‰å·®å¼‚ï¼Œæˆ‘è¦ä¸è¦æ¢ä¸€ä¸ªæ•°æ®åº“çš„å‚è€ƒåŸºå› ç»„ã€‚ç„¶åæˆ‘ä¸€é¡¿æ“ä½œï¼Œæ— æœ~~~ç®—äº†ï¼Œæˆ‘æ‰“ä¸€å±€æ¸¸æˆç¡è§‰å§ï¼Œæ˜å¤©å†æƒ³ã€‚</h6><h6 id="èµ·åºŠåƒæ—©é¥­éª‘è½¦å»å®éªŒå®¤æ€è€ƒæŸ¥èµ„æ–™æœ‰ä¸€ç‚¹å°æƒ³æ³•äº†æˆ‘å®è·µä¸€ä¸‹çœ‹çœ‹å¯ä¸å¯"><a href="#èµ·åºŠåƒæ—©é¥­éª‘è½¦å»å®éªŒå®¤æ€è€ƒæŸ¥èµ„æ–™æœ‰ä¸€ç‚¹å°æƒ³æ³•äº†æˆ‘å®è·µä¸€ä¸‹çœ‹çœ‹å¯ä¸å¯" class="headerlink" title="èµ·åºŠåƒæ—©é¥­éª‘è½¦å»å®éªŒå®¤æ€è€ƒæŸ¥èµ„æ–™æœ‰ä¸€ç‚¹å°æƒ³æ³•äº†æˆ‘å®è·µä¸€ä¸‹çœ‹çœ‹å¯ä¸å¯"></a>èµ·åºŠ<del>åƒæ—©é¥­</del>éª‘è½¦å»å®éªŒå®¤<del>æ€è€ƒ</del>æŸ¥èµ„æ–™<del>æœ‰ä¸€ç‚¹å°æƒ³æ³•äº†</del>æˆ‘å®è·µä¸€ä¸‹çœ‹çœ‹å¯ä¸å¯</h6><p>å‚è€ƒèµ„æ–™ï¼š<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/5ceda0350d0d">https://www.jianshu.com/p/5ceda0350d0d</a></p>
<h6 id="ç¬¬ä¸€æ­¥ï¼šç»™bamæ–‡ä»¶æ·»åŠ chr"><a href="#ç¬¬ä¸€æ­¥ï¼šç»™bamæ–‡ä»¶æ·»åŠ chr" class="headerlink" title="ç¬¬ä¸€æ­¥ï¼šç»™bamæ–‡ä»¶æ·»åŠ chr"></a><font color=blue>ç¬¬ä¸€æ­¥ï¼šç»™bamæ–‡ä»¶æ·»åŠ chr</font></h6><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#æ·»åŠ chr</span></span><br><span class="line">ls *.bam|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup samtools view -h <span class="variable">$id</span> | sed -e <span class="string">&#x27;/^@SQ/s/SN\:/SN\:chr/&#x27;</span> -e <span class="string">&#x27;/^[^@]/s/\t/\tchr/2&#x27;</span>|awk -F <span class="string">&#x27; &#x27;</span> <span class="string">&#x27;$7=($7==&quot;=&quot; || $7==&quot;*&quot;?$7:sprintf(&quot;chr%s&quot;,$7))&#x27;</span> |tr <span class="string">&quot; &quot;</span> <span class="string">&quot;\t&quot;</span> | samtools view -h -b -@ 10 -S - &gt; <span class="variable">$&#123;id%%.*&#125;</span>.Aligned.sortedByCoord.out.chr.bam &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h6 id="ç¬¬äºŒæ­¥ï¼šä½¿ç”¨æ·»åŠ chrçš„bamæ–‡ä»¶æ‰§è¡Œç¨‹åº"><a href="#ç¬¬äºŒæ­¥ï¼šä½¿ç”¨æ·»åŠ chrçš„bamæ–‡ä»¶æ‰§è¡Œç¨‹åº" class="headerlink" title="ç¬¬äºŒæ­¥ï¼šä½¿ç”¨æ·»åŠ chrçš„bamæ–‡ä»¶æ‰§è¡Œç¨‹åº"></a><font color=blue>ç¬¬äºŒæ­¥ï¼šä½¿ç”¨æ·»åŠ chrçš„bamæ–‡ä»¶æ‰§è¡Œç¨‹åº</font></h6><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">gff3=/home/gongyuqi/ref/hg38/ensembl/Homo_sapiens.GRCh38.102.gff3</span><br><span class="line">output_dir=/home/gongyuqi/project/AS/rMATS/rmats2sashimiplot/</span><br><span class="line">rmats2sashimiplot \</span><br><span class="line">--b1 SRR974968.Aligned.sortedByCoord.out.chr.bam,SRR974972.Aligned.sortedByCoord.out.chr.bam,SRR974978.Aligned.sortedByCoord.out.chr.bam,SRR974984.Aligned.sortedByCoord.out.chr.bam \</span><br><span class="line">--b2 SRR974974.Aligned.sortedByCoord.out.chr.bam,SRR974976.Aligned.sortedByCoord.out.chr.bam,SRR974980.Aligned.sortedByCoord.out.chr.bam,SRR974982.Aligned.sortedByCoord.out.chr.bam \</span><br><span class="line">-c chr21:+:46636438:46636547:<span class="variable">$gff3</span> \</span><br><span class="line">-e /home/gongyuqi/project/AS/rMATS/rmats_output/SE.plot.PRMT2.txt \</span><br><span class="line">--l1 treatment --l2 control \</span><br><span class="line">--exon_s 1 --intron_s 5 -t SE \</span><br><span class="line">-o <span class="variable">$output_dir</span>/SE.plot</span><br></pre></td></tr></table></figure>
<p>å¯è§†åŒ–ç»“æœå¦‚ä¸‹ï¼ˆä½ ä¼šå‘ç°ï¼Œå’¦~ä¸¤ç»„é—´å·®å¼‚ä¸æ˜¾è‘—å‘€ï¼Ÿå› ä¸ºæˆ‘éšä¾¿æŒ‘äº†ä¸€ä¸ªåŸºå› å¯è§†åŒ–å‘€ï¼æˆ‘æœ‰å·å·å¯è§†åŒ–æŸä¸ªåŸºå› ï¼Œå¯è§†åŒ–å‡ºæ¥çš„å·®å¼‚æ€§ä¸æˆ‘å®šé‡å’ŒåŠå®šé‡qPCRçš„ç»“æœä¸€è‡´å“Ÿï¼Œæ‰€ä»¥ä¸Šè¿°åˆ†ææµç¨‹åº”è¯¥æ˜¯OKçš„~~~å˜»å˜»å˜»ğŸ¤­ä¸ç®¡æ€æ ·ï¼Œè¿™ä¸ªåŸºå› ç¡®å®å­˜åœ¨Exon Skippingã€‚ï¼‰<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ABC_GCB_PRMT2_rMATs.JPG"/></p>
<h3 id="æ€»ç»“"><a href="#æ€»ç»“" class="headerlink" title="æ€»ç»“"></a>æ€»ç»“</h3><p><strong>æŠ€æœ¯ä¸Š</strong><br>1ã€éœ€è¦æ­å»ºä¸¤ä¸ªé‡è¦çš„ç¯å¢ƒï¼šè¿è¡ŒrMATsçš„ç¯å¢ƒï¼Œè¿è¡Œrmats2sashimiplotçš„ç¯å¢ƒã€‚<br>2ã€é€‰ç”¨STARè¿›è¡Œå‚è€ƒåŸºå› ç»„çš„æ¯”å¯¹ï¼ˆæ³¨æ„äº†ï¼Œå‚è€ƒåŸºå› ç»„å’Œæ³¨é‡Šæ–‡ä»¶çš„ç‰ˆæœ¬è¦ä¸€è‡´å“Ÿï¼‰ã€‚<br>3ã€åœ¨rMATsçš„ç¯å¢ƒä¸‹,ç»“åˆä¸Šè¿°ç”Ÿæˆçš„bamæ–‡ä»¶ï¼Œè¿è¡ŒrMATsè½¯ä»¶ï¼Œç”Ÿæˆå¯å˜å‰ªåˆ‡äº‹ä»¶çš„æ–‡ä»¶ã€‚<br>4ã€åœ¨rmats2sashimiplotç¯å¢ƒä¸‹ï¼Œå¯è§†åŒ–ä¸Šè¿°ç”Ÿæˆçš„å¯å˜å‰ªåˆ‡äº‹ä»¶ã€‚</p>
<p><strong>å¿ƒæ€ä¸Š</strong><br>ä¸Šåˆè§£å†³ä¸äº†å°±ä¸‹åˆè§£å†³ï¼Œä»Šå¤©è§£å†³ä¸äº†å°±æ˜å¤©è§£å†³ã€‚è¿™ä¸ªæ–¹æ³•ä¸è¡Œå°±æ¢ä¸ªæ–¹æ³•ã€‚æ€è€ƒè‰¯ä¹…ä¹‹åå¯ä»¥é€‚å½“çš„é—®é—®è€å¸ˆã€‚æœ€å…³é”®çš„è¿˜æ˜¯è¦æŒ‰æ—¶åƒé¥­æŒ‰æ—¶ç¡è§‰ã€‚</p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/" data-id="ckl4ra1mf0010agvf9az82jph" data-title="rMART" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-å¯å˜å‰ªåˆ‡ä¹‹MISO" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/" class="article-date">
  <time class="dt-published" datetime="2021-01-09T06:05:52.000Z" itemprop="datePublished">2021-01-09</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>


<h1 id="å¯å˜å‰ªåˆ‡åˆ†æä¹‹MISO"><a href="#å¯å˜å‰ªåˆ‡åˆ†æä¹‹MISO" class="headerlink" title="å¯å˜å‰ªåˆ‡åˆ†æä¹‹MISO"></a>å¯å˜å‰ªåˆ‡åˆ†æä¹‹MISO</h1><p>MISOè½¯ä»¶åŸæ–‡æ¡£ï¼š<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso">https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso</a></p>
<h2 id="æ•°æ®ä¸‹è½½"><a href="#æ•°æ®ä¸‹è½½" class="headerlink" title="æ•°æ®ä¸‹è½½"></a>æ•°æ®ä¸‹è½½</h2><ul>
<li>å†™ä¸€ä¸ªbashè„šæœ¬ä¸‹è½½æ•°æ®<br>ä¸€æ¬¡æ€§ä¸‹è½½æ‰€æœ‰çš„.fastq.gzæ–‡ä»¶ï¼Œæœ¬æ¬¡å®éªŒæ•°æ®æ˜¯å•ç«¯æµ‹åºæ•°æ®<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ï¼/bin/bash</span></span><br><span class="line">dir=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line"><span class="keyword">for</span> id <span class="keyword">in</span>  &#123;84,78,76,72,82,80,74,68&#125; </span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line">       nohup ascp -QT -l 300m -P33001 -i <span class="variable">$dsa</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR974/SRR9749<span class="variable">$id</span>/SRR9749<span class="variable">$id</span>.fastq.gz . &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>æ‰§è¡Œè„šæœ¬</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chmod +x download.sh</span><br><span class="line">nohup ./download.sh &amp;</span><br></pre></td></tr></table></figure>
<h2 id="æ¯”å¯¹"><a href="#æ¯”å¯¹" class="headerlink" title="æ¯”å¯¹"></a>æ¯”å¯¹</h2><p>å‚è€ƒèµ„æ–™ï¼š<a target="_blank" rel="noopener" href="http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml">http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml</a></p>
<blockquote>
<ul>
<li>tophatè¿›è¡Œè½¬å½•ç»„çš„æ¯”å¯¹æ—¶ï¼Œè¾“å‡ºçš„æ˜¯bamï¼ˆäºŒè¿›åˆ¶çš„samï¼‰æ–‡ä»¶ã€‚</li>
<li>tophatä¾èµ–bowtie2ï¼ˆbowtie1ä¹Ÿå¯ä»¥ï¼‰ï¼Œå‚è€ƒåŸºå› ç»„ç´¢å¼•ç”¨bowtie2-buildè¿›è¡Œæ„å»ºã€‚</li>
<li>å¦‚æœFASTAæ ¼å¼çš„åŸºå› ç»„ä¸ç´¢å¼•æ–‡ä»¶ä¸åœ¨åŒä¸€ä¸ªæ–‡ä»¶å¤¹ä¸‹é¢ï¼Œtophatä¼šåœ¨è¿è¡Œä¸­è‡ªåŠ¨ç”Ÿæˆï¼Œæ‰€ä»¥è®°å¾—æŠŠå‚è€ƒåŸºå› ç»„å’Œç´¢å¼•æ–‡ä»¶æ”¾åœ¨åŒä¸€ä¸ªæ–‡ä»¶å¤¹ä¸‹é¢ï¼Œä»¥å‡å°‘ç³»ç»Ÿè¿è¡Œæ—¶çš„é«˜æ¶ˆè€—ã€‚</li>
<li>å¦‚æœGTFæˆ–è€…GFFæ ¼å¼çš„åŸºå› ç»„æ³¨é‡Šæ–‡ä»¶å­˜åœ¨ï¼Œ tophatä¼šä¼˜å…ˆæ¯”å¯¹åˆ°æ³¨é‡Šæ–‡ä»¶çš„è½¬å½•ç»„ä¸Šã€‚æœ€å¥½äº‹å…ˆå‡†å¤‡å¥½è½¬å½•ç»„ç´¢å¼•ä»¥èŠ‚çº¦åç»­çš„æ¯”å¯¹æ—¶é—´ã€‚</li>
<li>å¦‚æœæ²¡æœ‰äº‹å…ˆå‡†å¤‡å¥½è½¬å½•ç»„ç´¢å¼•ï¼Œtophatä¼šåœ¨è¿è¡Œè¿‡ç¨‹ä¸­ç”Ÿæˆè½¬å½•ç»„ç´¢å¼•ï¼Œè¿™æ ·å°±æ¯”è¾ƒè€—æ—¶è€—åŠ›äº†ã€‚</li>
</ul>
</blockquote>
<p><strong><table><tr><td bgcolor=GreenYellow>**å‚è€ƒåŸºå› ç»„ç´¢å¼•æ„å»ºã€è½¬å½•ç»„ç´¢å¼•æ„å»º</strong></td></tr></table>**</p>
<p>hg19å‚è€ƒåŸºå› ç»„gtfæ–‡ä»¶ä¸‹è½½é“¾æ¥ï¼š<a target="_blank" rel="noopener" href="https://www.gencodegenes.org/human/release_36lift37.html">https://www.gencodegenes.org/human/release_36lift37.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å»ºç«‹hg19å‚è€ƒåŸºå› ç»„ç´¢å¼•æ–‡ä»¶</span></span><br><span class="line">nohup time bowtie2-build path/to/hg19.fa ./hg19 &gt; hg19.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="comment">#å»ºç«‹hg19è½¬å½•ç»„ç´¢å¼•æ–‡ä»¶</span></span><br><span class="line"><span class="comment">#../index/bowtie/hg19ä¸ºä¸Šä¸€æ­¥æ„å»ºçš„hg19å‚è€ƒåŸºå› ç»„ç´¢å¼•æ–‡ä»¶çš„ç›¸å¯¹è·¯å¾„åŠå‰ç¼€</span></span><br><span class="line">nohup tophat -G ../gencode.v36lift37.annotation.gtf --transcriptome-index=hg19.tr ../index/bowtie/hg19 &amp;</span><br></pre></td></tr></table></figure>

<p><strong><table><tr><td bgcolor=GreenYellow>**å‚è€ƒåŸºå› ç»„çš„æ¯”å¯¹</strong></td></tr></table>** </p>
<h5 id="æ¯”å¯¹æ–¹æ³•ä¸€"><a href="#æ¯”å¯¹æ–¹æ³•ä¸€" class="headerlink" title="æ¯”å¯¹æ–¹æ³•ä¸€"></a>æ¯”å¯¹æ–¹æ³•ä¸€</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ä»¥å…¶ä¸­ä¸€ä¸ªæ ·æœ¬ä¸ºä¾‹ï¼Œèµ°ä¸€æ³¢æ¯”å¯¹è½¬å½•ç»„æµç¨‹ã€‚</span></span><br><span class="line"><span class="comment">#ç†è®ºä¸Šå¯ä»¥èŠ‚çº¦æ—¶é—´ï¼Œä½†æ˜¯å®é™…è·‘èµ·æ¥éå¸¸è€—æ—¶ï¼</span></span><br><span class="line"><span class="comment">#1Gå·¦å³å¾—æ ·æœ¬è·‘äº†4ä¸ªå°æ—¶éƒ½æ²¡æœ‰è·‘å®Œï¼ï¼ï¼ä¸çŸ¥é“é—®é¢˜å‡ºåœ¨é‚£å„¿ï¼Œä½†æ˜¯è¿˜æ˜¯æœæ–­killæ‰</span></span><br><span class="line">tr_index=/home/gongyuqi/ref/hg19/transcriptome-index/hg19.tr</span><br><span class="line">gene_index=~/ref/hg19/index/bowtie/hg19</span><br><span class="line">tophat2 -o ../aligndata/SRR974980.out --transcriptome-index=<span class="variable">$tr</span> -p 30 --phred64-quals <span class="variable">$gene_index</span> SRR974980.fastq.gz</span><br></pre></td></tr></table></figure>
<h5 id="æ¯”å¯¹æ–¹æ³•äºŒ"><a href="#æ¯”å¯¹æ–¹æ³•äºŒ" class="headerlink" title="æ¯”å¯¹æ–¹æ³•äºŒ"></a>æ¯”å¯¹æ–¹æ³•äºŒ</h5><ul>
<li>å†™ä¸€ä¸ªæ¯”å¯¹çš„bashè„šæœ¬æ–‡ä»¶ï¼ˆalign.txtï¼‰ï¼Œå¦‚ä¸‹<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line">ls *.gz|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup tophat -G ~/ref/hg19/gencode.v36lift37.annotation.gtf -p 30 -o ../aligndata/<span class="variable">$id</span>.out ~/ref/hg19/index/bowtie/hg19 <span class="variable">$id</span> &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>æ‰§è¡Œè¿™ä¸ªè„šæœ¬ï¼Œç„¶åå»ç¡è§‰ï¼Œæ—©ä¸Šé†’æ¥çœ‹çœ‹æ¯”å¯¹å¾—ç»“æœå¦‚ä½•~<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup bash align.txt &amp;</span><br></pre></td></tr></table></figure>
è¿è¡Œç»“æœ<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG"/>


</li>
</ul>
<p>æ¯ä¸ªæ–‡ä»¶å¤¹æ‰“å¼€ç»“æœå¦‚ä¸‹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align2.JPG"/></p>
<p>å¤§æ¦‚äº†è§£ä¸€ä¸‹æ ·æœ¬çš„è¿è¡Œæ—¥å¿—</p>
<blockquote>
<p>æ³¨æ„readsçš„é•¿åº¦â€”â€”36ã€‚misoåç»­åˆ†æä¸­éœ€è¦çŸ¥é“è¿™ä¸ªå€¼<br>æ³¨æ„å…¶ä¸­æœ‰ä¸€ä¸ªerror,è‡³äºä¸ºä»€ä¹ˆæœ‰è¿™ä¸ªæŠ¥é”™ï¼Œå…ˆä¸è¿½ç©¶<br>æ³¨æ„æ¯”å¯¹ç»“æœçš„é‚£ä¸ªç»Ÿè®¡æ–‡ä»¶align_summary.txt,æŸ¥çœ‹è¿™ä¸ªæ–‡ä»¶å‘ç°å„ä¸ªæ ·æœ¬æ¯”å¯¹æƒ…å†µéƒ½æŒºå¥½çš„ã€‚è¿™å°±æ˜¯ä¸ºä»€ä¹ˆè¿™é‡Œæš‚æ—¶ä¸è¿½ç©¶é‚£ä¸ªæŠ¥é”™<br>æœ€åä¼šç»Ÿè®¡æ ·æœ¬æ¯”å¯¹æ—¶é•¿<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align4.JPG"/></p>
</blockquote>
<p>3ã€bamæ–‡ä»¶æ’åºåŠç´¢å¼•æ–‡ä»¶å»ºç«‹</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools sort -@ 30 -m 8G -O bam <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.accepted_hits.sorted.bam;<span class="keyword">done</span></span><br><span class="line">ls *sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> nohup samtools index -@ 30 <span class="variable">$id</span> &amp; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h2 id="è¿è¡ŒMISO"><a href="#è¿è¡ŒMISO" class="headerlink" title="è¿è¡ŒMISO"></a>è¿è¡ŒMISO</h2><p><strong><table><tr><td bgcolor=GreenYellow>**ä¸‹è½½hg19çš„GFFæ–‡ä»¶å¹¶æ„å»ºç´¢å¼•</strong></td></tr></table>**</p>
<p>å…·ä½“ä¸‹è½½ç½‘é¡µåŠè¯¦æƒ…ï¼š<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/annotation.html">https://miso.readthedocs.io/en/fastmiso/annotation.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/annotation/human/hg19</span><br><span class="line">wget hollywood.mit.edu/burgelab/miso/annotations/ver2/miso_annotations_hg19_v2.zip</span><br><span class="line">unzip miso_annotations_hg19_v2.zip</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/unzipgff.JPG"/>

<p>è¿™é‡Œæˆ‘ä»¬ä»¥å¤–æ˜¾å­çš„gff3ä¸ºä¾‹æ„å»ºç´¢å¼•</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">index_gff --index SE.hg19.gff3 ./index</span><br></pre></td></tr></table></figure>
<p>ç»“æœå¦‚ä¸‹<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ESgff3index.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow>**ä»¥å•ç«¯æµ‹åºæ ·æœ¬ä¸ºä¾‹è¿è¡Œmiso</strong></td></tr></table>**</p>
<p>è¿™ä¸€æ­¥ç”Ÿæˆçš„ç»“æœæ˜¯å¾ˆå…³é”®çš„ï¼Œä¼šè®¡ç®—alternative splicing eventsçš„PSIå€¼ã€‚è€Œè¿™ä¸ªåœ¨æœ€åå¯è§†åŒ–çš„é˜¶æ®µä¼šä»¥posterior distributions over Î¨(PSI)çš„å½¢å¼å‘ˆç°â€”â€”å³sashimi_plotå›¾æœ€å³è¾¹ä¸€åˆ—çš„æŸ±çŠ¶å›¾ã€‚</p>
<ul>
<li>å†™ä¸€ä¸ªmiso_runçš„è„šæœ¬ï¼Œå†…å®¹å¦‚ä¸‹<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/aligndata/accpted_bam</span><br><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line">out_dir=/home/gongyuqi/project/AS/miso_step1</span><br><span class="line">ls *.sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup miso --run <span class="variable">$index_db</span> <span class="variable">$id</span> --output-dir <span class="variable">$out_dir</span>/<span class="variable">$&#123;id%%.*&#125;</span> --read-len 36 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure></li>
<li>è¿è¡Œè„šæœ¬<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup ./miso_run.sh &amp;</span><br></pre></td></tr></table></figure>
ä¸¾ä¾‹æŸ¥çœ‹å…¶ä¸­ä¸€ä¸ªæ ·æœ¬çš„è¿è¡Œç»“æœ</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/miso_run_out.JPG"/>


<table><tr><td bgcolor=GreenYellow>**æ ·æœ¬é—´æ¯”è¾ƒ**</td></tr></table>

<p>è¿™ä¸ªè½¯ä»¶æœ€å¤§çš„é—®é¢˜åœ¨äºåªèƒ½è¿›è¡Œä¸¤ä¸ªæ ·æœ¬çš„æ¯”è¾ƒï¼Œè¿™ä¸€ç‚¹å°±å¾ˆæ°”ï¼ï¼ï¼<br>æ°”å½’æ°”ï¼Œè¿˜æ˜¯å…ˆè¿è¡Œä¸€ä¸‹ï¼Œè‡³å°‘å…ˆæŠŠæµç¨‹èµ°å®Œï¼Œå…ˆå­¦èµ·æ¥ã€‚</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">compare_miso --compare-samples SRR974978/ SRR974982/ comparison/</span><br></pre></td></tr></table></figure>

<p><strong><table><tr><td bgcolor=GreenYellow>**è¿‡æ»¤</strong></td></tr></table>**</p>
<p>è®¾ç½®å„ç§å‚æ•°çš„é˜ˆå€¼ï¼Œç›®çš„æ˜¯ç­›é€‰å·®å¼‚æ˜¾è‘—çš„alternative splicing eventsã€‚</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">filter_events \</span><br><span class="line">--filter  SRR974978_vs_SRR974982.miso_bf \</span><br><span class="line">--num-inc 1 \</span><br><span class="line">--num-exc 1 \</span><br><span class="line">--num-sum-inc-exc 10 \</span><br><span class="line">--delta-psi 0.20 \</span><br><span class="line">--bayes-factor 10 \</span><br><span class="line">--output-dir ./filter_out</span><br></pre></td></tr></table></figure>
<p>è¿‡æ»¤å‰å.misoz_bfæ–‡ä»¶ä¸­çš„alternative splicing eventsæœ‰æ˜æ˜¾å‡å°‘<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filter.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow>**å¯è§†åŒ–</strong></td></tr></table>**</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line"><span class="comment">#ä¸‹é¢åšæ¼”ç¤ºçš„å‰ªåˆ‡äº‹ä»¶æ˜¯ä»ä¸Šä¸€æ­¥è¿‡æ»¤åçš„.miso_bf.filteredæ–‡ä»¶ä¸­æå–çš„ä¸€ä¸ª</span></span><br><span class="line">sashimi_plot \</span><br><span class="line">--plot-event <span class="string">&quot;chr20:1093906:1094052:+@chr20:1099396:1099545:+@chr20:1106141:1106293:+&quot;</span> \</span><br><span class="line"><span class="variable">$index_db</span> \</span><br><span class="line">sashimi_plot_settings.txt  \</span><br><span class="line">--output-dir ./sashimi_plot</span><br></pre></td></tr></table></figure>

<p>å…¶ä¸­sashimi_plot_settings.txtæ˜¯é…ç½®æ–‡ä»¶ï¼Œå†…å®¹å¦‚ä¸‹ã€‚å½“ç„¶ï¼Œä¹Ÿå¯ä»¥æ ¹æ®è‡ªå·±çš„ä½œå›¾éœ€è¦ä¿®æ”¹ç›¸åº”çš„å‚æ•°</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br></pre></td><td class="code"><pre><span class="line">[data]</span><br><span class="line"><span class="comment"># directory where BAM files are</span></span><br><span class="line">bam_prefix = /home/gongyuqi/project/AS/aligndata/accpted_bam/</span><br><span class="line"><span class="comment"># directory where MISO output is</span></span><br><span class="line">miso_prefix = /home/gongyuqi/project/AS/miso_step1</span><br><span class="line"></span><br><span class="line">bam_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976.accepted_hits.sorted.bam&quot;</span>]</span><br><span class="line"></span><br><span class="line">miso_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976&quot;</span>]</span><br><span class="line"></span><br><span class="line">[plotting]</span><br><span class="line"><span class="comment"># Dimensions of figure to be plotted (in inches)</span></span><br><span class="line">fig_width = 4</span><br><span class="line">fig_height = 10</span><br><span class="line"><span class="comment"># Factor to scale down introns and exons by</span></span><br><span class="line">intron_scale = 30</span><br><span class="line">exon_scale = 4</span><br><span class="line"><span class="comment"># Whether to use a log scale or not when plotting</span></span><br><span class="line">logged = False</span><br><span class="line">font_size = 6</span><br><span class="line"></span><br><span class="line"><span class="comment"># Max y-axis</span></span><br><span class="line">ymax = 150</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot posterior distributions inferred by MISO</span></span><br><span class="line">show_posteriors = True</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to show posterior distributions as bar summaries</span></span><br><span class="line">bar_posteriors = False</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot the number of reads in each junction</span></span><br><span class="line">number_junctions = True</span><br><span class="line"></span><br><span class="line">resolution = .5</span><br><span class="line">posterior_bins = 40</span><br><span class="line">gene_posterior_ratio = 5</span><br><span class="line"></span><br><span class="line"><span class="comment"># List of colors for read denisites of each sample</span></span><br><span class="line">colors = [<span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Number of mapped reads in each sample</span></span><br><span class="line"><span class="comment"># (Used to normalize the read density for RPKM calculation)</span></span><br><span class="line">coverages = [27232306,</span><br><span class="line">             16799279,</span><br><span class="line">             31298963,</span><br><span class="line">             26308550,</span><br><span class="line">             27067722,</span><br><span class="line">             17372484,</span><br><span class="line">             26095030,</span><br><span class="line">             26980460]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Bar color for Bayes factor distribution</span></span><br><span class="line"><span class="comment"># plots (--plot-bf-dist)</span></span><br><span class="line"><span class="comment"># Paint them blue</span></span><br><span class="line">bar_color = <span class="string">&quot;b&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Bayes factors thresholds to use for --plot-bf-dist</span></span><br><span class="line">bf_thresholds = [0, 1, 2, 5, 10, 20]</span><br></pre></td></tr></table></figure>
<p>å¯è§†åŒ–ç»“æœå¦‚ä¸‹<br>å› ä¸ºæˆ‘ä»¬é€‰å–çš„å‰ªåˆ‡äº‹ä»¶æ˜¯SRR974978å’ŒSRR974982ä¹‹é—´çš„å·®å¼‚å‰ªåˆ‡äº‹ä»¶ï¼Œæ‰€ä»¥åœ¨å¤šä¸ªæ ·æœ¬é—´è¶‹åŠ¿ä¸æ˜¯å¾ˆæ˜æ˜¾ã€‚ä½†è¿™é‡Œæˆ‘ä»¬å…ˆå­¦ä¸ªæ–¹æ³•ã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sashimi_plot1.JPG"/></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/" data-id="ckl4ra1mg0012agvf8mae68f9" data-title="MISO" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-2020 ATAC-seq analysis" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/12/08/2020%20ATAC-seq%20analysis/" class="article-date">
  <time class="dt-published" datetime="2020-12-08T05:53:55.000Z" itemprop="datePublished">2020-12-08</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="2020-ATAC-seq-analysis"><a href="#2020-ATAC-seq-analysis" class="headerlink" title="2020 ATAC-seq analysis"></a><strong>2020 ATAC-seq analysis</strong></h1><h2 id="ä¸€ã€ATAC-seqä¸ªæ€§åŒ–çš„conda-environmentçš„æ­å»º"><a href="#ä¸€ã€ATAC-seqä¸ªæ€§åŒ–çš„conda-environmentçš„æ­å»º" class="headerlink" title="ä¸€ã€ATAC-seqä¸ªæ€§åŒ–çš„conda environmentçš„æ­å»º"></a>ä¸€ã€ATAC-seqä¸ªæ€§åŒ–çš„conda environmentçš„æ­å»º</h2><h3 id="1ã€git-clone"><a href="#1ã€git-clone" class="headerlink" title="1ã€git clone"></a>1ã€git clone</h3><p>git cloneæ–‡ç« ä¸­æåˆ°çš„atac-seq-pipeline (é‡Œé¢æœ‰ä½ æ¥ä¸‹æ¥åˆ†ææ‰€éœ€è¦çš„ç»å¤§éƒ¨åˆ†ä¸œè¥¿ã€‚å¦å¤–ä¸€å®šè¦å»çœ‹README.mdï¼Œè¿™ä¸ªæ–‡æ¡£ä¼šå‘Šè¯‰ä½ åˆ†æATAC-seqéœ€è¦åšäº›å‡†å¤‡ï¼Œè¿›è¡Œäº›ä»€ä¹ˆå·¥ä½œã€‚)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ <span class="built_in">cd</span> ~/ATAC</span><br><span class="line">$ git <span class="built_in">clone</span> https://github.com/ENCODE-DCC/atac-seq-pipeline</span><br></pre></td></tr></table></figure>
<h3 id="2ã€ç¯å¢ƒæ­å»º"><a href="#2ã€ç¯å¢ƒæ­å»º" class="headerlink" title="2ã€ç¯å¢ƒæ­å»º"></a>2ã€ç¯å¢ƒæ­å»º</h3><p>ï¼ˆ1ï¼‰conda base environmentæ­å»º</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ä¸€è·¯yesä¸‹å»</span></span><br><span class="line">$ wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh</span><br><span class="line">$ bash Miniconda3-4.6.14-Linux-x86_64.sh</span><br><span class="line"><span class="comment">#æœ€åä¸€ä¸ªyesæ—¶ä½ åŒæ„äº†æ¯æ¬¡è‡ªåŠ¨å¯åŠ¨condaçš„baseç¯å¢ƒï¼Œæ˜¾ç¤ºå®‰è£…å®Œæˆåï¼Œä¸€å®šè¦é€€å‡ºå½“å‰çš„ç»ˆç«¯ï¼Œå†é‡æ–°æ‰“å¼€ä¸€æ¬¡ï¼Œä½ å°±ä¼šå‘ç°åœ¨æ¸¸æ ‡çš„æœ€å·¦è¾¹å¤šäº†ä¸€ä¸ªï¼ˆbaseï¼‰,ä½ çš„base conda environmentå°±æ¿€æ´»å•¦ï¼ä½ çš„condaç¯å¢ƒæ­å»ºå°±æˆåŠŸäº†ä¸€å°æ­¥(*^__^*) å˜»å˜»â€¦â€¦</span></span><br></pre></td></tr></table></figure>
<p>ï¼ˆ2ï¼‰å¤±æ´»æ‰base conda environmentè‡ªåŠ¨æ¿€æ´»çš„è®¾ç½®</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda config --<span class="built_in">set</span> auto_activate_base <span class="literal">false</span></span><br><span class="line"><span class="comment">#åŒæ ·çš„å®Œæˆä¸Šè¯‰æ“ä½œåï¼Œè®°å¾—é€€å‡ºå½“å‰ç»ˆç«¯ï¼Œå†æ¬¡æ‰“å¼€ä¸€ä¸ªæ–°çš„ç»ˆç«¯ã€‚ä½ å°±ä¼šå‘ç°æ¸¸æ ‡çš„æœ€å·¦è¾¹æ²¡æœ‰ï¼ˆbaseï¼‰äº†</span></span><br></pre></td></tr></table></figure>
<p>ï¼ˆ3ï¼‰å®‰è£…pipelineâ€™s conda environment</p>
<p>ï¼ï¼Ÿä½ æ˜¯ä¸æ˜¯å……æ»¡äº†ç–‘é—®ï¼ï¼Ÿä¸‹é¢çš„ä¸¤ä¸ªshellè„šæœ¬å“ªé‡Œå†’å‡ºæ¥çš„ï¼Œå¹²ä»€ä¹ˆç”¨çš„ã€‚ä¸è¦å¿˜äº†ï¼Œæˆ‘ä»¬ç¬¬ä¸€æ­¥git cloneäº†è¿™ä¸ªatac-seq-pipelineå‘¢ï¼è¿™ä¸¤ä¸ªè„šæœ¬å°±æ˜¯å°±æ˜¯æ¥æºäºè¿™ä¸ªcloneä¸‹æ¥çš„pipelineçš„ã€‚</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># ç½‘ä¸ç»™åŠ›çš„æœ‹å‹ï¼Œæ¯”å¦‚è¯´æˆ‘ï¼Œè¿™ä¸€éƒ¨å°±æ˜¯æ•´ä¸ªåˆ†ææµç¨‹çš„é™é€Ÿæ­¥éª¤ï¼ï¼ï¼ç½‘ç»œç»™åŠ›è¿™ä¸€æ­¥ä¹Ÿéœ€è¦èŠ±ä¸€äº›æ—¶é—´ï¼Œè¯»ä¸€ä¸‹ä¸¤ä¸ªshellæ–‡ä»¶å°±çŸ¥é“ä¸ºä»€ä¹ˆå•¦ã€‚</span></span><br><span class="line">$ bash scripts/uninstall_conda_env.sh  <span class="comment"># uninstall it for clean-install</span></span><br><span class="line"><span class="comment">#ä¸‹é¢è¿™ä¸ªshellè„šæœ¬ä¼šåˆ›å»ºä¸¤ä¸ªåˆ†æATAC-seqæ•°æ®æ‰€éœ€è¦çš„ç¯å¢ƒï¼Œä¸€ä¸ªåŸºäºpython3ï¼Œä¸€ä¸ªåŸºäºpython2çš„</span></span><br><span class="line">$ bash scripts/install_conda_env.sh</span><br></pre></td></tr></table></figure>
<p>ï¼ˆ4ï¼‰æ¿€æ´»å®‰è£…çš„condaç¯å¢ƒ</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate encode-atac-seq-pipeline</span><br><span class="line"><span class="comment">#ä½ ä¸€å®šä¼šç–‘é—®ä¸ºä»€ä¹ˆæ˜¯encode-atac-seq-pipeline,é˜…è¯»ä¸€ä¸‹scripts/install_conda_env.shè¿™ä¸ªè„šæœ¬å°±çŸ¥é“å•¦ã€‚å› ä¸ºæºæ–‡æ¡£ä½œè€…æŠŠè¿™ä¸ªç¯å¢ƒåå‘½åä¸ºencode-atac-seq-pipelineã€‚</span></span><br></pre></td></tr></table></figure>
<p>ï¼ˆ5ï¼‰æŸ¥çœ‹æ‰€æœ‰çš„condaç¯å¢ƒ</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda env list</span><br></pre></td></tr></table></figure>
<p>æˆ‘ä»¬ä¼šçœ‹åˆ°æœ‰ä¸€ä¸ªåŸºäºpython2çš„ç¯å¢ƒï¼Œæ²¡æœ‰æ ‡æ˜pythonç‰ˆæœ¬çš„é‚£ä¸ªencode-atac-seq-pipelineæ˜¯åŸºäºpython3çš„ã€‚æœ‰äº›è½¯ä»¶æ˜¯è¦åŸºäºç‰¹å®šç‰ˆæœ¬çš„pythonè¿è¡Œçš„ã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG"/></p>
<h2 id="äºŒã€æ•°æ®ä¸‹è½½"><a href="#äºŒã€æ•°æ®ä¸‹è½½" class="headerlink" title="äºŒã€æ•°æ®ä¸‹è½½"></a>äºŒã€æ•°æ®ä¸‹è½½</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ä¸€æ¬¡æ€§ä¸‹è½½æ‰€æœ‰çš„......_1.fastq.gzå’Œ......_2.fastq.gzæ ·æœ¬</span></span><br><span class="line">dir=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line">x=_1</span><br><span class="line">y=_2</span><br><span class="line"><span class="keyword">for</span> id <span class="keyword">in</span> &#123;93,98,99&#125;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$id<span class="variable">$x</span>.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$id<span class="variable">$y</span>.fastq.gz . </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<h2 id="ä¸‰ã€bowtie2æ¯”å¯¹"><a href="#ä¸‰ã€bowtie2æ¯”å¯¹" class="headerlink" title="ä¸‰ã€bowtie2æ¯”å¯¹"></a>ä¸‰ã€bowtie2æ¯”å¯¹</h2><h3 id="1ã€-fastqc-multiqcæŸ¥çœ‹åŸå§‹æ•°æ®çš„è´¨é‡æƒ…å†µ"><a href="#1ã€-fastqc-multiqcæŸ¥çœ‹åŸå§‹æ•°æ®çš„è´¨é‡æƒ…å†µ" class="headerlink" title="1ã€ fastqc+multiqcæŸ¥çœ‹åŸå§‹æ•°æ®çš„è´¨é‡æƒ…å†µ"></a>1ã€ fastqc+multiqcæŸ¥çœ‹åŸå§‹æ•°æ®çš„è´¨é‡æƒ…å†µ</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">conda insatll -y fastqc multiqc</span><br><span class="line"><span class="built_in">cd</span> raw </span><br><span class="line">fastqc -t 8 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></table></figure>
<p>multiqcç»“æœè¡¨æ˜åŸå§‹æ•°æ®è´¨é‡å·²ç»å¾ˆå¥½äº†ï¼Œå®Œå…¨æ²¡æœ‰adaptoræ®‹ç•™,è¿PCRé‡å¤éƒ½æ˜¯å‹‰å¼ºè¿‡å…³çš„ã€‚per base sequence contentå’Œkmer contentä¸¤é¡¹æŒ‡æ ‡ä¸åˆæ ¼ï¼Œä½†æ˜¯å¹¶ä¸ä¼šå¯¹åç»­çš„æ¯”å¯¹é€ æˆå¤šå¤§çš„è´Ÿé¢å½±å“ï¼ŒåŒæ—¶è¿™ä¸¤é¡¹æŒ‡æ ‡é€šè¿‡è¿‡æ»¤è½¯ä»¶çš„å¤„ç†å¯èƒ½å¹¶ä¸ä¼šåå¾—åˆ°æ”¹å–„ã€‚è¿™æ ·çš„æ•°æ®å…¶å®å¯ä»¥ç›´æ¥æ¯”å¯¹çš„ã€‚<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/multiqc.JPG"/></p>
<h3 id="2ã€-è¿‡æ»¤ä½è´¨é‡çš„reads"><a href="#2ã€-è¿‡æ»¤ä½è´¨é‡çš„reads" class="headerlink" title="2ã€ è¿‡æ»¤ä½è´¨é‡çš„reads"></a>2ã€ è¿‡æ»¤ä½è´¨é‡çš„reads</h3><p>ï¼ˆå¦‚æœåŸå§‹æ•°æ®éœ€è¦è¿›è¡Œè¿‡æ»¤å¤„ç†ï¼Œé‚£ä¹ˆèµ°ä¸‹é¢è¿™ä¸ªè¿‡æ»¤æµç¨‹ï¼Œæˆ‘ä»¬è¿™é‡Œå°±ä¸è¿‡æ»¤äº†ï¼‰</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">nohup cat rawdata.txt|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">trim_galore -j 35 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o ../clean <span class="variable">$fq1</span> <span class="variable">$fq2</span></span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#è¿‡æ»¤åçš„æ•°æ®è´¨é‡æ§åˆ¶ï¼Œçœ‹æ•°æ®æ˜¯å¦æœ‰å¾—åˆ°æ”¹å–„ï¼Œæœ‰æ²¡æœ‰å»æ‰adaptoréå¸¸é‡è¦</span></span><br><span class="line"><span class="built_in">cd</span> clean</span><br><span class="line">fastqc -t 20 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></table></figure>
<h3 id="3ã€-æ¯”å¯¹å‚è€ƒåŸºå› ç»„"><a href="#3ã€-æ¯”å¯¹å‚è€ƒåŸºå› ç»„" class="headerlink" title="3ã€ æ¯”å¯¹å‚è€ƒåŸºå› ç»„"></a>3ã€ æ¯”å¯¹å‚è€ƒåŸºå› ç»„</h3><ul>
<li><p>æ¯”å¯¹</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup cat sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">bowtie2  --very-sensitive -X2000 --mm --threads 50 -x /home/gongyuqi/ref/mm10/index/mm10 -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span>|samtools sort -O bam -@ 15 - -o ../align/<span class="variable">$sample</span> </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure></li>
<li><p>æ¯”å¯¹æƒ…å†µç»Ÿè®¡åŠæŸ¥çœ‹</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">samtools flagstat -@ 4 <span class="variable">$id</span>  &gt; <span class="variable">$&#123;id%%.*&#125;</span>.<span class="built_in">stat</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line">ls *.<span class="built_in">stat</span> | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> | grep <span class="string">&quot;mapped (&quot;</span>;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mapped.JPG"/>
</li>
<li><p>ç”Ÿæˆbamæ–‡ä»¶çš„ç´¢å¼•æ–‡ä»¶</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup ls *sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools index -@ 10 <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>

</li>
</ul>
<h2 id="å››ã€PCRå»é™¤é‡åŠè´¨æ§åˆ¶"><a href="#å››ã€PCRå»é™¤é‡åŠè´¨æ§åˆ¶" class="headerlink" title="å››ã€PCRå»é™¤é‡åŠè´¨æ§åˆ¶"></a>å››ã€PCRå»é™¤é‡åŠè´¨æ§åˆ¶</h2><h3 id="1ã€åˆæ­¥è¿‡æ»¤"><a href="#1ã€åˆæ­¥è¿‡æ»¤" class="headerlink" title="1ã€åˆæ­¥è¿‡æ»¤"></a>1ã€åˆæ­¥è¿‡æ»¤</h3><p>#=============================<br>#Remove unmapped, mate unmapped<br>#not primary alignment, reads failing platform<br>#Only keep properly paired reads<br>#Obtain name sorted BAM file<br>#=============================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#fixmateè¦æ±‚è¾“å…¥çš„bamæ–‡ä»¶æŒ‰åç§°æ’åºï¼Œæ‰€ä»¥ç¬¬ä¸€æ­¥åˆæ­¥è¿‡æ»¤åè¦æŒ‰readsåç§°æ’åºä¸€ä¸‹</span></span><br><span class="line">nohup ls *.sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools view -@ 8 -F 524 -f 2 -u <span class="variable">$id</span> | samtools sort -@ 8 -n /dev/stdin -o ../filter/<span class="variable">$&#123;id%%.*&#125;</span>.filt.nmsrt.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#multimapping=4æ˜¯bowtie2çš„é»˜è®¤è®¾ç½®</span></span><br><span class="line"><span class="built_in">cd</span> ../filter</span><br><span class="line">multimapping=4 </span><br><span class="line">nohup ls *filt.nmsrt.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools view -@ 8 -h <span class="variable">$id</span>|assign_multimappers.py -k <span class="variable">$multimapping</span> --paired-end |samtools fixmate -@ 8 -r /dev/stdin <span class="variable">$&#123;id%%.*&#125;</span>.filt.nmsrt.fixmate.bam; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="2ã€è¿›ä¸€æ­¥"><a href="#2ã€è¿›ä¸€æ­¥" class="headerlink" title="2ã€è¿›ä¸€æ­¥"></a>2ã€è¿›ä¸€æ­¥</h3><p>#===============================================<br>#Remove orphan reads (pair was removed)<br>#and read pairs mapping to different chromosomes<br>#Obtain position sorted BAM<br>#===============================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å¤šä¸ªæ ·æœ¬å†™å¾ªç¯</span></span><br><span class="line">nohup ls *.filt.nmsrt.fixmate.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools view -@ 8 -F 1804 -f 2 -u <span class="variable">$id</span> | samtools sort -@ 8 /dev/stdin -o <span class="variable">$&#123;id%%.*&#125;</span>.prefilt.bam;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="3ã€æ ‡è®°PCRé‡å¤"><a href="#3ã€æ ‡è®°PCRé‡å¤" class="headerlink" title="3ã€æ ‡è®°PCRé‡å¤"></a>3ã€æ ‡è®°PCRé‡å¤</h3><p>#===============<br>#Mark duplicates<br>#===============</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">MARKDUP=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line"><span class="comment">#å¤šä¸ªæ ·æœ¬å†™å¾ªç¯</span></span><br><span class="line">nohup ls *.prefilt.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> java -Xmx16G -jar <span class="variable">$MARKDUP</span> MarkDuplicates INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">$&#123;id%%.*&#125;</span>.prefilt.dupmark.bam METRICS_FILE=<span class="variable">$&#123;id%%.*&#125;</span>.prefilt.dupmark.qc VALIDATION_STRINGENCY=LENIENT ASSUME_SORTED=<span class="literal">true</span> REMOVE_DUPLICATES=<span class="literal">false</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="4ã€å»é™¤PCRé‡å¤ã€å»ºç«‹ç´¢å¼•æ–‡ä»¶ã€æŒ‰nameæ’åºçš„bamæ–‡ä»¶ã€è´¨æ§æ–‡ä»¶"><a href="#4ã€å»é™¤PCRé‡å¤ã€å»ºç«‹ç´¢å¼•æ–‡ä»¶ã€æŒ‰nameæ’åºçš„bamæ–‡ä»¶ã€è´¨æ§æ–‡ä»¶" class="headerlink" title="4ã€å»é™¤PCRé‡å¤ã€å»ºç«‹ç´¢å¼•æ–‡ä»¶ã€æŒ‰nameæ’åºçš„bamæ–‡ä»¶ã€è´¨æ§æ–‡ä»¶"></a>4ã€å»é™¤PCRé‡å¤ã€å»ºç«‹ç´¢å¼•æ–‡ä»¶ã€æŒ‰nameæ’åºçš„bamæ–‡ä»¶ã€è´¨æ§æ–‡ä»¶</h3><p>#============================<br>#Remove duplicates<br>#Index final position sorted BAM<br>#Create final name sorted BAM<br>#============================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å¤šä¸ªæ ·æœ¬å†™å¾ªç¯</span></span><br><span class="line"><span class="comment">#å»é‡</span></span><br><span class="line">nohup ls *.prefilt.dupmark.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools view -@ 16 -F 1804 -f 2 -b <span class="variable">$id</span> &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#æ„å»ºç´¢å¼•æ–‡ä»¶</span></span><br><span class="line">nohup ls *nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools index <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="5ã€æµ‹åºæ–‡åº“å¤æ‚åº¦çš„æ£€éªŒ"><a href="#5ã€æµ‹åºæ–‡åº“å¤æ‚åº¦çš„æ£€éªŒ" class="headerlink" title="5ã€æµ‹åºæ–‡åº“å¤æ‚åº¦çš„æ£€éªŒ"></a>5ã€æµ‹åºæ–‡åº“å¤æ‚åº¦çš„æ£€éªŒ</h3><p>#=============================<br>#Compute library complexity<br>#=============================<br>#Sort by name<br>#convert to bedPE and obtain fragment coordinates<br>#sort by position and strand<br>#Obtain unique count statistics<br>#================================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å¤šä¸ªæ ·æœ¬å†™å¾ªç¯</span></span><br><span class="line">nohup ls *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools sort -@ 16 -n <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.bam;<span class="keyword">done</span> &amp;</span><br><span class="line">nohup ls *.nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools bamtobed -bedpe -i <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1,$2,$4,$6,$9,$10&#125;&#x27;</span> | grep -v <span class="string">&#x27;chrM&#x27;</span> | sort | uniq -c | awk <span class="string">&#x27;BEGIN&#123;mt=0;m0=0;m1=0;m2=0&#125; ($1==1)&#123;m1=m1+1&#125; ($1==2)&#123;m2=m2+1&#125; &#123;m0=m0+1&#125; &#123;mt=mt+$1&#125; END&#123;m1_m2=-1.0; if(m2&gt;0) m1_m2=m1/m2;printf &quot;%d\t%d\t%d\t%d\t%f\t%f\t%f\n&quot;,mt,m0,m1,m2,m0/mt,m1/m0,m1_m2&#125;&#x27;</span> &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.pbc.qc;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<p>Library complexity measuresè®¡ç®—ç»“æœå¦‚ä¸‹ï¼Œâ€¦nodup.pbc.qcæ–‡ä»¶æ ¼å¼ä¸ºï¼š<br>TotalReadPairs<br>DistinctReadPairs<br>OneReadPair<br>TwoReadPairs<br>NRF=Distinct/Total<br>PBC1=OnePair/Distinct<br>PBC2=OnePair/TwoPair</p>
<p>é’ˆå¯¹NRFã€PBC1ã€PBC2è¿™å‡ ä¸ªæŒ‡æ ‡ï¼ŒENCODEå®˜ç½‘æä¾›çš„æ ‡å‡†å¦‚ä¸‹:</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/NRF-PBC1-PBC2.jpeg"></p>
<p>æˆ‘ä»¬æ­¤æ¬¡4ä¸ªæ ·æœ¬çš„åˆ†æç»“æœæ˜¾ç¤ºè¿™å‡ ä¸ªæŒ‡æ ‡çš„å€¼å¦‚ä¸‹æ‰€ç¤ºï¼š</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NRF PBC1 PBC2.JPG"/>

<p>é™¤äº†ç¬¬ä¸€ä¸ªæ ·æœ¬ï¼ˆSRR12692092ï¼‰ï¼Œå…¶ä»–æ ·æœ¬è®¡ç®—ç»“æœæ˜¾ç¤ºNRFã€PBC1ã€PBC2çš„å€¼éƒ½éå¸¸å®Œç¾ï¼Œè¯´æ˜æˆ‘ä»¬è¿›è¡Œè¿‡æ»¤å’ŒPCRå»é‡çš„bamæ–‡ä»¶è´¨é‡ä¸Šæ²¡æœ‰é—®é¢˜ï¼Œå¯ä»¥ç”¨äºåç»­çš„åˆ†æã€‚</p>
<h3 id="6ã€æ’å…¥ç‰‡æ®µè´¨æ§æ£€æµ‹"><a href="#6ã€æ’å…¥ç‰‡æ®µè´¨æ§æ£€æµ‹" class="headerlink" title="6ã€æ’å…¥ç‰‡æ®µè´¨æ§æ£€æµ‹"></a>6ã€æ’å…¥ç‰‡æ®µè´¨æ§æ£€æµ‹</h3><table><tr><td bgcolor=GreenYellow>ç”¨äºè¯„ä¼°ATAC/Chipå®éªŒè´¨é‡å¥½åçš„ä¸€ä¸ªé‡è¦æŒ‡æ ‡</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#å¤šä¸ªæ ·æœ¬å¾ªç¯</span></span><br><span class="line">picard_jar=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line">nohup ls *.nodup.srt.name.bam | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> java -Xmx6G -XX:ParallelGCThreads=1 -jar <span class="variable">$&#123;picard_jar&#125;</span> CollectInsertSizeMetrics INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">$&#123;id%%.*&#125;</span>.INSERT.DATA H=<span class="variable">$&#123;id%%.*&#125;</span>.INSERT.PLOT.pdf VERBOSITY=ERROR QUIET=TRUE USE_JDK_DEFLATER=TRUE USE_JDK_INFLATER=TRUE W=1000 STOP_AFTER=5000000; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<p>æˆåŠŸçš„ATAC-seqå®éªŒä¼šç”Ÿæˆå…·æœ‰é€çº§é€’å‡å’Œå‘¨æœŸæ€§çš„å³°ï¼Œå¯¹åº”æ— æ ¸å°ä½“åŒºåŸŸï¼ˆNucleosome free region, NRF,100bpï¼‰,å•æ ¸åŒºåŸŸï¼ˆmono-nucleosome,200bpï¼‰ï¼ŒåŒæ ¸åŒºåŸŸï¼ˆ400bpï¼‰ï¼Œä¸‰æ ¸åŒºåŸŸï¼ˆ600bpï¼‰ã€‚</p>
<p>æœ¬æ¬¡ç»“æœæ˜¾ç¤ºæ ·æœ¬è´¨é‡OK</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg" width="50%"></p>
<h3 id="7ã€TSSå¯Œé›†"><a href="#7ã€TSSå¯Œé›†" class="headerlink" title="7ã€TSSå¯Œé›†"></a>7ã€TSSå¯Œé›†</h3><p><strong>ä»¥BRM014-10uM_24h_wtæ ·æœ¬ä¸ºä¾‹ï¼Œå¯è§†åŒ–peaksåœ¨åŸºå› ç»„ä¸Šçš„åˆ†å¸ƒç‰¹å¾(TSSåŒºåŸŸï¼›TSSã€gene-bodyã€TSEåŒºåŸŸ)</strong></p>
<ul>
<li>both -R and -S can accept multiple files</li>
<li>UCSCå®˜ç½‘ä¸‹è½½å°é¼ å‚è€ƒåŸºå› ç»„æ³¨é‡Šæ–‡ä»¶ï¼Œè¯¦æƒ…å‚è€ƒï¼š<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/2494.html">http://www.bio-info-trainee.com/2494.html</a></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ç”Ÿæˆbwæ–‡ä»¶ç”¨äºåç»­çš„computeMatrix</span></span><br><span class="line">nohup ls *.nodup.bam |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup bamCoverage --normalizeUsing CPM -b <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.nodup.bw</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#æ³¨é‡Šæ–‡ä»¶ç»å¯¹è·¯åŠ²</span></span><br><span class="line">refseq=/home/gongyuqi/ref/mm10/mouse.ucsc.refseq.bed</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS</span></span><br><span class="line"><span class="comment">##ç”ŸæˆplotHeatmapéœ€è¦çš„gzæ–‡ä»¶</span></span><br><span class="line">nohup computeMatrix reference-point  --referencePoint TSS  -p 30 \</span><br><span class="line">-b 3000 -a 3000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##å¯è§†åŒ–æ–‡ä»¶</span></span><br><span class="line">nohup plotHeatmap -m BRM014-10uM_24h_wt_TSS.gz  -out BRM014-10uM_24h_wt_TSS_rainbow.png --colorMap rainbow &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS_gene-body_TSE</span></span><br><span class="line"><span class="comment">##ç”ŸæˆplotHeatmapéœ€è¦çš„gzæ–‡ä»¶</span></span><br><span class="line">nohup computeMatrix scale-regions -p 30 \</span><br><span class="line">-a 3000 -b 3000 -m 5000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS_gene0body_TSE.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##å¯è§†åŒ–æ–‡ä»¶</span></span><br><span class="line">nohup plotHeatmap -m BRM014-10uM_24h_wt_TSS_genebody_TSE.gz  -out BRM014-10uM_24h_wt_TSS_genebody_TSE.png --colorMap rainbow &amp;</span><br></pre></td></tr></table></figure>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png" width="50%"></p>
<h2 id="äº”ã€ç”ŸæˆtagAlignæ ¼å¼æ–‡ä»¶"><a href="#äº”ã€ç”ŸæˆtagAlignæ ¼å¼æ–‡ä»¶" class="headerlink" title="äº”ã€ç”ŸæˆtagAlignæ ¼å¼æ–‡ä»¶"></a>äº”ã€ç”ŸæˆtagAlignæ ¼å¼æ–‡ä»¶</h2><h3 id="1-Convert-PE-BAM-to-tagAlign"><a href="#1-Convert-PE-BAM-to-tagAlign" class="headerlink" title="1. Convert PE BAM to tagAlign"></a>1. Convert PE BAM to tagAlign</h3><ul>
<li>å¯¹äºå•ç«¯åºåˆ—ã€‚ç›´æ¥ç”¨bedæ ¼å¼å°±å¯ä»¥ï¼›å¯¹äºåŒç«¯åºåˆ—ï¼Œæ¨èç”¨bedpeæ ¼å¼ã€‚è¿™ä¸¤ç§æ ¼å¼éƒ½å¯ä»¥ç§°ä¹‹ä¸ºtagAlignï¼Œå¯ä»¥ä½œä¸ºmacsçš„è¾“å…¥æ–‡ä»¶ã€‚</li>
<li>tagAligenæ ¼å¼ç›¸æ¯”bamï¼Œæ–‡ä»¶å¤§å°ä¼šå°å¾ˆå¤šï¼Œæ›´åŠ æ–¹ä¾¿æ–‡ä»¶çš„è¯»å–ã€‚åœ¨è½¬æ¢å¾—åˆ°tagAlignæ ¼å¼ä¹‹åï¼Œæˆ‘ä»¬å°±å¯ä»¥å¾ˆå®¹æ˜“çš„å°†åæ ‡è¿›è¡Œåç§»</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">nohup ls *nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools bamtobed -bedpe -mate1 -i <span class="variable">$id</span>  | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.bedpe.gz;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#å«æœ‰chrMçš„æŸ“è‰²ä½“çš„TagAlignæ–‡ä»¶</span></span><br><span class="line">nohup ls *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;printf &quot;%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n&quot;,$1,$2,$3,$9,$4,$5,$6,$10&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#å»é™¤chrMçš„æŸ“è‰²ä½“çš„TagAlignæ–‡ä»¶</span></span><br><span class="line">nohup ls *nodup.srt.name.bedpe.gz|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -P -v <span class="string">&quot;^chrM&quot;</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;printf &quot;%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n&quot;,$1,$2,$3,$9,$4,$5,$6,$10&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.nomit.srt.name.tagAlign.gz; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h3 id="2-Stand-Cross-Correlation-analysis"><a href="#2-Stand-Cross-Correlation-analysis" class="headerlink" title="2. Stand Cross Correlation analysis"></a>2. Stand Cross Correlation analysis</h3><table><tr><td bgcolor=GreenYellow>ç”¨äºè¯„ä¼°ATAC/Chipå®éªŒè´¨é‡å¥½åçš„ä¸€ä¸ªé‡è¦æŒ‡æ ‡</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">NREADS=25000000</span><br><span class="line">nohup ls *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -v â€œchrMâ€ | shuf -n <span class="variable">$&#123;NREADS&#125;</span> --random-source=&lt;(openssl enc -aes-256-ctr -pass pass:$(zcat -f <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.tagAlign.gz | wc -c) -nosalt &lt;/dev/zero 2&gt;/dev/null) | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1,$2,$3,&quot;N&quot;,&quot;1000&quot;,$9&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.nomit.srt.name.$((NREADS / <span class="number">1000000</span>)).tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#å‘½ä»¤æœ€ç»ˆä¼šç”Ÿæˆäº¤å‰ç›¸å…³è´¨é‡è¯„ä¼°æ–‡ä»¶ï¼Œ*.cc.qcæ–‡ä»¶ä¸­ä¼šè¾“å‡ºåŒ…å«11åˆ—çš„ä¿¡æ¯ï¼Œé‡ç‚¹å…³æ³¨9-11åˆ—çš„ä¿¡æ¯ï¼Œcc.plot.pdfæ–‡ä»¶ç›¸å½“äº*.cc.qcæ–‡ä»¶çš„å¯è§†åŒ–</span></span><br><span class="line">nohup ls *$((NREADS / <span class="number">1000000</span>)).tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> Rscript $(<span class="built_in">which</span> run_spp.R) -c=<span class="variable">$id</span> -p=10 -filtchr=chrM -savp=<span class="variable">$&#123;id%%.*&#125;</span>.cc.plot.pdf -out=<span class="variable">$&#123;id%%.*&#125;</span>.cc.qc; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#è´¨æ§ç»“æœæŸ¥çœ‹ï¼Œä¸»è¦çœ‹NSC,RSC,Quality tagä¸‰ä¸ªå€¼å³è¾“å‡ºæ–‡ä»¶çš„ç¬¬9åˆ—ï¼Œç¬¬10åˆ—ï¼Œç¬¬11åˆ—ã€‚</span></span><br><span class="line">ls *.cc.qc|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> | awk <span class="string">&#x27;&#123;print $9, &quot;\t&quot;, $10, &quot;\t&quot;, $11&#125;&#x27;</span>;<span class="keyword">done</span> </span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/cc.qc.JPG" width="50%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg" width="50%">

<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg" width="50%"></p>
<ul>
<li>è´¨æ§ç»“æœè§£è¯»<blockquote>
<p>Normalized strand cross-correlation coefficent (NSC)ï¼šNSCæ˜¯æœ€å¤§äº¤å‰ç›¸å…³å€¼é™¤ä»¥èƒŒæ™¯äº¤å‰ç›¸å…³çš„æ¯”ç‡(æ‰€æœ‰å¯èƒ½çš„é“¾è½¬ç§»çš„æœ€å°äº¤å‰ç›¸å…³å€¼)ã€‚NSCå€¼è¶Šå¤§è¡¨æ˜å¯Œé›†æ•ˆæœè¶Šå¥½ï¼ŒNSCå€¼ä½äº1.1è¡¨æ˜è¾ƒå¼±çš„å¯Œé›†ï¼Œå°äº1è¡¨ç¤ºæ— å¯Œé›†ã€‚NSCå€¼ç¨å¾®ä½äº1.05ï¼Œæœ‰è¾ƒä½çš„ä¿¡å™ªæ¯”æˆ–å¾ˆå°‘çš„å³°ï¼Œè¿™è‚¯èƒ½æ˜¯ç”Ÿç‰©å­¦çœŸå®ç°è±¡ï¼Œæ¯”å¦‚æœ‰çš„å› å­åœ¨ç‰¹å®šç»„ç»‡ç±»å‹ä¸­åªæœ‰å¾ˆå°‘çš„ç»“åˆä½ç‚¹ï¼›ä¹Ÿå¯èƒ½ç¡®å®æ˜¯æ•°æ®è´¨é‡å·®ã€‚</p>
<p>Relative strand cross-correlation coefficient (RSC)ï¼šRSCæ˜¯ç‰‡æ®µé•¿åº¦ç›¸å…³å€¼å‡å»èƒŒæ™¯ç›¸å…³å€¼é™¤ä»¥phantom-peakç›¸å…³å€¼å‡å»èƒŒæ™¯ç›¸å…³å€¼ã€‚RSCçš„æœ€å°å€¼å¯èƒ½æ˜¯0ï¼Œè¡¨ç¤ºæ— ä¿¡å·ï¼›å¯Œé›†å¥½çš„å®éªŒRSCå€¼å¤§äº1ï¼›ä½äº1è¡¨ç¤ºè´¨é‡ä½ã€‚</p>
<p>QualityTag: Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High,2:veryHigh)</p>
<p>æŸ¥çœ‹äº¤å‰ç›¸å…³æ€§è´¨é‡è¯„ä¼°ç»“æœï¼Œä¸»è¦çœ‹NSC,RSC,Quality tagä¸‰ä¸ªå€¼ï¼Œè¿™ä¸‰ä¸ªå€¼åˆ†åˆ«å¯¹åº”è¾“å‡ºæ–‡ä»¶çš„ç¬¬9åˆ—ï¼Œç¬¬10åˆ—ï¼Œç¬¬11åˆ—ã€‚</p>
</blockquote>
</li>
</ul>
<h2 id="å…­ã€Call-Peaks"><a href="#å…­ã€Call-Peaks" class="headerlink" title="å…­ã€Call Peaks"></a>å…­ã€Call Peaks</h2><h3 id="1ã€å»é™¤çº¿ç²’ä½“åŸºå› çš„TagAlignæ ¼å¼æ–‡ä»¶è¿›è¡Œshiftæ“ä½œï¼Œè¾“å…¥macs2è½¯ä»¶å»callpeak"><a href="#1ã€å»é™¤çº¿ç²’ä½“åŸºå› çš„TagAlignæ ¼å¼æ–‡ä»¶è¿›è¡Œshiftæ“ä½œï¼Œè¾“å…¥macs2è½¯ä»¶å»callpeak" class="headerlink" title="1ã€å»é™¤çº¿ç²’ä½“åŸºå› çš„TagAlignæ ¼å¼æ–‡ä»¶è¿›è¡Œshiftæ“ä½œï¼Œè¾“å…¥macs2è½¯ä»¶å»callpeak"></a>1ã€å»é™¤çº¿ç²’ä½“åŸºå› çš„TagAlignæ ¼å¼æ–‡ä»¶è¿›è¡Œshiftæ“ä½œï¼Œè¾“å…¥macs2è½¯ä»¶å»callpeak</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">smooth_window=150 <span class="comment"># default</span></span><br><span class="line">shiftsize=$(( -<span class="variable">$smooth_window</span>/<span class="number">2</span> ))</span><br><span class="line">pval_thresh=0.01</span><br><span class="line">nohup ls *nodup.nomit.srt.name.tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span> macs2 callpeak \</span><br><span class="line">-t <span class="variable">$id</span> -f BED -n <span class="string">&quot;<span class="variable">$&#123;id%%.*&#125;</span>&quot;</span> -g mm -p <span class="variable">$pval_thresh</span> \</span><br><span class="line">--<span class="built_in">shift</span> <span class="variable">$shiftsize</span> --extsize <span class="variable">$smooth_window</span> --nomodel -B --SPMR --keep-dup all --call-summits; \</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="2ã€å»é™¤ENCODEåˆ—å…¥é»‘åå•çš„åŒºåŸŸ"><a href="#2ã€å»é™¤ENCODEåˆ—å…¥é»‘åå•çš„åŒºåŸŸ" class="headerlink" title="2ã€å»é™¤ENCODEåˆ—å…¥é»‘åå•çš„åŒºåŸŸ"></a>2ã€å»é™¤ENCODEåˆ—å…¥é»‘åå•çš„åŒºåŸŸ</h3><ul>
<li>å»é™¤é»‘åå•çš„bedæ–‡ä»¶ï¼Œç”¨äºåç»­çš„peaksæ³¨é‡Š<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">BLACKLIST=/home/gongyuqi/project/ATAC/mm10.blacklist.bed.gz</span><br><span class="line"><span class="comment">#*_summits.bedä¸ºmacs2è½¯ä»¶callpeakçš„ç»“æœæ–‡ä»¶ä¹‹ä¸€</span></span><br><span class="line">nohup ls *_summits.bed | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools intersect -a <span class="variable">$id</span> -b <span class="variable">$BLACKLIST</span> -v &gt; <span class="variable">$&#123;id%%.*&#125;</span>_filt_blacklist.bed; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#æŸ¥çœ‹è¿‡æ»¤é»‘åå•çš„åŒºåŸŸå‰åçš„bedæ–‡ä»¶çš„peaksæ•°</span></span><br><span class="line">ls *summits.bed|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> |wc -l &gt;&gt;summits.bed.txt;<span class="keyword">done</span></span><br><span class="line">ls *summits_filt_blacklist.bed|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> |wc -l &gt;&gt;summits_filt_blacklist.bed.txt;<span class="keyword">done</span></span><br><span class="line">past summits.bed.txt summits_filt_blacklist.bed.txt </span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/before_after_blacklist-bed.JPG"/>





</li>
</ul>
<ul>
<li>å»é™¤é»‘åå•çš„narrowPeaksæ–‡ä»¶ï¼Œç”¨äºåç»­çš„IDRè¯„ä¼°<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ä½¿ç”¨IDRéœ€è¦å…ˆå¯¹MACS2çš„ç»“æœæ–‡ä»¶narrowPeakæ ¹æ®-log10(p-value)è¿›è¡Œæ’åº,-log10(p-value)åœ¨ç¬¬å…«åˆ—ã€‚</span></span><br><span class="line"><span class="comment"># Sort by Col8 in descending order and replace long peak names in Column 4 with Peak_&lt;peakRank&gt;</span></span><br><span class="line"><span class="comment">#*_peaks.narrowPeakä¸ºmacs2è½¯ä»¶callpeakçš„ç»“æœæ–‡ä»¶ä¹‹ä¸€</span></span><br><span class="line">NPEAKS=300000 </span><br><span class="line">ls *_peaks.narrowPeak | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> sort -k 8gr,8gr <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;$4=&quot;Peak_&quot;NR ; print $0&#125;&#x27;</span> | head -n <span class="variable">$&#123;NPEAKS&#125;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%_*&#125;</span>.narrowPeak.gz; <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">BLACKLIST=../BLACKLIST/mm10.blacklist.bed.gz</span><br><span class="line"></span><br><span class="line"><span class="comment">#ç”Ÿæˆä¸å‹ç¼©æ–‡ä»¶</span></span><br><span class="line">ls *.narrowPeak.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools intersect -v -a <span class="variable">$id</span> -b <span class="variable">$&#123;BLACKLIST&#125;</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; &#123;if ($5&gt;1000) $5=1000; print $0&#125;&#x27;</span> | grep -P <span class="string">&#x27;chr[\dXY]+[ \t]&#x27;</span>  &gt; <span class="variable">$&#123;id%%.*&#125;</span>.narrowPeak.filt_blacklist; <span class="keyword">done</span></span><br><span class="line"><span class="comment">#ç”Ÿæˆå‹ç¼©æ–‡ä»¶</span></span><br><span class="line"><span class="comment">#ls *.narrowPeak.gz | while read id; do bedtools intersect -v -a $id -b $&#123;BLACKLIST&#125; | awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; &#123;if ($5&gt;1000) $5=1000; print $0&#125;&#x27; | grep -P &#x27;chr[\dXY]+[ \t]&#x27; | gzip -nc &gt; $&#123;id%%.*&#125;.narrowPeak.filt_blacklist.gz; done</span></span><br></pre></td></tr></table></figure>
<h3 id="3ã€Irreproducibility-Discovery-Rate-IDR-è¯„ä¼°"><a href="#3ã€Irreproducibility-Discovery-Rate-IDR-è¯„ä¼°" class="headerlink" title="3ã€Irreproducibility Discovery Rate (IDR)è¯„ä¼°"></a>3ã€Irreproducibility Discovery Rate (IDR)è¯„ä¼°</h3><table><tr><td bgcolor=GreenYellow>ç”¨äºè¯„ä¼°é‡å¤æ ·æœ¬é—´peaksä¸€è‡´æ€§çš„é‡è¦æŒ‡æ ‡</td></tr></table>
é¦–å…ˆç”ŸæˆnarrowPeak_sample.txtæ–‡ä»¶ï¼Œæ–¹ä¾¿åç»­å¾ªç¯å¤„ç†ï¼Œç”Ÿæˆæ–‡ä»¶å†…å®¹å¦‚ä¸‹ï¼š
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/narrowPeak for idr.JPG"/>

</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup cat narrowPeak_sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">Rep1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">Rep2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;Rep1%%.*&#125;</span>_<span class="variable">$&#123;Rep2%%.*&#125;</span>_idr</span><br><span class="line">idr --samples <span class="variable">$Rep1</span> <span class="variable">$Rep2</span> --input-file-type narrowPeak -o <span class="variable">$sample</span> --plot  </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<ul>
<li>DMSO_24h_wt ï¼ˆæ ·æœ¬å¤„ç†æƒ…å†µï¼‰</li>
<li>SRR12692092.filt_blacklist.narrowPeak SRR12692093.filt_blacklist.narrowPeak</li>
<li>æ²¡æœ‰é€šè¿‡IDRé˜ˆå€¼çš„æ˜¾ç¤ºä¸ºçº¢è‰²</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png"/>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png"/>

<ul>
<li>BRM014-10uM_24h_wt ï¼ˆæ ·æœ¬å¤„ç†æƒ…å†µï¼‰</li>
<li>SRR12692098.filt_blacklist.narrowPeak SRR12692099.filt_blacklist.narrowPeak</li>
<li>æ²¡æœ‰é€šè¿‡IDRé˜ˆå€¼çš„æ˜¾ç¤ºä¸ºçº¢è‰²</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png"/>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png"/>

<blockquote>
<p>IDRè¯„ä¼°ä¼šåŒæ—¶è€ƒè™‘peaksé—´çš„overlapå’Œå¯Œé›†å€æ•°çš„ä¸€è‡´æ€§ã€‚é€šè¿‡IDRé˜ˆå€¼ï¼ˆ0.05ï¼‰çš„å æ¯”è¶Šå¤§ï¼Œè¯´æ˜é‡å¤æ ·æœ¬é—´peaksä¸€è‡´æ€§è¶Šå¥½ã€‚ä»idrçš„åˆ†æç»“æœçœ‹ï¼Œæˆ‘ä»¬çš„æµ‹è¯•æ•°æ®è¿˜å¯ä»¥çš„å‘¢ã€‚</p>
</blockquote>
<p><strong>IDRè¯„ä¼°ç›¸å…³å‚è€ƒèµ„æ–™ï¼š</strong>  </p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/d8a7056b4294?utm_campaign=maleskine&utm_content=note&utm_medium=seo_notes&utm_source=recommendation">é‡å¤æ ·æœ¬çš„å¤„ç†â€”â€”IDR</a></p>
<h3 id="4ã€Fraction-of-reads-in-peaks-FRiP-è¯„ä¼°"><a href="#4ã€Fraction-of-reads-in-peaks-FRiP-è¯„ä¼°" class="headerlink" title="4ã€Fraction of reads in peaks (FRiP)è¯„ä¼°"></a>4ã€Fraction of reads in peaks (FRiP)è¯„ä¼°</h3><table><tr><td bgcolor=GreenYellow>åæ˜ æ ·æœ¬å¯Œé›†æ•ˆæœå¥½åçš„è¯„ä»·æŒ‡æ ‡</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ç”Ÿæˆbedæ–‡ä»¶</span></span><br><span class="line">nohup ls *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> (bedtools bamtobed -i <span class="variable">$id</span> &gt;<span class="variable">$&#123;id%%.*&#125;</span>.nodup.bed) ;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#æ‰¹é‡è®¡ç®—FRiP</span></span><br><span class="line">ls *_summits_filt_blacklist.bed|<span class="keyword">while</span>  <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span></span><br><span class="line">bed=<span class="variable">$&#123;id%%_*&#125;</span>.nodup.bed </span><br><span class="line">Reads=$(bedtools intersect -a <span class="variable">$bed</span> -b <span class="variable">$id</span> |wc -l|awk <span class="string">&#x27;&#123;print $1&#125;&#x27;</span>)</span><br><span class="line">totalReads=$(wc -l <span class="variable">$bed</span>|awk <span class="string">&#x27;&#123;print $1&#125;&#x27;</span>)</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$Reads</span>  <span class="variable">$totalReads</span> </span><br><span class="line"><span class="built_in">echo</span> <span class="string">&#x27;==&gt; FRiP value:&#x27;</span> $(bc &lt;&lt;&lt; <span class="string">&quot;scale=2;100*<span class="variable">$Reads</span>/<span class="variable">$totalReads</span>&quot;</span>)<span class="string">&#x27;%&#x27;</span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>FRiPå€¼åœ¨5%ä»¥ä¸Šç®—æ¯”è¾ƒå¥½çš„ã€‚ä½†ä¹Ÿä¸ç»å¯¹ï¼Œè¿™æ˜¯ä¸ªè½¯é˜ˆå€¼ï¼Œå¯ä»¥ä½œä¸ºä¸€ä¸ªå‚è€ƒã€‚</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FRiP.JPG"/>

<p><strong>FRiPè¯„ä¼°ç›¸å…³å‚è€ƒèµ„æ–™ï¼š</strong></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation">https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation</a></p>
<h2 id="ä¸ƒã€Peak-annotation"><a href="#ä¸ƒã€Peak-annotation" class="headerlink" title="ä¸ƒã€Peak annotation"></a>ä¸ƒã€Peak annotation</h2><h3 id="1ã€Feature-Distribution"><a href="#1ã€Feature-Distribution" class="headerlink" title="1ã€Feature Distribution"></a>1ã€Feature Distribution</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">&quot;path to bed file&quot;</span>)</span><br><span class="line">library(ChIPpeakAnno)</span><br><span class="line">library(TxDb.Mmusculus.UCSC.mm10.knownGene)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(BiocInstaller)</span><br><span class="line">library(ChIPseeker)</span><br><span class="line">txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter &lt;- getPromoters(TxDb=txdb, upstream=<span class="number">3000</span>, downstream=<span class="number">3000</span>)</span><br><span class="line">files = <span class="built_in">list</span>(DMSO_24h_wt_rep1 = <span class="string">&quot;SRR12692092_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             DMSO_24h_wt_rep2 = <span class="string">&quot;SRR12692093_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             BRM014_10uM_24h_wt_rep1 = <span class="string">&quot;SRR12692098_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             BRM014_10uM_24h_wt_rep2 = <span class="string">&quot;SRR12692099_summits_filt_blacklist.bed&quot;</span>)</span><br><span class="line"><span class="comment">#æ±‡æ€»æ‰€æœ‰æ ·æœ¬</span></span><br><span class="line"><span class="comment">#plotAnnoBarå’ŒplotDistToTSSè¿™ä¸¤ä¸ªæŸ±çŠ¶å›¾éƒ½æ”¯æŒå¤šä¸ªæ•°æ®åŒæ—¶å±•ç¤º</span></span><br><span class="line">peakAnnoList &lt;- lapply(files, annotatePeak, </span><br><span class="line">                       TxDb=txdb,</span><br><span class="line">                       tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>))</span><br><span class="line">plotAnnoBar(peakAnnoList,title = <span class="string">&quot;                        Feature Distribution&quot;</span>)</span><br><span class="line">plotDistToTSS(peakAnnoList,title = <span class="string">&quot;                 Feature Distribution relative to TSS&quot;</span>)</span><br><span class="line"><span class="comment">#ä¾‹ä¸¾å•ä¸ªæ ·æœ¬</span></span><br><span class="line">peakAnno &lt;- annotatePeak(files[[<span class="number">1</span>]],<span class="comment"># åˆ†åˆ«æ”¹æˆ2æˆ–è€…3æˆ–è€…4å³å¯ï¼Œåˆ†åˆ«å¯¹åº”å››ä¸ªæ–‡ä»¶</span></span><br><span class="line">                        tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                        TxDb=txdb,</span><br><span class="line">                        annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">plotAnnoPie(peakAnnoLipeakAnnost)</span><br><span class="line">upsetplot(peakAnno, vennpie=<span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution TSS.jpeg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotAnnoPie.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/upsetplot(peakAnno, vennpie=TRUE).jpeg" width="50%"></p>
<h3 id="2ã€æŸ¥çœ‹peaksåœ¨å…¨åŸºå› ç»„ä¸Šçš„åˆ†å¸ƒ"><a href="#2ã€æŸ¥çœ‹peaksåœ¨å…¨åŸºå› ç»„ä¸Šçš„åˆ†å¸ƒ" class="headerlink" title="2ã€æŸ¥çœ‹peaksåœ¨å…¨åŸºå› ç»„ä¸Šçš„åˆ†å¸ƒ"></a>2ã€æŸ¥çœ‹peaksåœ¨å…¨åŸºå› ç»„ä¸Šçš„åˆ†å¸ƒ</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#è¾“å…¥æ–‡ä»¶çš„å‡†å¤‡</span></span><br><span class="line">DMSO_24h_wt_rep1&lt;-read.csv(<span class="string">&quot;SRR12692092_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">DMSO_24h_wt_rep1&lt;-DMSO_24h_wt_rep1[,-<span class="number">4</span>]</span><br><span class="line">DMSO_24h_wt_rep2&lt;-read.csv(<span class="string">&quot;SRR12692093_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">DMSO_24h_wt_rep2&lt;-DMSO_24h_wt_rep2[,-<span class="number">4</span>]</span><br><span class="line">BRM014_10uM_24h_wt_rep1&lt;-read.csv(<span class="string">&quot;SRR12692098_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">BRM014_10uM_24h_wt_rep1&lt;-BRM014_10uM_24h_wt_rep1[,-<span class="number">4</span>]</span><br><span class="line">BRM014_10uM_24h_wt_rep2&lt;-read.csv(<span class="string">&quot;SRR12692099_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">BRM014_10uM_24h_wt_rep2&lt;-BRM014_10uM_24h_wt_rep2[,-<span class="number">4</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment">#ä»¥DMSO_24h_wt_rep1ä¸ºä¾‹</span></span><br><span class="line">set.seed(<span class="number">123</span>)</span><br><span class="line">circos.initializeWithIdeogram(plotType = <span class="built_in">c</span>(<span class="string">&quot;axis&quot;</span>, <span class="string">&quot;labels&quot;</span>))</span><br><span class="line">circos.track(ylim = <span class="built_in">c</span>(<span class="number">0</span>, <span class="number">1</span>), panel.fun = <span class="keyword">function</span>(x, y) &#123;</span><br><span class="line">  chr = CELL_META$sector.index</span><br><span class="line">  xlim = CELL_META$xlim</span><br><span class="line">  ylim = CELL_META$ylim</span><br><span class="line">  circos.rect(xlim[<span class="number">1</span>], <span class="number">0</span>, xlim[<span class="number">2</span>], <span class="number">1</span>)</span><br><span class="line">&#125;, track.height = <span class="number">0.15</span>, bg.border = <span class="literal">NA</span>, bg.col=rainbow(<span class="number">24</span>))</span><br><span class="line">text(<span class="number">0</span>, <span class="number">0</span>, <span class="string">&quot;DMSO_24h_wt_rep1&quot;</span>, cex = <span class="number">1.5</span>)</span><br><span class="line">circos.genomicDensity(DMSO_24h_wt_rep1, col = <span class="built_in">c</span>(<span class="string">&quot;#000080&quot;</span>), track.height = <span class="number">0.2</span>)</span><br><span class="line">circos.clear()</span><br></pre></td></tr></table></figure>
<p>çœ‹åˆ°è¿™æ ·çš„ç»“æœï¼Œç¬¬ä¸€ååº”å°±æ˜¯â€”â€”â€”â€”ä¸ºä»€ä¹ˆä¸¤ç§å¤„ç†æƒ…å†µä¸‹æŸ“è‰²ä½“å¼€æ”¾ç¨‹åº¦é‚£ä¹ˆåƒï¼ï¼Ÿéš¾é“æˆ‘ä»£ç æœ‰é—®é¢˜ï¼ï¼Ÿç»è¿‡åå¤æ£€æŸ¥éªŒè¯ï¼ˆå°†ä¸€ä¸ªæ ·æœ¬chr1ä¸Šçš„peakséƒ½åˆ æ‰ï¼Œå†æ¬¡è¿è¡Œä¸Šè¿°ä»£ç ï¼Œå°±ä¼šå‘ç°æ˜¾è‘—çš„æ”¹å˜ï¼‰ï¼Œå¯ä»¥ç¡®å®šåˆ†æä¸Šæ˜¯æ²¡æœ‰é—®é¢˜çš„ã€‚è¿™ä¸¤ç§å¤„ç†å¯¼è‡´çš„å·®å¼‚å¯èƒ½ä¸æ˜¯å¾ˆæ˜¾è‘—ã€‚å†åŠ ä¸Š20ä¸‡+çš„peaksæ”¾åœ¨è¿™ä¸ªå°å°çš„circoså›¾ä¸Šå±•ç¤ºï¼Œæœ‰äº›å·®å¼‚ä¼šè¢«æ©ç›–æ‰ã€‚å°±å¦‚åœ¨åšTSSå¯Œé›†åˆ†æçš„æ—¶å€™ï¼Œå•ç‹¬çœ‹TSSå‰å3KbåŒºåŸŸï¼Œå¯ä»¥çœ‹åˆ°æœ‰ä¸¤ä¸ªå³°ï¼Œä½†åœ¨çœ‹TSS-genebody-TSEåŒºåŸŸæ˜¯ï¼ŒTSSå¤„ç›¸å¯¹å¾®å¼±çš„é‚£ä¸ªå³°å°±è¢«æ©ç›–æ‰äº†ã€‚</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep2.jpeg" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep2.jpeg" width="50%"></p>
<h3 id="3ã€æ‹¿åˆ°æ¯ä¸ªæ ·æœ¬ä¸­peakså¯¹åº”å¾—åŸºå› å"><a href="#3ã€æ‹¿åˆ°æ¯ä¸ªæ ·æœ¬ä¸­peakså¯¹åº”å¾—åŸºå› å" class="headerlink" title="3ã€æ‹¿åˆ°æ¯ä¸ªæ ·æœ¬ä¸­peakså¯¹åº”å¾—åŸºå› å"></a>3ã€æ‹¿åˆ°æ¯ä¸ªæ ·æœ¬ä¸­peakså¯¹åº”å¾—åŸºå› å</h3><p>è¿™ä¸€æ­¥éå¸¸é‡è¦ï¼Œæ‹¿åˆ°åŸºå› åå°±å¯ä»¥æ ¹æ®è¯¾é¢˜éœ€è¦è¿›è¡Œå·®å¼‚åˆ†æç­‰</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#ä»¥DMSO_24h_wtæ ·æœ¬ä¸ºä¾‹</span></span><br><span class="line"><span class="comment">#replicate 1</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep1 &lt;- annotatePeak(files[[<span class="number">1</span>]],</span><br><span class="line">                                          tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                                          TxDb=txdb,</span><br><span class="line">                                          annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">genelist_DMSO_24h_wt_rep1_uniqe&lt;-as.data.frame(unique(peakAnno_DMSO_24h_wt_rep1@anno@elementMetadata@listData[[<span class="string">&quot;SYMBOL&quot;</span>]]))</span><br><span class="line">colnames(genelist_DMSO_24h_wt_rep1_uniqe)&lt;-<span class="string">&quot;symbol&quot;</span></span><br><span class="line"><span class="comment">#replicate 2</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep2 &lt;- annotatePeak(files[[<span class="number">2</span>]],</span><br><span class="line">                                          tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                                          TxDb=txdb,</span><br><span class="line">                                          annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">genelist_DMSO_24h_wt_rep2_uniqe&lt;-as.data.frame(unique(peakAnno_DMSO_24h_wt_rep2@anno@elementMetadata@listData[[<span class="string">&quot;SYMBOL&quot;</span>]]))</span><br><span class="line">colnames(genelist_DMSO_24h_wt_rep2_uniqe)&lt;-<span class="string">&quot;symbol&quot;</span></span><br><span class="line"><span class="comment">#é‡å¤æ ·æœ¬é—´å…±åŒçš„å¼€æ”¾åŸºå› </span></span><br><span class="line">venn.diagram(</span><br><span class="line">  x=<span class="built_in">list</span>(</span><br><span class="line">    DMSO_24h_wt_rep1=genelist_DMSO_24h_wt_rep1_uniqe$symbol,</span><br><span class="line">    DMSO_24h_wt_rep2=genelist_DMSO_24h_wt_rep2_uniqe$symbol</span><br><span class="line">  ),</span><br><span class="line">  filename = <span class="string">&quot;DMSO_24h_wt.png&quot;</span>,</span><br><span class="line">  lty=<span class="string">&quot;dotted&quot;</span>,</span><br><span class="line">  lwd=<span class="number">3</span>,</span><br><span class="line">  col=<span class="string">&quot;transparent&quot;</span>,</span><br><span class="line">  fill=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;cornflowerblue&quot;</span>),</span><br><span class="line">  alpha=<span class="number">0.5</span>,</span><br><span class="line">  </span><br><span class="line">  label.col=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;white&quot;</span>,<span class="string">&quot;darkblue&quot;</span>) ,</span><br><span class="line">  cex=<span class="number">1</span>,</span><br><span class="line">  fontfamily=<span class="string">&quot;serif&quot;</span>,</span><br><span class="line">  fontface=<span class="string">&quot;bold&quot;</span>,</span><br><span class="line">  cat.default.pos=<span class="string">&quot;text&quot;</span>,</span><br><span class="line">  cat.col=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;darkblue&quot;</span>),</span><br><span class="line">  cat.cex=<span class="number">0.6</span>,</span><br><span class="line">  cat.fontfamily=<span class="string">&quot;serif&quot;</span>,</span><br><span class="line">  cat.dist=<span class="built_in">c</span>(<span class="number">0.3</span>,<span class="number">0.3</span>),</span><br><span class="line">  cat.pos=<span class="number">0</span></span><br><span class="line">)</span><br><span class="line"></span><br><span class="line"><span class="comment">#æŸ¥çœ‹å„ç»„å†…æ ·æœ¬é—´çš„overlapping readsï¼šDMSO_24h_wtï¼Œ BRM014_10uM_24h_wtï¼›</span></span><br><span class="line"><span class="comment">#ä»¥åŠç»„é—´peaksçš„å¼‚åŒæƒ…å†µï¼šDMSO_24h_wt vs. BRM014_10uM_24h_wt</span></span><br><span class="line"><span class="comment">#ä»£ç ç±»ä¼¼ä¸Šé¢çš„ï¼Œå°±ä¸ä¸€ä¸€å±•ç¤ºäº†</span></span><br></pre></td></tr></table></figure>
<p>ä»ä¸‹å›¾å¯ä»¥çœ‹å‡ºï¼Œä¸ç®¡æ˜¯ç»„é—´è¿˜æ˜¯ç»„å†…ï¼Œå·®å¼‚çš„peaksæ•°ç›®éƒ½ä¸æ˜¯å¾ˆå¤šäº†ï¼Œè¿™ä¸€ç‚¹ä¹ŸéªŒè¯äº†æˆ‘ä»¬ä¸Šé¢åšçš„å†å…¨åŸºå› ç»„èŒƒå›´å†…æŸ¥çœ‹peaksçš„åˆ†å¸ƒç»“æœã€‚</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png" width="50%"></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/12/08/2020%20ATAC-seq%20analysis/" data-id="ckl4ra1n2001aagvf49ugfsmu" data-title="2020 ATAC-seq analysis" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-NGSæ•°æ®åˆ†æä¹‹è¡¨è§‚" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/" class="article-date">
  <time class="dt-published" datetime="2020-05-14T03:16:18.000Z" itemprop="datePublished">2020-05-14</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGSæ•°æ®åˆ†æä¹‹è¡¨è§‚</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p> NGSæ•°æ®åˆ†æä¹‹è¡¨è§‚ä¸­çš„ä¸€äº›æ³¨æ„äº‹é¡¹</p>
        
          <p class="article-more-link">
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/" data-id="ckl4ra1ls0009agvf744q336w" data-title="NGSæ•°æ®åˆ†æä¹‹è¡¨è§‚" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-UEFI" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/04/04/UEFI/" class="article-date">
  <time class="dt-published" datetime="2020-04-04T06:44:56.000Z" itemprop="datePublished">2020-04-04</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/04/04/UEFI/">UEFI</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>windowså¯åŠ¨æ¨¡å¼åŒ…æ‹¬ä¸¤ç§ï¼šUEFI+GPTã€Legacy+MBRã€‚<br>åœ¨æ­¤ä»‹ç»3ä¸­åœ¨windowsä¸ŠæŸ¥çœ‹ç³»ç»Ÿå¯åŠ¨æ¨¡å¼çš„æ–¹æ³•ã€‚</p>
        
          <p class="article-more-link">
            <a href="/2020/04/04/UEFI/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/04/04/UEFI/" data-id="ckl4ra1ly000gagvfa0qfehby" data-title="UEFI" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/System/" rel="tag">System</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-gawk-è¿›é˜¶" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/03/16/gawk-%E8%BF%9B%E9%98%B6/" class="article-date">
  <time class="dt-published" datetime="2020-03-16T10:18:53.000Z" itemprop="datePublished">2020-03-16</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/03/16/gawk-%E8%BF%9B%E9%98%B6/">gawk è¿›é˜¶</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>ä½¿ç”¨å˜é‡ã€å¤„ç†æ•°ç»„ã€å¤„ç†æ¨¡å¼</p>
        
          <p class="article-more-link">
            <a href="/2020/03/16/gawk-%E8%BF%9B%E9%98%B6/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/03/16/gawk-%E8%BF%9B%E9%98%B6/" data-id="ckl4ra1lz000hagvfho5p9dpv" data-title="gawk è¿›é˜¶" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-sed-è¿›é˜¶" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/03/09/sed-%E8%BF%9B%E9%98%B6/" class="article-date">
  <time class="dt-published" datetime="2020-03-09T01:43:03.000Z" itemprop="datePublished">2020-03-09</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/03/09/sed-%E8%BF%9B%E9%98%B6/">sed è¿›é˜¶</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>å¤šè¡Œå‘½ä»¤ã€ä¿æŒç©ºé—´ã€æ’é™¤å‘½ä»¤ã€æ”¹å˜æµã€æ¨¡å¼æ›¿ä»£ã€åœ¨è„šæœ¬ä¸­ä½¿ç”¨sedã€åˆ›å»ºsedå®ç”¨ç¨‹åº</p>
        
          <p class="article-more-link">
            <a href="/2020/03/09/sed-%E8%BF%9B%E9%98%B6/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/03/09/sed-%E8%BF%9B%E9%98%B6/" data-id="ckl4ra1ma000tagvf6nlj1na9" data-title="sed è¿›é˜¶" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-sedç¼–è¾‘å™¨åŸºç¡€" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/03/07/sed%E7%BC%96%E8%BE%91%E5%99%A8%E5%9F%BA%E7%A1%80/" class="article-date">
  <time class="dt-published" datetime="2020-03-07T10:41:22.000Z" itemprop="datePublished">2020-03-07</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/03/07/sed%E7%BC%96%E8%BE%91%E5%99%A8%E5%9F%BA%E7%A1%80/">sedç¼–è¾‘å™¨åŸºç¡€</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>sedç¼–è¾‘å™¨æ›´å¤šçš„å‘½ä»¤å’Œæ ¼å¼ï¼š<br>æ›´å¤šçš„æ›¿æ¢é€‰é¡¹ã€ä½¿ç”¨åœ°å€ã€åˆ é™¤è¡Œã€æ’å…¥å’Œé™„åŠ æ–‡æœ¬ã€ä¿®æ”¹è¡Œã€è½¬æ¢å‘½ä»¤ã€æ‰“å°ã€ä½¿ç”¨sedå¤„ç†æ–‡ä»¶</p>
        
          <p class="article-more-link">
            <a href="/2020/03/07/sed%E7%BC%96%E8%BE%91%E5%99%A8%E5%9F%BA%E7%A1%80/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/03/07/sed%E7%BC%96%E8%BE%91%E5%99%A8%E5%9F%BA%E7%A1%80/" data-id="ckl4ra1mc000wagvf9hqe3nsy" data-title="sedç¼–è¾‘å™¨åŸºç¡€" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-åˆæ­¥è®¤è¯†sedå’Œgwak" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/03/07/%E5%88%9D%E6%AD%A5%E8%AE%A4%E8%AF%86sed%E5%92%8Cgwak/" class="article-date">
  <time class="dt-published" datetime="2020-03-07T06:31:16.000Z" itemprop="datePublished">2020-03-07</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/03/07/%E5%88%9D%E6%AD%A5%E8%AE%A4%E8%AF%86sed%E5%92%8Cgwak/">åˆæ­¥è®¤è¯†sedå’Œgwak</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>æƒ³åœ¨shellè„šæœ¬ä¸­å¤„ç†ä»»ä½•ç±»å‹çš„æ•°æ®ï¼Œå°±è¦ç†Ÿæ‚‰Linuxä¸­çš„sedå’Œgawkå·¥å…·ã€‚Linuxä¸–ç•Œä¸­å¹¿æ³›ä½¿ç”¨çš„ä¸¤ä¸ªå‘½ä»¤è¡Œç¼–è¾‘å™¨ï¼šsedå’Œgawk</p>
        
          <p class="article-more-link">
            <a href="/2020/03/07/%E5%88%9D%E6%AD%A5%E8%AE%A4%E8%AF%86sed%E5%92%8Cgwak/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/03/07/%E5%88%9D%E6%AD%A5%E8%AE%A4%E8%AF%86sed%E5%92%8Cgwak/" data-id="ckl4ra1md000yagvf9k8zg1hr" data-title="åˆæ­¥è®¤è¯†sedå’Œgwak" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li></ul>

    </footer>
  </div>
  
</article>



  


  <nav id="page-nav">
    
    <span class="page-number current">1</span><a class="page-number" href="/page/2/">2</a><a class="page-number" href="/page/3/">3</a><a class="extend next" rel="next" href="/page/2/">Next &raquo;</a>
  </nav>

</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/System/" rel="tag">System</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/notes/" rel="tag">notes</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Linux/" style="font-size: 16.67px;">Linux</a> <a href="/tags/MySQL/" style="font-size: 13.33px;">MySQL</a> <a href="/tags/NGS/" style="font-size: 20px;">NGS</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/System/" style="font-size: 10px;">System</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/notes/" style="font-size: 10px;">notes</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/01/">January 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">December 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/05/">May 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/04/">April 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/03/">March 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/02/">February 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
          </li>
        
          <li>
            <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
          </li>
        
          <li>
            <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
          </li>
        
          <li>
            <a href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
          </li>
        
          <li>
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGSæ•°æ®åˆ†æä¹‹è¡¨è§‚</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2021 gongyuqiqi<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>