<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>scRNA-seq数据分析流程 | gyq&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="scRNA-seq数据分析流程以下分析流程以GSE111229数据为例(为什么用它呢，因为它数据量小呀，嘿嘿🤭)在进行分析之前，首先了解一下该套数据的基本情况。   由此可基本了解，此数据为单端测序数据，物种为小鼠。768个样本即768个单细胞测序数据，总大小才10G。这在单细胞测序数据中真的是非常小！非常罕见啦！ 😄数据分析上游(Linux)😄因为是单细胞转录组测序数据，和转录组测序的">
<meta property="og:type" content="article">
<meta property="og:title" content="scRNA-seq数据分析流程">
<meta property="og:url" content="http://example.com/2020/02/07/scRNA-seq/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="scRNA-seq数据分析流程以下分析流程以GSE111229数据为例(为什么用它呢，因为它数据量小呀，嘿嘿🤭)在进行分析之前，首先了解一下该套数据的基本情况。   由此可基本了解，此数据为单端测序数据，物种为小鼠。768个样本即768个单细胞测序数据，总大小才10G。这在单细胞测序数据中真的是非常小！非常罕见啦！ 😄数据分析上游(Linux)😄因为是单细胞转录组测序数据，和转录组测序的">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GSE111229information.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GSE111229datasize.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/environment.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI2.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI3.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/hisat2mm10.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mm10注释.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ALIGN.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/bamsamfastq.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/df.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tSNE.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wholePCA.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wholetSNE.jpeg">
<meta property="article:published_time" content="2020-02-07T08:30:23.000Z">
<meta property="article:modified_time" content="2021-02-14T06:31:44.564Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GSE111229information.JPG">
  
    <link rel="alternate" href="/atom.xml" title="gyq's blog" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 5.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">gyq&#39;s blog</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><article id="post-scRNA-seq" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/07/scRNA-seq/" class="article-date">
  <time class="dt-published" datetime="2020-02-07T08:30:23.000Z" itemprop="datePublished">2020-02-07</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      scRNA-seq数据分析流程
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="scRNA-seq数据分析流程"><a href="#scRNA-seq数据分析流程" class="headerlink" title="scRNA-seq数据分析流程"></a>scRNA-seq数据分析流程</h1><p>以下分析流程以GSE111229数据为例(为什么用它呢，因为它数据量小呀，嘿嘿🤭)在进行分析之前，首先了解一下该套数据的基本情况。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GSE111229information.JPG" width="30%" height="30%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GSE111229datasize.JPG"/>  </p>
<p>由此可基本了解，此数据为<font color=red>单端测序</font>数据，物种为<font color=red>小鼠</font>。768个样本即768个单细胞测序数据，总大小才10G。这在单细胞测序数据中真的是<strong>非常小</strong>！<strong>非常罕见</strong>啦！</p>
<h2 id="😄数据分析上游-Linux-😄"><a href="#😄数据分析上游-Linux-😄" class="headerlink" title="😄数据分析上游(Linux)😄"></a>😄数据分析上游(Linux)😄</h2><p>因为是单细胞<strong>转录组</strong>测序数据，和转录组测序的上游分析没什么区别。所以以下分析流程都是在conda构建的rna环境下进行，这个环境也是很久之前搭建好的，在此不再赘述。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda activat rna</span><br></pre></td></tr></table></figure>

<p>运行上述代码后，可以看到提示符(prompt)左边多了一个<u>(rna)</u>环境，此环境里面，预先装好了处理RNA-seq数据所需要的相关软件。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/environment.JPG"/></p>
<h3 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h3><blockquote>
<p><strong>如果样本量不多，网速又不理想</strong>，推荐<a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/">EBI-ENA</a>数据库，此数据库下载网速还可以，另外该数据库除了提供SRA文件外，还提供FASTQ文件，省去了从NCBI下载SRA文件转为FASTQ文件的麻烦(如何单个文件数据量大，此格式转换步骤非常耗时)。<br><strong>如果样本量特别多，网速又比较理想</strong>，推荐使用Aspera、prfetch、wget、curl。前两者大大优于后两者。Aspera Connect的下载速度是最快了，此方法也可以下载FASTQ和SRA文件，免去了SRA转至FASTQ的过程，该过程很耗时，十分耗时。 </p>
</blockquote>
<h4 id="进入EBI官网，输入SRP号"><a href="#进入EBI官网，输入SRP号" class="headerlink" title="进入EBI官网，输入SRP号"></a>进入EBI官网，输入SRP号</h4><p><em>1</em><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI.JPG"/><br><em>2</em><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI2.JPG"/><br><em>3</em><br>以其中一个样本SRR6791132为例，点击SRR6791132进入下载界面，鼠标右键红色箭头处——“复制链接地址”<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/EBI3.JPG"/></p>
<h4 id="进入linux进行数据下载"><a href="#进入linux进行数据下载" class="headerlink" title="进入linux进行数据下载"></a>进入linux进行数据下载</h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p scRNA-seq/fastq &amp;&amp; <span class="built_in">cd</span> scRNA-seq/fastq  </span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR679/002/SRR6791132/SRR6791132.fastq.gz    </span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR679/005/SRR6791135/SRR6791135.fastq.gz  </span><br><span class="line">wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR679/005/SRR6791135/SRR6791135.fastq.gz</span><br><span class="line">```  </span><br><span class="line">&lt;font color=blue&gt;注意注意，特别注意，要下载整套数据最好使用Aspera或者prefetch，本宝宝没有办法呀，网不好，空间也不大，linux中的数据操作只能用三个样本做演示了，熟悉一下分析流程。如果是整套数据，记得写小循环，放后台，就不用一直守着它了，最好是睡觉的时候跑，不耽误事。&lt;/font&gt;    </span><br><span class="line"></span><br><span class="line">   </span><br><span class="line"><span class="comment">### 质控  </span></span><br><span class="line">```bash   </span><br><span class="line"><span class="comment">#首先查看数据质量情况，进行如下操作 </span></span><br><span class="line">mkdir -p /scRNA-seq/html &amp;&amp; <span class="built_in">cd</span> /scRNA-seq/html</span><br><span class="line">ls /scRNA-seq/fastq/*gz|xargs fastqc -t 4 -o ./</span><br><span class="line">```   </span><br><span class="line">上述操作会得到html文件和zip文件  </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/quality.JPG&quot;</span>/&gt;</span><br><span class="line"></span><br><span class="line">网页打开其中一个html文件可查看测序数据的质量，该数据整体质量还不错（[质控报告解读教程](https://www.jianshu.com/p/835fd925d6ee)） </span><br><span class="line"></span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/html.JPG&quot;</span> width=<span class="string">&quot;50%&quot;</span> height=<span class="string">&quot;50%&quot;</span>&gt;  </span><br><span class="line"></span><br><span class="line">大部分测序数据都会出现如下情况，不过这个问题不大  </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pbsc.JPG&quot;</span> width=<span class="string">&quot;50%&quot;</span> height=<span class="string">&quot;50%&quot;</span>&gt;  </span><br><span class="line">```bash  </span><br><span class="line"><span class="comment">#接下来过滤掉质量不好的数据，使trim_galore软件  </span></span><br><span class="line"><span class="comment">#mkdir /scRNA-seq/clean</span></span><br><span class="line"><span class="comment">#ls /scRNA-seq/fastq/*gz|while read id; do trim_galore -q 25 --phred33 --length 35 -e 0.1 --stringency 5 -o . $id;done  </span></span><br><span class="line"><span class="comment">#在此会发现过滤后的数据质量反而没有过滤前的质量好，有时候数据分析就是会这样，过滤参数很难调整。。。所以在此后续步骤使用未过滤的数据进行处理。  </span></span><br></pre></td></tr></table></figure>
<h3 id="比对"><a href="#比对" class="headerlink" title="比对"></a>比对</h3><p>1、小鼠参考基因组<br>首先下载<font color=red>小鼠参考基因组</font>，hisat2或者bowtie2构建<font color=red>小鼠参考基因组索引</font>。也可以直接下载参考基因组索引文件。另外还需要下载<font color=red>小鼠参考基因组注释文件</font>。(此步骤很早前处理RNA-seq数据时就已经完成了，就不再赘述)<br><em>hisat2构建的mm10参考基因组索引文件</em><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/hisat2mm10.JPG"/><br><em>mm10注释文件</em><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mm10注释.JPG"/>   </p>
<p>2、进行参考基因组比对</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#同参考基因组索引进行比对，生成bam文件。参考基因组索引一定是引用前缀&quot;genome&quot;。另外注意，此处数据为单端测序数据！</span></span><br><span class="line">ls *.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> hisat2 -p 4 -x /addfirst/reference/mouse/index/hisat2-build-index/mm10/genome -U <span class="variable">$id</span> -S <span class="variable">$&#123;id%%.*&#125;</span>.hisat2.sam;<span class="keyword">done</span>   </span><br><span class="line">mkdir /scRNA-seq/align &amp;&amp; <span class="built_in">cd</span> /scRNA-seq/align   </span><br><span class="line">cp /scRNA-seq/fastq/*.sam /scRNA-seq/fastq/*.bam ./  </span><br><span class="line"><span class="comment">#使用samtools软件将sam文件转成bam文件  </span></span><br><span class="line">ls *.sam | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools sort -O bam -@ 4 -o <span class="variable">$&#123;id%%.*&#125;</span>.hisat2.bam <span class="variable">$id</span>; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>比对结果如下：82.25%的比对率，比较的凑合了，一般情况最好90%以上，后两个数据的比对率实在不能看。真实处理环境一定要<strong>仔细调节过滤参数！！！</strong><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ALIGN.JPG"/></p>
<p>可以比较一下bam、fasatq、sam文件的大小。<br>sam文件远远大于bam和fastq文件。生信分析上游一定要对文件大小保持高度敏感，避免不必要的错误。另外，数据量大的话，可以在生成sam文件时通过<code>|</code>将结果进一步生成bam文件，这样可以大大节省存储空间。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/bamsamfastq.JPG"/>   </p>
<h3 id="构建index文件"><a href="#构建index文件" class="headerlink" title="构建index文件"></a>构建index文件</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | xargs -i samtools index &#123;&#125;</span><br><span class="line">```  </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/bamindex.JPG&quot;</span>/&gt;</span><br><span class="line"></span><br><span class="line"><span class="comment">### 生成reads表达矩阵  </span></span><br><span class="line">```bash </span><br><span class="line">gtf=/addfirst/reference/mouse/gencode.vM23.annotation.gtf   </span><br><span class="line">featureCounts -T 4 -p -t exon -g gene_id -a <span class="variable">$gtf</span> -o all.id.txt *.bam 1&gt;counts.id.log 2&gt;&amp;1   </span><br><span class="line">查看counts矩阵  </span><br><span class="line">less -S all.id.txt</span><br><span class="line">```   </span><br><span class="line"> &lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/reads.JPG&quot;</span>/&gt;    </span><br><span class="line"> 查看counts矩阵会发现绝大多数基因表达量都是0，因为对于大多数细胞来说，大多数基因都是测不到的，这个属于正常现象。在此，我们只用了3个样本做演示，总数据则有768个样本。  </span><br><span class="line"> &lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/countsmatrix.JPG&quot;</span>/&gt;    </span><br><span class="line"> 理论上按照上面的分析流程生成的表达矩阵数据和下图中NCBI官网上的rawCounts.txt.gz的内容应该是差不多的，只是所用处理软件不同会有一点点区别。  </span><br><span class="line"> &lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NCBImatrix.JPG&quot;</span>/&gt;</span><br><span class="line"></span><br><span class="line">至此，scRNA-seq上游分析结束，这里比较重要的是**数据质量**，数据质量直接决定比对的效率，影响最终生成的表达矩阵的可靠性。所以质控步骤要严格把控，送样测序也要选靠谱的公司。   </span><br><span class="line"></span><br><span class="line"><span class="comment">## 😄数据分析下游(R)😄   </span></span><br><span class="line"><span class="comment">### 数据(data.frame)的构建</span></span><br><span class="line">如果是使用rawCounts数据进行下游分析，首先要对数据进行标准化，然后再做后续分析。该篇文章是使用rpkm的数据进行下游分析，这里也直接使用rpkm的数据进行下游分析。   </span><br><span class="line">```R   </span><br><span class="line">a=read.table(<span class="string">&#x27;../GSE111229_Mammary_Tumor_fibroblasts_768samples_rpkmNormalized.txt.gz&#x27;</span>,header = T ,sep = <span class="string">&#x27;\t&#x27;</span>)  <span class="comment">##把表达矩阵文件载入R，header=T :保留文件头部信息，seq=&#x27;\t&#x27;以tap为分隔符</span></span><br><span class="line"><span class="comment"># 记得检测数据</span></span><br><span class="line">a[1:6,1:4] <span class="comment">#对于a矩阵取第1~6行，第1~4列</span></span><br><span class="line"><span class="comment">## 读取RNA-seq的counts定量结果，表达矩阵需要进行简单的过滤</span></span><br><span class="line">dat=a[apply(a,1, <span class="keyword">function</span>(x) sum(x&gt;0) &gt; floor(ncol(a)/50)),]   </span><br><span class="line"><span class="comment">#筛选表达量合格的行,列数不变   </span></span><br><span class="line"><span class="comment">#上面的apply()指令代表对矩阵a进行行计算，判断每行表达量&gt;1的样本总个数，并筛选出细胞表达量合格的基因（行）</span></span><br><span class="line"><span class="comment">#第一个参数是指要参与计算的矩阵——a</span></span><br><span class="line"><span class="comment">#第二个参数是指按行计算还是按列计算，1——表示按行计算，2——按列计算；</span></span><br><span class="line"><span class="comment">#第三个参数是指具体的运算参数,定义一个函数x（即表达量为x）</span></span><br><span class="line">dat[1:4,1:4]      </span><br><span class="line">```   </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/a.JPG&quot;</span>/&gt;  </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/dat.JPG&quot;</span>/&gt;</span><br><span class="line"></span><br><span class="line">   </span><br><span class="line">```R  </span><br><span class="line"><span class="comment">#层次聚类</span></span><br><span class="line">hc=hclust(dist(t(<span class="built_in">log</span>(dat+0.1)))) <span class="comment">##样本间层次聚类</span></span><br><span class="line"><span class="comment"># 如果是基因聚类，可以选择 wgcna 等算法  </span></span><br><span class="line">plot(hc,labels = F) <span class="comment">##此图可以看出678个样本大致可以分为几类，在此大致可分为4类   </span></span><br><span class="line">clus = cutree(hc, 4) <span class="comment">#对hclust()函数的聚类结果进行剪枝，即选择输出指定类别数的系谱聚类结果。</span></span><br><span class="line">group_list= as.factor(clus) <span class="comment">##转换为因子属性</span></span><br><span class="line">table(group_list) <span class="comment">##统计频数    </span></span><br><span class="line">```  </span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/hclust.JPG&quot;</span> width=<span class="string">&quot;60%&quot;</span> height=<span class="string">&quot;60%&quot;</span>&gt;    </span><br><span class="line"></span><br><span class="line">&lt;img src=<span class="string">&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/grouplist.JPG&quot;</span>/&gt;</span><br><span class="line"></span><br><span class="line">```R  </span><br><span class="line"><span class="comment">#提取批次信息</span></span><br><span class="line">colnames(dat) <span class="comment">#取列名</span></span><br><span class="line">library(stringr)</span><br><span class="line">plate=str_split(colnames(dat),<span class="string">&#x27;_&#x27;</span>,simplify = T)[,3] <span class="comment">#str_split()函数可以分割字符串。取列名，以&#x27;_&#x27;号分割，提取第三列。</span></span><br><span class="line">table(plate) <span class="comment">#以为这768个样本是放在两个384孔检测的，后续要检测测序数据是否存在批次效应，所以要预先提取批次信息 </span></span><br><span class="line">n_g = apply(a,2,<span class="keyword">function</span>(x) sum(x&gt;0)) <span class="comment">#统计每个样本有表达的基因有多少行</span></span><br><span class="line"><span class="comment">#reads数量大于1的那些基因为有表达，一般来说单细胞转录组过半数的基因是不会表达的。  </span></span><br><span class="line">df=data.frame(g=group_list,plate=plate,n_g=n_g) <span class="comment">#新建数据框(细胞的属性信息) </span></span><br><span class="line"><span class="comment">##(样本为行名，列分别为：样本分类信息，样本批次，样本表达的基因数——不是表达量的和，而是种类数或者说个数) </span></span><br><span class="line">df<span class="variable">$all</span>=<span class="string">&#x27;all&#x27;</span> <span class="comment">#添加列，列名为&quot;all&quot;，没事意思，就是后面有需要</span></span><br><span class="line">metadata=df  </span><br><span class="line">df[1:4,1:4]</span><br><span class="line">save(dat,metadata,file = <span class="string">&#x27;../input_rpkm.Rdata&#x27;</span>) <span class="comment">#保存a,dat,df这变量到上级目录的input.Rdata   </span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/df.JPG"/>    

<h3 id="检测是否存在批次效应"><a href="#检测是否存在批次效应" class="headerlink" title="检测是否存在批次效应"></a>检测是否存在批次效应</h3><p>因为该篇文章使用两块384孔进行测序，所以要先确定两块板是否存在批次效应，如果存在，则两块板的样本不能合并分析，如果不存在批次效应，则两块板可以进行批次分析。</p>
<p><strong>PCA主成份分析</strong>  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br></pre></td><td class="code"><pre><span class="line">rm(<span class="built_in">list</span> = ls())  <span class="comment">## 魔幻操作，一键清空~</span></span><br><span class="line">options(stringsAsFactors = <span class="built_in">F</span>)</span><br><span class="line">load(file = <span class="string">&#x27;../input_rpkm.Rdata&#x27;</span>)</span><br><span class="line">dat[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">head(metadata) </span><br><span class="line">plate=metadata$plate</span><br><span class="line"><span class="comment">## 下面是画PCA的必须操作</span></span><br><span class="line">dat_back=dat</span><br><span class="line">dat=t(dat)  <span class="comment">#做PCA之前一定要进行数据转置</span></span><br><span class="line">dat=as.data.frame(dat)</span><br><span class="line">dat=cbind(dat,plate ) <span class="comment">#cbind根据列进行合并，即叠加所有列，矩阵添加批次信息</span></span><br><span class="line">dat[<span class="number">1</span>:<span class="number">4</span>,<span class="number">12688</span>:<span class="number">12690</span>]  <span class="comment">#检测是否添加上了批次信息</span></span><br><span class="line">table(dat$plate)</span><br><span class="line">library(<span class="string">&quot;FactoMineR&quot;</span>)</span><br><span class="line">library(<span class="string">&quot;factoextra&quot;</span>) </span><br><span class="line">dat.pca &lt;- PCA(dat[,-ncol(dat)], graph = <span class="literal">FALSE</span>)</span><br><span class="line">fviz_pca_ind(dat.pca,<span class="comment">#repel =T, geom.ind = &quot;point&quot;, </span></span><br><span class="line">             col.ind = dat$plate, <span class="comment"># color by groups</span></span><br><span class="line">             addEllipses = <span class="literal">FALSE</span>, <span class="comment"># Concentration ellipses</span></span><br><span class="line">             legend.title = <span class="string">&quot;Groups&quot;</span></span><br><span class="line">)   </span><br><span class="line">```   </span><br><span class="line">&lt;img src=&quot;https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PCA.jpeg&quot; width=&quot;60%&quot; height=&quot;60%&quot;&gt;</span><br><span class="line">      </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">**tSNE分析**  </span><br><span class="line">```R </span><br><span class="line">rm(<span class="built_in">list</span> = ls())  <span class="comment">## 魔幻操作，一键清空~</span></span><br><span class="line">options(stringsAsFactors = <span class="built_in">F</span>)   </span><br><span class="line">library(Rtsne) </span><br><span class="line">load(file = <span class="string">&#x27;../input_rpkm.Rdata&#x27;</span>)</span><br><span class="line">dat[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">dat_matrix &lt;- t(dat)  </span><br><span class="line">dat_matrix=log2(dat_matrix+<span class="number">0.01</span>)</span><br><span class="line">set.seed(<span class="number">42</span>) <span class="comment"># 如果想得到可重复的结果，就种一颗种子吧😄</span></span><br><span class="line">tsne_out &lt;- Rtsne(dat_matrix,pca=<span class="literal">FALSE</span>,perplexity=<span class="number">30</span>,theta=<span class="number">0.0</span>)   </span><br><span class="line"><span class="comment">##添加颜色</span></span><br><span class="line">tsnes=tsne_out$Y</span><br><span class="line">tsnes=as.data.frame(tsnes)</span><br><span class="line">group=<span class="built_in">c</span>(<span class="built_in">rep</span>(<span class="string">&#x27;plate1&#x27;</span>,<span class="number">384</span>),<span class="built_in">rep</span>(<span class="string">&#x27;plate2&#x27;</span>,<span class="number">384</span>))</span><br><span class="line">colnames(tsnes)&lt;-<span class="built_in">c</span>(<span class="string">&quot;tSNE1&quot;</span>,<span class="string">&quot;tSNE2&quot;</span>)  </span><br><span class="line">library(ggfortify)</span><br><span class="line">ggplot(tsnes, aes(x = tSNE1, y = tSNE2))+ geom_point(aes(col=group))+ theme_classic()</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tSNE.jpeg" width="60%" height="60%">

<h3 id="细胞亚群"><a href="#细胞亚群" class="headerlink" title="细胞亚群"></a>细胞亚群</h3><pre><code class="R">#########################
##针对所有细胞做PCA分析##  
#########################
rm(list = ls())  ## 魔幻操作，一键清空
options(stringsAsFactors = F)
load(file = &#39;../input_rpkm.Rdata&#39;)
dat[1:4,1:4]
head(metadata) 
group&lt;-metadata$g
#做好数据备份
dat_back=dat
dat=t(dat) #记得转置数据
dat=as.data.frame(dat)
dat=cbind(dat,group ) #cbind根据列进行合并，即叠加所有列，为矩阵添加分组信息
dat[1:4,12688:12690]
table(dat$group)
library(&quot;FactoMineR&quot;)
library(&quot;factoextra&quot;) 
dat.pca &lt;- PCA(dat[,-ncol(dat)], graph = FALSE)
fviz_pca_ind(dat.pca,#repel =T,
             geom.ind = &quot;point&quot;, # show points only (nbut not &quot;text&quot;)
             col.ind = dat$group, # color by groups
             addEllipses = FALSE, # Concentration ellipses
             legend.title = &quot;Groups&quot;  
             )   

#########################
##针对所有细胞做tSNE分析##  
#########################
rm(list = ls())  ## 魔幻操作，一键清空~
options(stringsAsFactors = F)
library(Rtsne) 
load(file = &#39;../input_rpkm.Rdata&#39;)
dat[1:4,1:4]
dat_matrix &lt;- t(dat)
dat_matrix=log2(dat_matrix+0.01)
# Set a seed if you want reproducible results
set.seed(42)
tsne_out &lt;- Rtsne(dat_matrix,pca=FALSE,perplexity=30,theta=0.0) 
group=metadata$g
##添加颜色
tsnes=tsne_out$Y
tsnes=as.data.frame(tsnes)
colnames(tsnes)&lt;-c(&quot;tSNE1&quot;,&quot;tSNE2&quot;)
ggplot(tsnes, aes(x = tSNE1, y = tSNE2))+ geom_point(aes(col=group))+ theme_classic()
</code></pre>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wholePCA.jpeg" width="60%" height="60%">  
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/wholetSNE.jpeg" width="60%" height="60%">   

<p>单细胞测序分析旨在分出不同细胞亚群(展现细胞间的异质性)，力求寻找某类亚群与某特定生物学功能之间的联系。而要探讨细胞分群，就必须要知道PCA和tSNE。<br><strong>PCA(Principal Component Analysis)</strong><br><strong>tSNE(t-Distributed Stochastic Neighbor Embedding)</strong><br>什么样的算法才是最理想的能够将细胞分群的算法呢？<br>1）局部结构：属于同一个亚群的细胞，聚类尽可能近<br>2）全局结构：属于不同亚群的细胞，聚类尽可能分来   </p>
<ul>
<li>PCA的方法侧重于去抓样本中隐含的主要效应，从而让差异大的样本在图上呈现较远的距离。常规RNA-seq项目中，处理效应、批次效应、离群效应等属于较大的效应，PCA的方法可以良好的去展示这些效应。   </li>
<li>如果影响样本分组的不是主要效应，而是一些更小的效应，PCA则无法对样本进行准确的区分。scRNA-seq的可视化主要期望对各个细胞亚群有良好的区分。每次检测的上万甚至几十万个细胞中，几乎肯定可以区分出几十中细胞亚群，包括一些稀有的细胞。而区分这些细胞亚群（尤其是稀有细胞类型）的效应，往往不是主要效应（即大量基因的差异），而是一些微小效应（少量标记基因差异）。如果使用PCA进行分析的话，就会掩盖掉某些微小的但是有可能非常重要的细胞群。   </li>
<li>tSNE算法就属于可以同时兼顾局部结构和全局结构的非线性降维可视化算法。tSNE不同于PCA（PCA主要目标是尽量去抓取群体中的主要效益），tSNE算法的主要目标是：降维后的数据依然保持最为相似的紧密成簇。这就保证了哪怕某些稀有细胞只是少量基因区别于其他细胞亚群，在tSNE中依然可以与其他细胞有良好的区分。<font color=blue>结论：PCA是常规转录组常用的数据降维和样本关系可视化的方法，但针对群体单细胞转录组数据，tSNE是明显胜过PCA的方法！</font>   </li>
</ul>
<p>参考资料<br><a target="_blank" rel="noopener" href="http://www.sohu.com/a/320164491_278730">http://www.sohu.com/a/320164491_278730</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/07/scRNA-seq/" data-id="ckl4ra1n8001bagvf5f86b28r" data-title="scRNA-seq数据分析流程" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2020/02/08/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%80%EF%BC%89/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          scRNA-seq数据分析实战（一）
        
      </div>
    </a>
  
  
    <a href="/2020/02/01/Markdown%E5%9F%BA%E7%A1%80%E8%AF%AD%E6%B3%95%E6%95%99%E7%A8%8B/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">Markdown学习教程</div>
    </a>
  
</nav>

  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/System/" rel="tag">System</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/notes/" rel="tag">notes</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Linux/" style="font-size: 16.67px;">Linux</a> <a href="/tags/MySQL/" style="font-size: 13.33px;">MySQL</a> <a href="/tags/NGS/" style="font-size: 20px;">NGS</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/System/" style="font-size: 10px;">System</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/notes/" style="font-size: 10px;">notes</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/01/">January 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">December 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/05/">May 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/04/">April 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/03/">March 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/02/">February 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
          </li>
        
          <li>
            <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
          </li>
        
          <li>
            <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
          </li>
        
          <li>
            <a href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
          </li>
        
          <li>
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGS数据分析之表观</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2021 gongyuqiqi<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>