<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>scRNA-seq数据分析学习（四） | gyq&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="学习R包monocle对scRNA-seq数据进行分析">
<meta property="og:type" content="article">
<meta property="og:title" content="scRNA-seq数据分析学习（四）">
<meta property="og:url" content="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="学习R包monocle对scRNA-seq数据进行分析">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sc_sds.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filtergene.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filtercell.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotPC.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plottsne.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tableBiologicalananotation.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/difftestres.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/headdiffgene.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotdiffgenehead.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotdiffgeneheadcol.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotoderinggenes.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/trace.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tracemaker.JPG">
<meta property="article:published_time" content="2020-02-15T08:30:23.000Z">
<meta property="article:modified_time" content="2021-02-14T06:33:38.558Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sc_sds.JPG">
  
    <link rel="alternate" href="/atom.xml" title="gyq's blog" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 5.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">gyq&#39;s blog</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><article id="post-scRNA-seq数据分析学习（四）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-15T08:30:23.000Z" itemprop="datePublished">2020-02-15</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      scRNA-seq数据分析学习（四）
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>学习R包monocle对scRNA-seq数据进行分析</p>
<a id="more"></a>

<h1 id="scRNA-seq数据分析（四）"><a href="#scRNA-seq数据分析（四）" class="headerlink" title="scRNA-seq数据分析（四）"></a>scRNA-seq数据分析（四）</h1><h2 id="创建数据集"><a href="#创建数据集" class="headerlink" title="创建数据集"></a>创建数据集</h2><p>后续分析的前提就是将数据构建成monocle需要的对象<br>因此这里先介绍一下monocle需要的用来构建CellDataSet对象的三个数据集   </p>
<ul>
<li>表达量矩阵<code>exprs</code>：数据矩阵，行名是基因，列明是细胞编号</li>
<li>细胞的表型信息<code>phenoData</code>：第一列是细胞编号，其他列是细胞的相关信息</li>
<li>基因注释<code>featureData</code>：第一列是基因编号，其他列是基因对应的信息<br>这三个数据集要满足如下要求  </li>
</ul>
<p><strong>表达量矩阵</strong></p>
<ul>
<li>保证它的列数等于<code>phenoData</code>的行数</li>
<li>保证它的行数等于<code>featureData</code>的行数</li>
<li><code>phenoData</code>的行名需要和表达矩阵的列明匹配</li>
<li><code>featureData</code>和表达矩阵的行名要匹配</li>
<li><code>featureData</code>至少要有一列”gene_short_name”，就是基因的symbol  </li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">rm(<span class="built_in">list</span> = ls())<span class="comment">#clear the environment</span></span><br><span class="line">options(warn=-<span class="number">1</span>)<span class="comment">#turn off warning message globally</span></span><br><span class="line">suppressMessages(library(monocle))</span><br></pre></td></tr></table></figure>
<p><strong>这里同样使用scRNAseq R包中的数据集，构建monocle对象</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#首先需要一个表达矩阵，还需要临床信息</span></span><br><span class="line">library(scRNAseq)</span><br><span class="line">data(fluidigm)<span class="comment">#load examaple data</span></span><br><span class="line">assay(fluidigm)&lt;-assays(fluidigm)$rsem_counts<span class="comment">#set assay to RSEM estimated counts</span></span><br><span class="line">ct&lt;-<span class="built_in">floor</span>(assays(fluidigm)$rsem_counts)</span><br><span class="line">ct[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]<span class="comment">#表达矩阵</span></span><br><span class="line">sample_ann&lt;-as.data.frame(colData(fluidigm))<span class="comment">#临床信息</span></span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#准备monocle对象需要的phenotype和feature data以及表达矩阵，从scRNA-seq这个R包里面提取这三个数据</span></span><br><span class="line">gene_ann&lt;-data.frame(</span><br><span class="line">  gene_short_name=row.names(ct),</span><br><span class="line">  row.names = row.names(ct)</span><br><span class="line">)</span><br><span class="line">pd&lt;-new(<span class="string">&quot;AnnotatedDataFrame&quot;</span>,data=sample_ann)</span><br><span class="line">fd&lt;-new(<span class="string">&quot;AnnotatedDataFrame&quot;</span>,data=gene_ann)</span><br></pre></td></tr></table></figure>
<p>#构建monocle后续分析的所有对象，主要是根据包的说明书，仔细探索其需要的构建对象的必备元素<br>#因为表达矩阵是counts值，所以注意expressionFamily参数,其他类型的值用其他参数</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">sc_cds&lt;-newCellDataSet(</span><br><span class="line">  ct,</span><br><span class="line">  phenoData = pd,</span><br><span class="line">  featureData = fd,</span><br><span class="line">  expressionFamily = negbinomial.size(),</span><br><span class="line">  lowerDetectionLimit = <span class="number">1</span></span><br><span class="line">)</span><br><span class="line">sc_cds</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sc_sds.JPG"/>

<p><font color=red><strong>在此示范如何加载加载RPKM数据，在此并不运行该部分代码</strong></font><br>从本地读入RPKM值文件，构造CellDataSet对象  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#读入所需要的数据：表达矩阵，样本信息，基因信息</span></span><br><span class="line">expression_matrix&lt;-read.table(<span class="string">&quot;fpkm_matrix.txt&quot;</span>)</span><br><span class="line">sample_sheet&lt;-read.delin(<span class="string">&quot;cell_sample_sheet.txt&quot;</span>)</span><br><span class="line">gene_annotation&lt;-read.delin(<span class="string">&quot;gene_annotation.txt&quot;</span>)</span><br><span class="line"><span class="comment">#创建CellDataSet对象</span></span><br><span class="line">pd&lt;-new(<span class="string">&quot;AnnotatedDataFrame&quot;</span>,data=sample_sheet)</span><br><span class="line">fd&lt;-new(<span class="string">&quot;AnnotatedDataFrame&quot;</span>,data=gene_annotation)</span><br><span class="line">sc_cds&lt;-newCellDataSet(as.matrix(expression_matrix),</span><br><span class="line">                       phenoData=pd,featureData=fd)</span><br></pre></td></tr></table></figure>
<p>monocle和scater、seurat他们基于的对象不一样，所以monocle还提供了转换函数  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">lymphomadata&lt;-sc_cds</span><br><span class="line"><span class="comment">#转换成seurat对象</span></span><br><span class="line">lymphomadata_seurat&lt;-exportCDS(sc_cds,<span class="string">&#x27;Seurat&#x27;</span>)</span><br><span class="line"><span class="comment">#转换成SCESet对象</span></span><br><span class="line">lymphomadata_scater&lt;-exportCDS(sc_cds,<span class="string">&#x27;Scater&#x27;</span>)</span><br></pre></td></tr></table></figure>
<p><strong>下面是monocle对新构建的CellDataSet对象的标准操作</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">library(dplyr)</span><br><span class="line">colnames(phenoData(sc_cds)@data)</span><br><span class="line"><span class="comment">#estimateDispersions这步的时间和电脑配置密切相关,所以我们这套分析不选用monocle包自带的数据集，因为太大了，仍然采用scRNAseq内置数据集</span></span><br><span class="line">sc_cds&lt;-estimateSizeFactors(sc_cds)<span class="comment">#一定要运行</span></span><br><span class="line">sc_cds&lt;-estimateDispersions(sc_cds)<span class="comment">#一定要运行</span></span><br></pre></td></tr></table></figure>
<h1 id="质控"><a href="#质控" class="headerlink" title="质控"></a>质控</h1><p>对基因和细胞进行质量控制，指控指标根据课题进行具体探索，没有统一标准，这里只是简单演示 </p>
<p><strong>过滤基因</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">cds&lt;-sc_cds</span><br><span class="line">cds</span><br><span class="line">cds&lt;-detectGenes(cds,min_expr = <span class="number">0.1</span>)<span class="comment">#设置基因最小表达量为0.1</span></span><br><span class="line">print(head(fData(cds)))</span><br><span class="line">expressed_gene&lt;-row.names(subset(fData(cds),num_cells_expressed&gt;=<span class="number">5</span>))</span><br><span class="line"><span class="built_in">length</span>(expressed_gene)</span><br><span class="line">cds&lt;-cds[expressed_gene,]</span><br><span class="line">cds</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filtergene.JPG"/>


<p><strong>过滤细胞</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#过滤细胞</span></span><br><span class="line">print(head(pData(cds)))<span class="comment">#如果没有运行上述步骤的estimateSizeFactors、estimateDispersions，这一步就会发现某个属性出现NA值</span></span><br><span class="line">tmp&lt;-pData(cds)</span><br><span class="line">fivenum(tmp[,<span class="number">1</span>])</span><br><span class="line">fivenum(tmp[,<span class="number">30</span>])</span><br><span class="line"><span class="comment">#这里展示就先不过滤细胞了，有需要就自行过滤</span></span><br><span class="line">valid_cell&lt;-row.names(pData(cds))<span class="comment">#其实这里没有过滤细胞，valid_cell全为TRUE</span></span><br><span class="line">cds&lt;-cds[,valid_cell]</span><br><span class="line">cds</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filtercell.JPG"/>


<h2 id="聚类"><a href="#聚类" class="headerlink" title="聚类"></a>聚类</h2><p>单细胞转录组最重要的就是把细胞分群，相关的算法非常多，这里选用最常用的tSNE  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plot_pc_variance_explained(cds,return_all = <span class="built_in">F</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotPC.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#由上图决定用前6个组成份进行降维,降维采用tSNE的方法</span></span><br><span class="line">cds&lt;-reduceDimension(cds,max_components = <span class="number">2</span>,num_dim=<span class="number">6</span>,</span><br><span class="line">                     reduction_method = <span class="string">&#x27;tSNE&#x27;</span>,verbose = <span class="built_in">T</span>)</span><br><span class="line">cds&lt;-clusterCells(cds,num_clusters = <span class="number">4</span>)</span><br><span class="line">plot_cell_clusters(cds,<span class="number">1</span>,<span class="number">2</span>,color=<span class="string">&quot;Biological_Condition&quot;</span>)</span><br><span class="line">table(pData(cds)$Biological_Condition)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plottsne.JPG"/>


<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tableBiologicalananotation.JPG"/>

<p><strong>可以看出，用monocle的方法进行单细胞转录组数据分析，GW的三群细胞依然无法很好的区分开，NPC倒是聚类得非常完美</strong>  </p>
<h2 id="寻找差异基因"><a href="#寻找差异基因" class="headerlink" title="寻找差异基因"></a>寻找差异基因</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#这一步需要计算量，如果数据量大，这一步很耗时</span></span><br><span class="line">Sys.time()</span><br><span class="line">diff_test_res&lt;-differentialGeneTest(cds,fullModelFormulaStr = <span class="string">&quot;~Biological_Condition&quot;</span>)</span><br><span class="line">Sys.time()</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/difftestres.JPG"/>

<p><strong>选出差异基因</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">sig_gene&lt;-subset(diff_test_res,qval&lt;<span class="number">0.1</span>)<span class="comment">#筛选条件qval&lt;0.1</span></span><br><span class="line">head(sig_gene[,<span class="built_in">c</span>(<span class="string">&quot;gene_short_name&quot;</span>,<span class="string">&quot;pval&quot;</span>,<span class="string">&quot;qval&quot;</span>)])</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/headdiffgene.JPG"/>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">choosed_gene&lt;-<span class="built_in">as.character</span>(head(sig_gene$gene_short_name))</span><br><span class="line">plot_genes_jitter(cds[choosed_gene,],grouping = <span class="string">&quot;Biological_Condition&quot;</span>,ncol = <span class="number">2</span>)</span><br><span class="line">plot_genes_jitter(cds[choosed_gene,],grouping = <span class="string">&quot;Biological_Condition&quot;</span>,</span><br><span class="line">                  color_by = <span class="string">&quot;Biological_Condition&quot;</span>,nrow = <span class="number">3</span>,ncol = <span class="literal">NULL</span></span><br><span class="line">                  )</span><br><span class="line"><span class="comment">#自己画图也可以</span></span><br><span class="line"> <span class="comment">#boxplot(log10(exprs(cds)[&#x27;A1BG&#x27;,]+1)~pData(cds)$Biological_Condition)</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotdiffgenehead.JPG" width="49%" height="50%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotdiffgeneheadcol.JPG" width="50%" height="50%">  


<h2 id="推断发育轨迹-这是monocle包最大的亮点"><a href="#推断发育轨迹-这是monocle包最大的亮点" class="headerlink" title="推断发育轨迹 这是monocle包最大的亮点"></a>推断发育轨迹 <font color=red>这是monocle包最大的亮点</font></h2><p><strong>第一步：挑选合适的基因</strong><br>有多种方法，例如可以提供已经的基因集，这里选取统计学上有差异的基因   </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ordering_gene&lt;-row.names(subset(diff_test_res,qval&lt; <span class="number">0.01</span>))</span><br><span class="line">cds&lt;-setOrderingFilter(cds,ordering_gene)</span><br><span class="line">plot_ordering_genes(cds)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotoderinggenes.JPG" width="60%" height="50%">

<p><strong>第二部：降维</strong><br>降维的目的是更好的展示数据。函数里提供很多方法，不同方法最后展示的图会有所不同，DDRTree是Monocle2使用的默认方法</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cds&lt;-reduceDimension(cds,max_components = <span class="number">2</span>,method=<span class="string">&#x27;DDRTree&#x27;</span>)</span><br></pre></td></tr></table></figure>
<p><strong>第三步：对细胞进行排序</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cds&lt;-orderCells(cds)</span><br><span class="line">plot_cell_trajectory(cds,color_by = <span class="string">&quot;Biological_Condition&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/trace.JPG" width="60%" height="50%">


<p><strong>第四步：可视化marker gene随发育阶段变化的表达情况</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#plot_genes_in_pseudotime可以展示marker基因，本例子随便选取了6个差异表达基因</span></span><br><span class="line">plot_genes_in_pseudotime(cds[choosed_gene,],color_by = <span class="string">&quot;Biological_Condition&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tracemaker.JPG"/>


<h2 id="参考资料"><a href="#参考资料" class="headerlink" title="参考资料"></a>参考资料</h2><p>生信技能数（<a target="_blank" rel="noopener" href="https://search.bilibili.com/all?keyword=%E7%94%9F%E4%BF%A1%E6%8A%80%E8%83%BD%E6%A0%91&from_source=nav_search&spm_id_from=333.851.b_696e7465726e6174696f6e616c486561646572.9">B站</a>、公众号）   </p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/" data-id="ckl4ra1m5000nagvf6ksr541k" data-title="scRNA-seq数据分析学习（四）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          scRNA-seq数据分析学习（三）
        
      </div>
    </a>
  
  
    <a href="/2020/02/14/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%BA%8C%EF%BC%89/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">scRNA-seq数据分析学习（二）</div>
    </a>
  
</nav>

  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/System/" rel="tag">System</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/notes/" rel="tag">notes</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Linux/" style="font-size: 16.67px;">Linux</a> <a href="/tags/MySQL/" style="font-size: 13.33px;">MySQL</a> <a href="/tags/NGS/" style="font-size: 20px;">NGS</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/System/" style="font-size: 10px;">System</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/notes/" style="font-size: 10px;">notes</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/01/">January 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">December 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/05/">May 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/04/">April 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/03/">March 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/02/">February 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
          </li>
        
          <li>
            <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
          </li>
        
          <li>
            <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
          </li>
        
          <li>
            <a href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
          </li>
        
          <li>
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGS数据分析之表观</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2021 gongyuqiqi<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>