<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.3.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","images":"/images","scheme":"Pisces","version":"8.2.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}};
  </script>
<meta name="description" content="普通RNA-seq数据分析">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-seq数据分析流程">
<meta property="og:url" content="http://example.com/2020/02/23/RNA-seq%20analysis/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="普通RNA-seq数据分析">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/1qc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/2qc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/3qc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/4qc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/5qc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/featureCounts.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_geneid.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_exprSet.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ggplot2_ang_counts.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/normalization.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GO.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/heatmap.jpeg">
<meta property="article:published_time" content="2020-02-23T08:30:23.000Z">
<meta property="article:modified_time" content="2021-02-14T07:35:03.005Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/1qc.png">


<link rel="canonical" href="http://example.com/2020/02/23/RNA-seq%20analysis/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>
<title>RNA-seq数据分析流程 | gyq's blog</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">gyq's blog</h1>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>







</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#RNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="nav-number">1.</span> <span class="nav-text">RNA-seq数据分析流程</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E4%B8%80%EF%BC%89%E5%8E%9F%E5%A7%8B%E6%95%B0%E6%8D%AE%E7%9A%84%E8%B4%A8%E6%8E%A7"><span class="nav-number">1.1.</span> <span class="nav-text">（一）原始数据的质控</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E4%BA%8C%EF%BC%89%E6%AF%94%E5%AF%B9"><span class="nav-number">1.2.</span> <span class="nav-text">（二）比对</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AC%AC%E4%B8%80%E6%AD%A5%EF%BC%9A%E5%BA%8F%E5%88%97%E6%AF%94%E5%AF%B9"><span class="nav-number">1.2.1.</span> <span class="nav-text">第一步：序列比对</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AC%AC%E4%BA%8C%E6%AD%A5%EF%BC%9A%E6%9E%84%E5%BB%BA%E7%B4%A2%E5%BC%95"><span class="nav-number">1.2.2.</span> <span class="nav-text">第二步：构建索引</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AC%AC%E4%B8%89%E9%83%A8%EF%BC%9A%E7%94%9F%E6%88%90%E7%8A%B6%E6%80%81%E6%96%87%E4%BB%B6-%E7%BB%9F%E8%AE%A1%E6%AF%94%E5%AF%B9%E6%83%85%E5%86%B5"><span class="nav-number">1.2.3.</span> <span class="nav-text">第三部：生成状态文件(统计比对情况)</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E4%B8%89%EF%BC%89%E5%88%B6%E4%BD%9C%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5"><span class="nav-number">1.3.</span> <span class="nav-text">（三）制作表达矩阵</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%EF%BC%88%E5%9B%9B%EF%BC%89%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90"><span class="nav-number">2.</span> <span class="nav-text">（四）差异分析</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AC%AC%E4%B8%80%E6%AD%A5%EF%BC%9A%E5%B0%86reads%E7%9F%A9%E9%98%B5%E5%AF%BC%E5%85%A5R%E4%B8%AD"><span class="nav-number">2.0.1.</span> <span class="nav-text">第一步：将reads矩阵导入R中</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%AC%AC%E4%BA%8C%E6%AD%A5%EF%BC%9A%E8%8E%B7%E5%8F%96%E6%AD%A3%E7%A1%AE%E7%9A%84Geneid"><span class="nav-number">2.0.2.</span> <span class="nav-text">第二步：获取正确的Geneid</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%AC%AC%E4%B8%89%E6%AD%A5%EF%BC%9Aid%E8%BD%AC%E6%8D%A2"><span class="nav-number">2.1.</span> <span class="nav-text">第三步：id转换</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%AC%AC%E5%9B%9B%E6%AD%A5%EF%BC%9ADESeq2%E8%BF%9B%E8%A1%8C%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90"><span class="nav-number">2.2.</span> <span class="nav-text">第四步：DESeq2进行差异分析</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E7%AC%AC%E4%BA%94%E6%AD%A5%EF%BC%9A%E5%8F%AF%E8%A7%86%E5%8C%96-%E2%80%A6%E2%80%A6"><span class="nav-number">2.3.</span> <span class="nav-text">第五步：可视化 ……</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">gongyuqiqi</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives">
          <span class="site-state-item-count">28</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/02/23/RNA-seq%20analysis/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="gongyuqiqi">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="gyq's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNA-seq数据分析流程
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-02-23 16:30:23" itemprop="dateCreated datePublished" datetime="2020-02-23T16:30:23+08:00">2020-02-23</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2021-02-14 15:35:03" itemprop="dateModified" datetime="2021-02-14T15:35:03+08:00">2021-02-14</time>
      </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <p>普通RNA-seq数据分析</p>
<a id="more"></a>

<h1 id="RNA-seq数据分析流程"><a href="#RNA-seq数据分析流程" class="headerlink" title="RNA-seq数据分析流程"></a>RNA-seq数据分析流程</h1><h2 id="（一）原始数据的质控"><a href="#（一）原始数据的质控" class="headerlink" title="（一）原始数据的质控"></a>（一）原始数据的质控</h2><p>以其中一个样本为例（其他样本指控结果基本类似）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/raw-data</span><br><span class="line"><span class="comment">#对每个样本进行质控</span></span><br><span class="line">fastqc -t 7 /root/project/atac/raw/*gz -o ./</span><br><span class="line"><span class="comment">#统计合并所有指控结果</span></span><br><span class="line">multiqc *fastqc.zip --ignore *.html</span><br></pre></td></tr></table></figure>
<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/1qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/2qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/3qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/4qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/5qc.png"/>


<ul>
<li>1、帮忙分析的数据，从公司给的结果的样本名中包含“clean”，我猜想公司大概已经进行过一次过滤，而且也完全没有接头。</li>
<li>2、另外，没有过关的这些数据也很难通过过滤得到改善（因为我使用trim_galore过滤，质控结果更差了）</li>
<li><strong>3、但是我还是不死心呀，我分别取了一个原始的和自己过滤的样本的前10000条reads,进行比对。结果发现原始的比对率可以达到95%+，自己过滤后的样本比对率比较低。</strong></li>
<li>4、所以，这批数据，公司应该已经进行过滤操作了。</li>
</ul>
<h2 id="（二）比对"><a href="#（二）比对" class="headerlink" title="（二）比对"></a>（二）比对</h2><p><strong>比对软件：hisat2</strong></p>
<p><font color=blue>不同数据类型所需要使用的比对软件也是有讲究的呢，可千万不能乱使用，会出问题的，刚开始接触数据分析时，知道有很多比对软件，但一直用bowtie2进行比对，后面就翻车啦～～～～～～</font></p>
<p>比对软件的选择：<br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/849f8ada0ab7">https://www.jianshu.com/p/849f8ada0ab7</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/681e02e7f9af">https://www.jianshu.com/p/681e02e7f9af</a><br><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/26506787">https://zhuanlan.zhihu.com/p/26506787</a> </p>
<p><font color=blue>NGS—-bwa<br>Chip（也包括ATAC）—-bowtie2<br>RNA—-hisat2（推荐）、star、Tophat、subjunc</font></p>
<h3 id="第一步：序列比对"><a href="#第一步：序列比对" class="headerlink" title="第一步：序列比对"></a>第一步：序列比对</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#cat sample</span></span><br><span class="line">KO1455_FRAS190065880-1a_1.clean.fq.gz	KO1455_FRAS190065880-1a_2.clean.fq.gz</span><br><span class="line">KO1462_FRAS190065881-1a_1.clean.fq.gz	KO1462_FRAS190065881-1a_2.clean.fq.gz</span><br><span class="line">KO1471_FRAS190065882-1a_1.clean.fq.gz	KO1471_FRAS190065882-1a_2.clean.fq.gz</span><br><span class="line">WT1441_FRAS190065878-1a_1.clean.fq.gz	WT1441_FRAS190065878-1a_2.clean.fq.gz</span><br><span class="line">WT1465_FRAS190065879-1a_1.clean.fq.gz	WT1465_FRAS190065879-1a_2.clean.fq.gz</span><br><span class="line">WT1474_FRAS190065877-1a_1.clean.fq.gz	WT1474_FRAS190065877-1a_2.clean.fq.gz</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#因为不想让本机太辛苦，要让它歇息一会儿,所以将KO和WT分批次跑</span></span><br><span class="line">cat sample | head -n 3 &gt; KO_sample</span><br><span class="line">cat sample | tail -n 3 &gt; WT_sample</span><br><span class="line">ref=/home/reference/mm10/genome</span><br><span class="line"></span><br><span class="line">cat KO_sample|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">hisat2 -p 7 -x <span class="variable">$ref</span> -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span> | samtools sort -@ 7 -o /home/align/<span class="variable">$sample</span> -</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">cat WT_sample|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">hisat2 -p 7 -x <span class="variable">$ref</span> -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span> | samtools sort -@ 7 -o /home/align/<span class="variable">$sample</span> -</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#各样本比对率都在95%以上，这样的比对结果才可以比较有信心的进行后续分析了。</span></span><br></pre></td></tr></table></figure>
<h3 id="第二步：构建索引"><a href="#第二步：构建索引" class="headerlink" title="第二步：构建索引"></a>第二步：构建索引</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|xargs -i samtools index &#123;&#125;</span><br></pre></td></tr></table></figure>
<h3 id="第三部：生成状态文件-统计比对情况"><a href="#第三部：生成状态文件-统计比对情况" class="headerlink" title="第三部：生成状态文件(统计比对情况)"></a>第三部：生成状态文件(统计比对情况)</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|xargs -i samtools flagstat -@ 7 &#123;&#125;</span><br></pre></td></tr></table></figure>
<h2 id="（三）制作表达矩阵"><a href="#（三）制作表达矩阵" class="headerlink" title="（三）制作表达矩阵"></a>（三）制作表达矩阵</h2><p>参考资料：<a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/407/">https://www.bioinfo-scrounger.com/archives/407/</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/reference/gtf/gencode.v33.annotation.gtf  </span><br><span class="line">featureCounts -T 8 -p -t exon -g gene_id -a <span class="variable">$gtf</span> -o RNA-matrix.txt *.bam 1&gt;counts.id.log 2&gt;&amp;1</span><br></pre></td></tr></table></figure>
<p>查看featureCounts的情况</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat counts.id.log</span><br></pre></td></tr></table></figure>

<p>在此，要记录一个错误。第一次操作时误使用了人类的gtf文件，结果日志文件显示比对率在10%左右……使用小鼠gtf文件后，比对结果正常多了（因为这批数据是小鼠组织细胞测序的结果）。如下所示。<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/featureCounts.png"/></p>
<h1 id="（四）差异分析"><a href="#（四）差异分析" class="headerlink" title="（四）差异分析"></a>（四）差异分析</h1><h3 id="第一步：将reads矩阵导入R中"><a href="#第一步：将reads矩阵导入R中" class="headerlink" title="第一步：将reads矩阵导入R中"></a>第一步：将reads矩阵导入R中</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">rm(<span class="built_in">list</span> = ls())</span><br><span class="line">options(stringsAsFactors = <span class="built_in">F</span>)</span><br><span class="line">setwd(<span class="string">&quot;G:\\seq-analysis-data\\RNA-seq&quot;</span>)</span><br><span class="line">a=read.table(<span class="string">&#x27;RNA-matrix.txt&#x27;</span>,header = <span class="built_in">T</span>)</span><br><span class="line">ckeck_a=a[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">12</span>]</span><br></pre></td></tr></table></figure>
<h3 id="第二步：获取正确的Geneid"><a href="#第二步：获取正确的Geneid" class="headerlink" title="第二步：获取正确的Geneid"></a>第二步：获取正确的Geneid</h3><ul>
<li>首先看看矩阵中的Geneid长什么样</li>
</ul>
<img width="20%" height="20%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_geneid.JPG"/>


<ul>
<li>所以要去掉小数点及后面的数字，获取正确的Geneid<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(stringr)</span><br><span class="line"><span class="built_in">class</span>(str_split(a$Geneid,<span class="string">&#x27;[.]&#x27;</span>,simplify = <span class="built_in">T</span>))</span><br><span class="line">a$ensembl_id=str_split(a$Geneid,<span class="string">&#x27;[.]&#x27;</span>,simplify = <span class="built_in">T</span>)[,<span class="number">1</span>]</span><br><span class="line">rownames(a)&lt;-a$ensembl_id</span><br><span class="line">check_a_again=a[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">12</span>]</span><br><span class="line">exprSet&lt;-a[,<span class="number">7</span>:<span class="number">12</span>]</span><br><span class="line">ckeck_exprSet&lt;-head(exprSet)</span><br></pre></td></tr></table></figure>
表达矩阵现在的样子（Geneid名去掉了多余的部分）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_exprSet.JPG"/>


<h2 id="第三步：id转换"><a href="#第三步：id转换" class="headerlink" title="第三步：id转换"></a>第三步：id转换</h2><p>Geneid和symbol并不是完全一一对应的，存在部分多个Geneid对应同一个基因名的情况。</p>
<p>因为rownames必须唯一，所以就要去掉重复的Geneid或者只保留其中一个（这个保留就比较的人为了~）</p>
<ul>
<li><p>方法一：使用biomaRt包</p>
<p>此方法几乎可以转换全部的id(<font color=orange>推荐</font>)</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">library(biomaRt)</span><br><span class="line">library(curl)</span><br><span class="line"><span class="comment">#此步骤需要网速，网速不好会报错（curl相关的），多试几遍，会有好运气的</span></span><br><span class="line">mart &lt;- useDataset(<span class="string">&quot;mmusculus_gene_ensembl&quot;</span>, useMart(<span class="string">&quot;ensembl&quot;</span>))<span class="comment">##人类选择hsapiens_gene_ensembl</span></span><br><span class="line">gene&lt;-row.names(exprSet)<span class="comment">#要转换成SYMBOL的Geneid</span></span><br><span class="line">gene_name&lt;-getBM(<span class="built_in">attributes</span>=<span class="built_in">c</span>(<span class="string">&quot;description&quot;</span>,<span class="string">&quot;external_gene_name&quot;</span>,<span class="string">&quot;ensembl_gene_id&quot;</span>),</span><br><span class="line">                 filters = <span class="string">&quot;ensembl_gene_id&quot;</span>,</span><br><span class="line">                 values = gene, mart = mart)</span><br><span class="line">head(gene_name)</span><br><span class="line">exprSet&lt;-a[,<span class="number">7</span>:<span class="number">12</span>]</span><br><span class="line">head(exprSet)</span><br><span class="line">table(rownames(exprSet)%in%gene_name$ensembl_gene_id)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%gene_name$ensembl_gene_id,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">symbol&lt;-gene_name[match(rownames(exprSet),gene_name$ensembl_gene_id),]</span><br><span class="line">head(symbol)</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">6</span>,<span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">tmp&lt;-by(exprSet,symbol$external_gene_name,<span class="keyword">function</span>(x)rownames(x)[which.max(rowMeans(x))])</span><br><span class="line">probes&lt;-<span class="built_in">as.character</span>(tmp)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%probes,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">rownames(exprSet)&lt;-symbol[match(rownames(exprSet),symbol$ensembl_gene_id),<span class="number">2</span>]</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line">write.csv(exprSet,<span class="string">&#x27;raw_exprSet.csv&#x27;</span>)</span><br><span class="line"><span class="comment">#exprSet&lt;-read.csv(&#x27;raw_exprSet.csv&#x27;)</span></span><br><span class="line"><span class="comment">#rownames(exprSet)&lt;-exprSet$X</span></span><br><span class="line"><span class="comment">#exprSet&lt;-exprSet[,-1]</span></span><br></pre></td></tr></table></figure></li>
<li><p>方法二：使用org.Mm.eg.db包<br>可能因为此包不怎么更新，id转换率只有50%左右，这样回丢掉很多信息的，可能会导致后续分析不可靠（不推荐）</p>
</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">library(clusterProfiler)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">symbol&lt;-bitr(rownames(exprSet),fromType = <span class="string">&quot;ENSEMBL&quot;</span>,</span><br><span class="line">             toType=<span class="built_in">c</span>(<span class="string">&quot;SYMBOL&quot;</span>,<span class="string">&quot;ENTREZID&quot;</span>),OrgDb = org.Mm.eg.db)</span><br><span class="line">colnames(symbol)&lt;-<span class="built_in">c</span>(<span class="string">&quot;ensembl_id&quot;</span>,<span class="string">&quot;SYMBOL&quot;</span>,<span class="string">&quot;ENTREZID&quot;</span>)</span><br><span class="line">check_symbol&lt;-symbol[<span class="number">1</span>:<span class="number">10</span>,<span class="number">1</span>:<span class="number">3</span>]</span><br><span class="line">table(rownames(exprSet)%in%symbol$ensembl_id)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%symbol$ensembl_id,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">symbol&lt;-symbol[match(rownames(exprSet),symbol$ensembl_id),]</span><br><span class="line">head(symbol)</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">6</span>,<span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">tmp&lt;-by(exprSet,symbol$SYMBOL,<span class="keyword">function</span>(x)rownames(x)[which.max(rowMeans(x))])</span><br><span class="line">probes&lt;-<span class="built_in">as.character</span>(tmp)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%probes,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">rownames(exprSet)&lt;-symbol[match(rownames(exprSet),symbol$ensembl_id),<span class="number">2</span>]</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br></pre></td></tr></table></figure>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/3a0e1e3e41d0">https://www.jianshu.com/p/3a0e1e3e41d0</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c06fea33b60f">https://www.jianshu.com/p/c06fea33b60f</a></p>
<ul>
<li>在进行后续分析之前，检验目的基因的敲除效果<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Ang表达量探索</span></span><br><span class="line">Ang&lt;-raw_exprSet[rownames(raw_exprSet)==<span class="string">&quot;Ang&quot;</span>,]</span><br><span class="line">colnames(Ang)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">ang&lt;-as.data.frame(t(Ang))</span><br><span class="line">ang$Group&lt;-<span class="built_in">c</span>(<span class="built_in">rep</span>(<span class="string">&quot;KO&quot;</span>,<span class="number">3</span>),<span class="built_in">rep</span>(<span class="string">&quot;WT&quot;</span>,<span class="number">3</span>))</span><br><span class="line">ang$Sample&lt;-rownames(ang)</span><br><span class="line">colnames(ang)&lt;-<span class="built_in">c</span>(<span class="string">&quot;Counts&quot;</span>,<span class="string">&quot;Group&quot;</span>,<span class="string">&quot;Sample&quot;</span>)</span><br><span class="line">library(ggplot2)</span><br><span class="line">p&lt;-ggplot(ang,aes(Sample,Counts,group=Group))+</span><br><span class="line">  geom_bar(aes(fill=Group),stat=<span class="string">&#x27;identity&#x27;</span>)+</span><br><span class="line">  scale_fill_manual(values = <span class="built_in">c</span>(<span class="string">&quot;#FFA500&quot;</span>,<span class="string">&quot;#228B22&quot;</span>))+</span><br><span class="line">  ylab(<span class="string">&quot;The expression of Ang&quot;</span>)+</span><br><span class="line">  xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line">p+theme_bw()+theme(panel.border = element_blank(),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),axis.line = element_line(colour = <span class="string">&quot;black&quot;</span>))</span><br></pre></td></tr></table></figure>
由图可看出：目的基因敲除效果显著；另外，也可以一定程度上反应上游数据分析的可靠性（因为在做测序之前，一定是先实验验证过敲低效率的）。</li>
</ul>
<img width="50%" height="50%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ggplot2_ang_counts.jpeg"/>


<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/92473504">https://zhuanlan.zhihu.com/p/92473504</a></p>
<p><a target="_blank" rel="noopener" href="https://www.sohu.com/a/319083452_785442">https://www.sohu.com/a/319083452_785442</a></p>
<h2 id="第四步：DESeq2进行差异分析"><a href="#第四步：DESeq2进行差异分析" class="headerlink" title="第四步：DESeq2进行差异分析"></a>第四步：DESeq2进行差异分析</h2><p>1、DESeq2差异分析的主要步骤</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">library(DESeq2)</span><br><span class="line"><span class="comment">#DEseq2要求输入数据是由整数组成的矩阵。</span></span><br><span class="line"><span class="comment">#DESeq2要求矩阵是没有标准化的。</span></span><br><span class="line"><span class="comment">##构建一个dds对象</span></span><br><span class="line">exprSet_integer=<span class="built_in">round</span>(exprSet)  </span><br><span class="line">(colData &lt;- data.frame(row.names=colnames(exprSet_integer), </span><br><span class="line">                       group_list=group_list) )</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData = exprSet_integer,</span><br><span class="line">                              colData = colData,</span><br><span class="line">                              design = ~ group_list)</span><br><span class="line">dds2 &lt;- DESeq(dds)<span class="comment">##使用DESeq函数对dds进行normalize</span></span><br><span class="line">resultsNames(dds2)<span class="comment">##查看结果的名称 </span></span><br><span class="line"></span><br><span class="line">res &lt;- results(dds2, contrast=<span class="built_in">c</span>(<span class="string">&quot;group_list&quot;</span>,<span class="string">&quot;KO&quot;</span>,<span class="string">&quot;WT&quot;</span>))</span><br><span class="line"><span class="comment">##提取想要的差异分析结果，这里是&quot;KO&quot;组对&quot;WT&quot;组进行比较。</span></span><br><span class="line">resOrdered &lt;- res[order(res$padj),]<span class="comment">##按padj从小到大排序</span></span><br><span class="line">summary(res)</span><br><span class="line">DEG&lt;-data.frame(resOrdered)<span class="comment">##得到差异分析的data.frame</span></span><br><span class="line">head(DEG)</span><br><span class="line">DEG = na.omit(DEG)</span><br><span class="line">write.csv(DEG,<span class="string">&#x27;DESeq2_KO_VS_WT DEG.csv&#x27;</span>)</span><br><span class="line"><span class="comment">#DEG&lt;-read.csv(&quot;DESeq2_KO_VS_WT DEG.csv&quot;)</span></span><br><span class="line"><span class="comment">#rownames(DEG)&lt;-DEG$X</span></span><br><span class="line"><span class="comment">#DEG&lt;-DEG[,-1]</span></span><br></pre></td></tr></table></figure>
<p>2、DESeq2标准化前后的对比</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#数据离散度非常大的RNA-seq的基因的reads的counts矩阵经过normalization后</span></span><br><span class="line"><span class="comment">#变成了类似于芯片表达矩阵的表达矩阵。生成的2行2列的图可以看出normalization前后的变化</span></span><br><span class="line">rld&lt;-rlogTransformation(dds2)<span class="comment">##得到经过DESeq2软件normalization的表达矩阵。</span></span><br><span class="line">exprSet_normalized&lt;-assay(rld)</span><br><span class="line">par(cex = <span class="number">0.7</span>)</span><br><span class="line">n.sample=ncol(exprSet)</span><br><span class="line"><span class="keyword">if</span>(n.sample&gt;<span class="number">40</span>) par(cex = <span class="number">0.5</span>)</span><br><span class="line">cols &lt;- rainbow(n.sample*<span class="number">1.2</span>)</span><br><span class="line">par(mfrow=<span class="built_in">c</span>(<span class="number">2</span>,<span class="number">2</span>))</span><br><span class="line">colnames(exprSet_integer)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">colnames(exprSet_normalized)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">boxplot(exprSet_integer, col = cols,main=<span class="string">&quot;expression value&quot;</span>,las=<span class="number">2</span>)</span><br><span class="line">boxplot(exprSet_normalized, col = cols,main=<span class="string">&quot;expression value&quot;</span>,las=<span class="number">2</span>)</span><br><span class="line">hist(as.matrix(exprSet))</span><br><span class="line">hist(exprSet_normalized)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/normalization.jpeg"/>

<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/30350531">https://zhuanlan.zhihu.com/p/30350531</a></p>
<h2 id="第五步：可视化-……"><a href="#第五步：可视化-……" class="headerlink" title="第五步：可视化 ……"></a>第五步：可视化 ……</h2><ul>
<li>富集分析（可以使用KEGG、GO，这里使用GO）<br>大致看一下受目的基因影响的信号通路<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#1、差异基因的提取</span></span><br><span class="line">topT &lt;- as.data.frame(DEG)</span><br><span class="line">logFC_Cutoff&lt;-with(topT,mean(<span class="built_in">abs</span>(log2FoldChange))+<span class="number">2</span>*sd(<span class="built_in">abs</span>(log2FoldChange)))</span><br><span class="line">logFC_cutoff&lt;-logFC_Cutoff</span><br><span class="line">select_gene=(<span class="built_in">abs</span>(DEG$log2FoldChange)&gt;=logFC_cutoff)&amp;(DEG$pvalue&lt;<span class="number">0.05</span>)</span><br><span class="line">select_gene.sig=DEG[select_gene,]</span><br><span class="line">write.csv(select_gene.sig,<span class="string">&#x27;DESeq2_select_gene.sig.csv&#x27;</span>)</span><br><span class="line">gene=rownames(select_gene.sig)</span><br><span class="line"><span class="comment">#2、差异基因的通路富集</span></span><br><span class="line">library(clusterProfiler)</span><br><span class="line">library(topGO)</span><br><span class="line">library(Rgraphviz)</span><br><span class="line">library(pathview)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">DEG.entrez_id = mapIds(x = org.Mm.eg.db,</span><br><span class="line">                       keys = gene,</span><br><span class="line">                       keytype = <span class="string">&quot;SYMBOL&quot;</span>,</span><br><span class="line">                       column = <span class="string">&quot;ENTREZID&quot;</span>)</span><br><span class="line">DEG.entrez_id = na.omit(DEG.entrez_id)</span><br><span class="line">erich.go.BP = enrichGO(gene = DEG.entrez_id,</span><br><span class="line">                       OrgDb = org.Mm.eg.db,</span><br><span class="line">                       keyType = <span class="string">&quot;ENTREZID&quot;</span>,</span><br><span class="line">                       ont = <span class="string">&quot;BP&quot;</span>,</span><br><span class="line">                       pvalueCutoff = <span class="number">0.5</span>,</span><br><span class="line">                       qvalueCutoff = <span class="number">0.5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">##分析完成后，作图</span></span><br><span class="line">dotplot(erich.go.BP)</span><br></pre></td></tr></table></figure>
<img width="80%" height="80%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GO.jpeg"/>

</li>
</ul>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/47b5ea646932?utm_source=desktop&amp;utm_medium=timeline">https://www.jianshu.com/p/47b5ea646932?utm_source=desktop&amp;utm_medium=timeline</a></p>
<ul>
<li>热图绘制<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">library(pheatmap)</span><br><span class="line">colnames(exprSet)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">choose_upgene=<span class="built_in">names</span>(sort(apply(up_gene, <span class="number">1</span>, mad),decreasing = <span class="built_in">T</span>)[<span class="number">1</span>:<span class="number">40</span>])</span><br><span class="line">choose_upgene_matrix=exprSet[choose_upgene,]</span><br><span class="line">choose_downgene=<span class="built_in">names</span>(sort(apply(down_gene, <span class="number">1</span>, mad),decreasing = <span class="built_in">T</span>)[<span class="number">1</span>:<span class="number">40</span>])</span><br><span class="line">choose_downgene_matrix=exprSet[choose_downgene,]</span><br><span class="line">choose_matrix=rbind(choose_upgene_matrix,choose_downgene_matrix)</span><br><span class="line">choose_matrix=t(scale(t(choose_matrix)))</span><br><span class="line">annotation_col = data.frame(CellType = factor(<span class="built_in">rep</span>(<span class="built_in">c</span>(<span class="string">&quot;KO&quot;</span>, <span class="string">&quot;WT&quot;</span>), each=<span class="number">3</span>)),Time = <span class="number">1</span>:<span class="number">3</span>)</span><br><span class="line">rownames(annotation_col)&lt;-colnames(exprSet)</span><br><span class="line">pheatmap(choose_matrix,fontsize = <span class="number">6</span>,cexCol=<span class="number">1</span>,cellwidth=<span class="number">10</span>,cellheigh=<span class="number">5</span>,color = colorRampPalette(<span class="built_in">c</span>(<span class="string">&#x27;navy&#x27;</span>,<span class="string">&#x27;white&#x27;</span>,<span class="string">&#x27;firebrick3&#x27;</span>))(<span class="number">50</span>),annotation_col=annotation_col)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/heatmap.jpeg"/>

</li>
</ul>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/NGS/" rel="tag"># NGS</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/02/22/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%89%EF%BC%89/" rel="prev" title="scRNA-seq数据分析实战（三）">
                  <i class="fa fa-chevron-left"></i> scRNA-seq数据分析实战（三）
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2020/02/23/R%EF%BF%BD%00apply%01lapply%01strsplit%01%25in%25(%EF%BF%BD%EF%BF%BD%EF%BF%BD/" rel="next" title="R语言apply、lapply、strsplit、&#37in&#37用法介绍">
                  R语言apply、lapply、strsplit、&#37in&#37用法介绍 <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>







<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>
</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">gongyuqiqi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a>
  </div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  






  





</body>
</html>
