<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.3.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/avatar.jpg">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/avatar.jpg">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/avatar.jpg">
  <link rel="mask-icon" href="/images/avatar.jpg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"gyqgyq.com","root":"/","images":"/images","scheme":"Pisces","version":"8.2.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":true,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":true,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"/search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>
<meta name="description" content="单细胞转录组数据降维聚类分群图表复现初尝试1. 这次我们要复现的单细胞数据来自International Journal of Cancer 2020年发表的一篇单细胞相关的纯生信文章。该文章数据链接：GSE150321   2. 简单浏览文章，抓取作者针对单细胞数据分析给出的一些参数信息，如下    Step 0、加载需要的R包12345library(Seurat)library(dply">
<meta property="og:type" content="article">
<meta property="og:title" content="scRNA-seq_数据复现">
<meta property="og:url" content="https://gyqgyq.com/2021/05/13/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="单细胞转录组数据降维聚类分群图表复现初尝试1. 这次我们要复现的单细胞数据来自International Journal of Cancer 2020年发表的一篇单细胞相关的纯生信文章。该文章数据链接：GSE150321   2. 简单浏览文章，抓取作者针对单细胞数据分析给出的一些参数信息，如下    Step 0、加载需要的R包12345library(Seurat)library(dply">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_analysis_information.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sparse_matrix.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pre_filter.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/aft_filter.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/top_10_highly_variable_genes.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/UMAP_cluster5_new.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/marker_genes_for_total.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plot_marker_on_UMAP_new.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/KRT5.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PTPRC.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CLU.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/AQP1.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/COL3A1.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/COL1A2.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/dotplot_marker.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/subset_account_paper.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pie.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/UMAP_cluster5_lable_new.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_total_cluster.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Keratinocyte_cancer_cluster.png">
<meta property="article:published_time" content="2021-05-13T06:08:18.000Z">
<meta property="article:modified_time" content="2021-05-15T15:28:11.544Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="tag1">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper.JPG">


<link rel="canonical" href="https://gyqgyq.com/2021/05/13/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>
<title>scRNA-seq_数据复现 | gyq's blog</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">gyq's blog</h1>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li>
        <li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li>
        <li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8D%95%E7%BB%86%E8%83%9E%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E9%99%8D%E7%BB%B4%E8%81%9A%E7%B1%BB%E5%88%86%E7%BE%A4%E5%9B%BE%E8%A1%A8%E5%A4%8D%E7%8E%B0%E5%88%9D%E5%B0%9D%E8%AF%95"><span class="nav-number">1.</span> <span class="nav-text">单细胞转录组数据降维聚类分群图表复现初尝试</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-0%E3%80%81%E5%8A%A0%E8%BD%BD%E9%9C%80%E8%A6%81%E7%9A%84R%E5%8C%85"><span class="nav-number">2.</span> <span class="nav-text">Step 0、加载需要的R包</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-1%E3%80%81%E6%95%B0%E6%8D%AE%E5%87%86%E5%A4%87"><span class="nav-number">3.</span> <span class="nav-text">Step 1、数据准备</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-2%E3%80%81%E5%88%9B%E5%BB%BASeurat%E5%AF%B9%E8%B1%A1"><span class="nav-number">4.</span> <span class="nav-text">Step 2、创建Seurat对象</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-3%E3%80%81%E6%A0%87%E5%87%86%E5%8C%96"><span class="nav-number">5.</span> <span class="nav-text">Step 3、标准化</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-4%E3%80%81%E9%89%B4%E5%AE%9A%E9%AB%98%E5%8F%98%E5%9F%BA%E5%9B%A0"><span class="nav-number">6.</span> <span class="nav-text">Step 4、鉴定高变基因</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-5%E3%80%81%E9%99%8D%E7%BB%B4"><span class="nav-number">7.</span> <span class="nav-text">Step 5、降维</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-6%E3%80%81%E7%BB%86%E8%83%9E%E8%81%9A%E7%B1%BB"><span class="nav-number">8.</span> <span class="nav-text">Step 6、细胞聚类</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-7%E3%80%81%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE"><span class="nav-number">9.</span> <span class="nav-text">Step 7、差异表达</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-8%E3%80%81%E7%BB%86%E8%83%9E%E4%BA%9A%E7%BE%A4%E5%91%BD%E5%90%8D"><span class="nav-number">10.</span> <span class="nav-text">Step 8、细胞亚群命名</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Step-9%E3%80%81%E7%BB%86%E8%83%9E%E4%BA%9A%E7%BE%A4%E6%8F%90%E5%8F%96%E5%8F%8A%E5%86%8D%E5%88%86%E7%BE%A4"><span class="nav-number">11.</span> <span class="nav-text">Step 9、细胞亚群提取及再分群</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%80%BB%E7%BB%93"><span class="nav-number">11.1.</span> <span class="nav-text">总结</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="gongyuqiqi"
      src="/assets/img/ava.jpg">
  <p class="site-author-name" itemprop="name">gongyuqiqi</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">35</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">7</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author site-overview-item animated">
      <span class="links-of-author-item">
        <a href="mailto:gongyuqi163@163.com" title="E-Mail → mailto:gongyuqi163@163.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
  </div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

  <a href="https://github.com/gongyuqiqi" class="github-corner" title="Follow me on GitHub" aria-label="Follow me on GitHub" rel="noopener" target="_blank"><svg width="80" height="80" viewBox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="https://gyqgyq.com/2021/05/13/scRNA-seq-%E6%95%B0%E6%8D%AE%E5%A4%8D%E7%8E%B0/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/assets/img/ava.jpg">
      <meta itemprop="name" content="gongyuqiqi">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="gyq's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          scRNA-seq_数据复现
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2021-05-13 14:08:18" itemprop="dateCreated datePublished" datetime="2021-05-13T14:08:18+08:00">2021-05-13</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2021-05-15 23:28:11" itemprop="dateModified" datetime="2021-05-15T23:28:11+08:00">2021-05-15</time>
      </span>

  
      </div>
      <div class="post-meta">
    <span class="post-meta-item" title="Symbols count in article">
      <span class="post-meta-item-icon">
        <i class="far fa-file-word"></i>
      </span>
      <span class="post-meta-item-text">Symbols count in article: </span>
      <span>8.6k</span>
    </span>
    <span class="post-meta-item" title="Reading time">
      <span class="post-meta-item-icon">
        <i class="far fa-clock"></i>
      </span>
      <span class="post-meta-item-text">Reading time &asymp;</span>
      <span>8 mins.</span>
    </span>
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <a id="more"></a>

<h1 id="单细胞转录组数据降维聚类分群图表复现初尝试"><a href="#单细胞转录组数据降维聚类分群图表复现初尝试" class="headerlink" title="单细胞转录组数据降维聚类分群图表复现初尝试"></a>单细胞转录组数据降维聚类分群图表复现初尝试</h1><p><strong>1. 这次我们要复现的单细胞数据来自International Journal of Cancer 2020年发表的一篇单细胞相关的纯生信文章。该文章数据链接：<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150321">GSE150321</a></strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper.JPG" width="50%">

<p><strong>2. 简单浏览文章，抓取作者针对单细胞数据分析给出的一些参数信息，如下</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_analysis_information.JPG" width="50%">


<h1 id="Step-0、加载需要的R包"><a href="#Step-0、加载需要的R包" class="headerlink" title="Step 0、加载需要的R包"></a>Step 0、加载需要的R包</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">library(Seurat)</span><br><span class="line">library(dplyr)</span><br><span class="line">library(patchwork)</span><br><span class="line">library(mindr)</span><br><span class="line">library(Matrix)</span><br></pre></td></tr></table></figure>
<h1 id="Step-1、数据准备"><a href="#Step-1、数据准备" class="headerlink" title="Step 1、数据准备"></a>Step 1、数据准备</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">&quot;//home//data//gongyuqi//R&quot;</span>)</span><br><span class="line"><span class="comment">#文章测序了两个病人的样本，正文figure展示的是其中一个病人的单细胞测序结果。我们下面的分析使用正文中展示的样本</span></span><br><span class="line"><span class="comment">#读入数据并将表达矩阵转成稀疏矩阵，减少数据对空间资源的消耗以及运算资源的消耗</span></span><br><span class="line">GSM4546857&lt;-read.csv(<span class="string">&quot;GSM4546857_LSCC01_DBEC_UMI.csv&quot;</span>,comment.char = <span class="string">&quot;#&quot;</span>)</span><br><span class="line"><span class="built_in">dim</span>(GSM4546857)</span><br><span class="line">GSM4546857[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">rownames(GSM4546857)&lt;-GSM4546857$Cell_Index</span><br><span class="line">GSM4546857&lt;-GSM4546857[,-<span class="number">1</span>]</span><br><span class="line">GSM4546857&lt;-t(GSM4546857)</span><br><span class="line">object.size(GSM4546857)<span class="comment">#2331231144 bytes</span></span><br><span class="line">GSM4546857_sparse&lt;-as(as.matrix(GSM4546857),<span class="string">&quot;dgCMatrix&quot;</span>)</span><br><span class="line">GSM4546857_sparse[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">object.size(GSM4546857_sparse)<span class="comment">#166367952 bytes</span></span><br><span class="line">save(GSM4546857_sparse,file = <span class="string">&quot;GSM4546857_sparse.Rdata&quot;</span>)</span><br><span class="line"><span class="built_in">dim</span>(GSM4546857_sparse)</span><br></pre></td></tr></table></figure>
<p><strong>在此，可以直观的感受一下两种矩阵的大小</strong><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sparse_matrix.JPG"/></p>
<h1 id="Step-2、创建Seurat对象"><a href="#Step-2、创建Seurat对象" class="headerlink" title="Step 2、创建Seurat对象"></a>Step 2、创建Seurat对象</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#下面这段代码中，最重要的就是创建Seurat对象以及去除线粒体基因，其他都是对Seurat对象的可视化，其目的在于提高初学者对该对象的了解</span></span><br><span class="line"><span class="comment">## =============== 创建Seurat对象</span></span><br><span class="line">tissu1 &lt;- CreateSeuratObject(counts = GSM4546857_sparse, project = <span class="string">&quot;LSCC&quot;</span>, min.cells = <span class="number">3</span>, min.features = <span class="number">200</span>)</span><br><span class="line">tissu1</span><br><span class="line"></span><br><span class="line"><span class="comment">## =============== 去除线粒体基因</span></span><br><span class="line"><span class="comment"># The [[ operator can add columns to object metadata. This is a great place to stash QC stats</span></span><br><span class="line"><span class="comment">#此次数据检测到大量线粒体基因</span></span><br><span class="line">grep(pattern = <span class="string">&quot;^MT\\.&quot;</span>,rownames(tissu1),value = <span class="built_in">T</span>)</span><br><span class="line">tissu1[[<span class="string">&quot;percent.mt&quot;</span>]] &lt;- PercentageFeatureSet(tissu1, pattern = <span class="string">&quot;^MT\\.&quot;</span>)</span><br><span class="line">head(tissu1@meta.data,<span class="number">5</span>)</span><br><span class="line">summary(tissu1@meta.data$nCount_RNA)</span><br></pre></td></tr></table></figure>
<p><strong>看一看过滤前数据情况</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Visualize QC metrics as a violin plot</span></span><br><span class="line">VlnPlot(tissu1, features = <span class="built_in">c</span>(<span class="string">&quot;nFeature_RNA&quot;</span>, <span class="string">&quot;nCount_RNA&quot;</span>, <span class="string">&quot;percent.mt&quot;</span>), ncol = <span class="number">3</span>)</span><br><span class="line"><span class="comment"># density</span></span><br><span class="line">data&lt;-tissu1@meta.data</span><br><span class="line">library(ggplot2)</span><br><span class="line">p&lt;-ggplot(data = data,aes(x=nFeature_RNA))+geom_density()</span><br><span class="line">p</span><br><span class="line"></span><br><span class="line"><span class="comment"># FeatureScatter is typically used to visualize feature-feature relationships, but can be used</span></span><br><span class="line"><span class="comment"># for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</span></span><br><span class="line">plot1 &lt;- FeatureScatter(tissu1, feature1 = <span class="string">&quot;nCount_RNA&quot;</span>, feature2 = <span class="string">&quot;percent.mt&quot;</span>)</span><br><span class="line">plot2 &lt;- FeatureScatter(tissu1, feature1 = <span class="string">&quot;nCount_RNA&quot;</span>, feature2 = <span class="string">&quot;nFeature_RNA&quot;</span>)</span><br><span class="line">plot1 + plot2</span><br></pre></td></tr></table></figure>
<p>这里我们主要展示数据<strong>过滤前</strong>nFeature_RNA, nCount_RNA, percent.mt的情况</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pre_filter.png" width="50%">


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Seurat官网给出的质控标准中，percent.mt &lt; 5，但不同组织不同细胞中线粒体含量存在差异,要根据自己的具体情况进行调整，</span></span><br><span class="line"><span class="comment">#但是！！！太夸张啦！太夸张啦！我们这里线粒体基因阈值设置在80%，最终得到的细胞数为9796个，如果设置成95%，得到的细胞数也小于原文中的10699。太夸张啦！为了复现文章，我们先定成80。</span></span><br><span class="line">tissu1 &lt;- subset(tissu1, subset = nFeature_RNA &gt; <span class="number">200</span> &amp; nFeature_RNA &lt; <span class="number">2500</span> &amp; percent.mt &lt; <span class="number">80</span>)</span><br><span class="line">tissu1</span><br><span class="line">VlnPlot(tissu1, features = <span class="built_in">c</span>(<span class="string">&quot;nFeature_RNA&quot;</span>, <span class="string">&quot;nCount_RNA&quot;</span>, <span class="string">&quot;percent.mt&quot;</span>), ncol = <span class="number">3</span>)</span><br></pre></td></tr></table></figure>
<p>这里我们主要展示数据<strong>过滤后</strong>nFeature_RNA, nCount_RNA, percent.mt的情况<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/aft_filter.png" width="50%"></p>
<h1 id="Step-3、标准化"><a href="#Step-3、标准化" class="headerlink" title="Step 3、标准化"></a>Step 3、标准化</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== Normalization</span></span><br><span class="line">tissu1 &lt;- NormalizeData(tissu1, normalization.method = <span class="string">&quot;LogNormalize&quot;</span>, scale.factor = <span class="number">10000</span>)</span><br><span class="line"><span class="comment">#标准化后的值保存在tissu1[[&quot;RNA&quot;]]@data</span></span><br><span class="line">normalized.data&lt;-tissu1[[<span class="string">&quot;RNA&quot;</span>]]@data</span><br><span class="line">normalized.data[<span class="number">1</span>:<span class="number">10</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br></pre></td></tr></table></figure>
<h1 id="Step-4、鉴定高变基因"><a href="#Step-4、鉴定高变基因" class="headerlink" title="Step 4、鉴定高变基因"></a>Step 4、鉴定高变基因</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== Identification of highly variable features</span></span><br><span class="line"><span class="comment">#Seurat官方默认找前2000高变基因，用于下游的PCA降维分析</span></span><br><span class="line"><span class="comment">#文章中作者是找的前2500，我们这里也就找前2500个高变基因</span></span><br><span class="line"><span class="comment">#高变基因：在一些细胞中高表达，在另一些细胞中低表达</span></span><br><span class="line"><span class="comment">#变异指标：mean-variance relationship</span></span><br><span class="line">tissu1 &lt;- FindVariableFeatures(tissu1, selection.method = <span class="string">&quot;vst&quot;</span>, nfeatures = <span class="number">2500</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Identify the 10 most highly variable genes</span></span><br><span class="line">top10 &lt;- head(VariableFeatures(tissu1), <span class="number">10</span>)</span><br><span class="line">top10</span><br><span class="line"></span><br><span class="line"><span class="comment"># plot variable features with and without labels</span></span><br><span class="line">plot1 &lt;- VariableFeaturePlot(tissu1)</span><br><span class="line">plot2 &lt;- LabelPoints(plot = plot1, points = top10, repel = <span class="literal">TRUE</span>)</span><br><span class="line">plot1 + plot2</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/top_10_highly_variable_genes.png"/>



<h1 id="Step-5、降维"><a href="#Step-5、降维" class="headerlink" title="Step 5、降维"></a>Step 5、降维</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== 1. Scaling the data</span></span><br><span class="line"><span class="comment">#Seurat默认是做前2000个基因的归一化</span></span><br><span class="line"><span class="comment">#如何有热图展示需求的小伙伴，可以做全部基因的归一化，方便后续分析，我们这里就全部基因进行归一化</span></span><br><span class="line">all.genes &lt;- rownames(tissu1)</span><br><span class="line">tissu1 &lt;- ScaleData(tissu1, features = all.genes)</span><br><span class="line"></span><br><span class="line"><span class="comment">## =============== 2. PCA降维</span></span><br><span class="line">tissu1 &lt;- RunPCA(tissu1, features = VariableFeatures(object = tissu1))</span><br><span class="line"><span class="comment"># Examine and visualize PCA results a few different ways</span></span><br><span class="line">print(tissu1[[<span class="string">&quot;pca&quot;</span>]], dims = <span class="number">1</span>:<span class="number">5</span>, nfeatures = <span class="number">5</span>)</span><br><span class="line"><span class="comment">#visualization</span></span><br><span class="line">VizDimLoadings(tissu1, dims = <span class="number">1</span>:<span class="number">2</span>, reduction = <span class="string">&quot;pca&quot;</span>)</span><br><span class="line">DimPlot(tissu1, reduction = <span class="string">&quot;pca&quot;</span>)</span><br><span class="line">DimHeatmap(tissu1, dims = <span class="number">1</span>, cells = <span class="number">500</span>, balanced = <span class="literal">TRUE</span>)</span><br><span class="line">DimHeatmap(tissu1, dims = <span class="number">1</span>:<span class="number">20</span>, cells = <span class="number">500</span>, balanced = <span class="literal">TRUE</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## =============== 3. PC个数的确定</span></span><br><span class="line"><span class="comment">#这个非常非常具有挑战性，针对这个问题有很多教程,我们这里不着重讨论，因为我们主要是图表复现</span></span><br><span class="line"><span class="comment"># <span class="doctag">NOTE:</span> This process can take a long time for big datasets, comment out for expediency. More</span></span><br><span class="line"><span class="comment"># approximate techniques such as those implemented in ElbowPlot() can be used to reduce</span></span><br><span class="line"><span class="comment"># computation time</span></span><br><span class="line">tissu1 &lt;- JackStraw(tissu1, num.replicate = <span class="number">100</span>)</span><br><span class="line">tissu1 &lt;- ScoreJackStraw(tissu1, dims = <span class="number">1</span>:<span class="number">20</span>)</span><br><span class="line">JackStrawPlot(tissu1, dims = <span class="number">1</span>:<span class="number">20</span>)</span><br><span class="line">ElbowPlot(tissu1)</span><br></pre></td></tr></table></figure>
<h1 id="Step-6、细胞聚类"><a href="#Step-6、细胞聚类" class="headerlink" title="Step 6、细胞聚类"></a>Step 6、细胞聚类</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== 细胞聚类</span></span><br><span class="line">tissu1 &lt;- FindNeighbors(tissu1, dims = <span class="number">1</span>:<span class="number">10</span>)</span><br><span class="line"><span class="comment">#为了和原文聚类结果尽量一致，设置resolution = 0.02，这个数值是试出来的，resolution值越大，聚类数越多</span></span><br><span class="line">tissu1 &lt;- FindClusters(tissu1, resolution = <span class="number">0.02</span>)</span><br><span class="line"><span class="comment"># Look at cluster IDs of the first 5 cells</span></span><br><span class="line">head(Idents(tissu1), <span class="number">5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#查看每个类别多少个细胞</span></span><br><span class="line">head(tissu1@meta.data)</span><br><span class="line">table(tissu1@meta.data$seurat_clusters)</span><br><span class="line"></span><br><span class="line"><span class="comment"># If you haven&#x27;t installed UMAP, you can do so via reticulate::py_install(packages =</span></span><br><span class="line"><span class="comment"># &#x27;umap-learn&#x27;)</span></span><br><span class="line">tissu1 &lt;- RunUMAP(tissu1, dims = <span class="number">1</span>:<span class="number">10</span>)</span><br><span class="line"><span class="comment"># note that you can set `label = TRUE` or use the LabelClusters function to help label</span></span><br><span class="line"><span class="comment"># individual clusters</span></span><br><span class="line">DimPlot(tissu1, reduction = <span class="string">&quot;umap&quot;</span>,label = <span class="built_in">T</span>,label.size = <span class="number">5</span>)</span><br><span class="line"><span class="comment">#也可以tsne的方式展示细胞分区，但是umap给出的空间分布信息是有意义的，所以更倾向与使用umap</span></span><br><span class="line"><span class="comment">#tissu1 &lt;- RunTSNE(tissu1, dims = 1:10)</span></span><br><span class="line"><span class="comment">#DimPlot(tissu1, reduction = &quot;tsne&quot;,label = T,label.size = 5)</span></span><br></pre></td></tr></table></figure>
<p><strong>聚类结果，一共5个群</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/UMAP_cluster5_new.png"/>


<h1 id="Step-7、差异表达"><a href="#Step-7、差异表达" class="headerlink" title="Step 7、差异表达"></a>Step 7、差异表达</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== 差异表达及细胞类型鉴定</span></span><br><span class="line"><span class="comment"># find markers for every cluster compared to all remaining cells, report only the positive ones</span></span><br><span class="line">tissu1.markers &lt;- FindAllMarkers(tissu1, only.pos = <span class="literal">TRUE</span>, min.pct = <span class="number">0.25</span>, logfc.threshold = <span class="number">0.25</span>)</span><br><span class="line">write.csv(tissu1.markers,file = <span class="string">&quot;tissu1.markers.csv&quot;</span>)</span><br><span class="line"><span class="comment">#展示每一个cluster中top2显著的marker</span></span><br><span class="line">tissu1.markers %&gt;% group_by(cluster) %&gt;% top_n(n = <span class="number">2</span>, wt = avg_log2FC)</span><br></pre></td></tr></table></figure>
<p><strong>可视化特异性marker基因</strong></p>
<p><strong>1、 首先看一下文章中给出的marker基因</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/marker_genes_for_total.JPG" width="90%">

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">markergene&lt;-<span class="built_in">c</span>(<span class="string">&quot;KRT5&quot;</span>,<span class="string">&quot;PTPRC&quot;</span>,<span class="string">&quot;CLU&quot;</span>,<span class="string">&quot;AQP1&quot;</span>,<span class="string">&quot;COL3A1&quot;</span>,<span class="string">&quot;COL1A2&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><strong>2、 根据上表将marker基因映射到亚群上，以此确定不同亚群的具体细胞类型</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#umap可视化</span></span><br><span class="line"><span class="comment">#根据文章supplementary table 1中提供的亚群的marker gene信息，在此绘制表达图谱</span></span><br><span class="line">FeaturePlot(tissu1, features = <span class="built_in">c</span>(<span class="string">&quot;KRT5&quot;</span>, <span class="string">&quot;PTPRC&quot;</span>, <span class="string">&quot;CLU&quot;</span>, <span class="string">&quot;AQP1&quot;</span>, <span class="string">&quot;COL3A1&quot;</span>, <span class="string">&quot;COL1A2&quot;</span>))</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plot_marker_on_UMAP_new.png"/>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#小提琴图可视化</span></span><br><span class="line">library(ggplot2)</span><br><span class="line">plot_list&lt;-<span class="built_in">list</span>()</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="number">5</span>) &#123;</span><br><span class="line">  p=VlnPlot(tissu1,features = markergene[i])+NoLegend()</span><br><span class="line">  print(p)</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/KRT5.png" width="30%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PTPRC.png" width="30%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CLU.png" width="30%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/AQP1.png" width="30%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/COL3A1.png" width="30%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/COL1A2.png" width="30%">

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#dotplot图可视化</span></span><br><span class="line">DotPlot(tissu1, features = markergene)+RotatedAxis()+</span><br><span class="line">  scale_x_discrete(<span class="string">&quot;&quot;</span>)+scale_y_discrete(<span class="string">&quot;&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/dotplot_marker.png"/>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#细胞亚群占比统计</span></span><br><span class="line">cluster&lt;-<span class="built_in">c</span>(<span class="string">&quot;Cancer cell&quot;</span>,<span class="string">&quot;Immune cell&quot;</span>,<span class="string">&quot;Epithelial cells&quot;</span>,<span class="string">&quot;Endothelial cells&quot;</span>,<span class="string">&quot;Fibroblasts&quot;</span>)</span><br><span class="line"></span><br><span class="line">subset_cell&lt;-<span class="built_in">c</span>()</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="number">5</span>) &#123;</span><br><span class="line">  subset_cell=<span class="built_in">c</span>(subset_cell,<span class="built_in">dim</span>(tissu1[,Idents(tissu1)%in%<span class="built_in">c</span>(cluster[i])])[<span class="number">2</span>])</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">subset_account&lt;-<span class="built_in">c</span>()</span><br><span class="line">total_cells&lt;-<span class="built_in">dim</span>(tissu1)[<span class="number">2</span>]</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="number">5</span>) &#123;</span><br><span class="line">  subset_account=<span class="built_in">c</span>(subset_account,subset_cell[i]/total_cells)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">subset_account&lt;-subset_account*<span class="number">100</span></span><br><span class="line">per.cell &lt;- paste(cluster,<span class="string">&quot;: &quot;</span>,<span class="built_in">ceiling</span>(subset_account),<span class="string">&quot;%&quot;</span>,sep = <span class="string">&quot;&quot;</span>)</span><br><span class="line">pie(subset_account,labels=per.cell) </span><br></pre></td></tr></table></figure>
<p><strong>文章中的各亚群占比</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/subset_account_paper.JPG"/>

<p><strong>我们的计算结果如下，结果显示我们的计算结果和原文结果类似</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pie.JPG" width="60%">







<h1 id="Step-8、细胞亚群命名"><a href="#Step-8、细胞亚群命名" class="headerlink" title="Step 8、细胞亚群命名"></a>Step 8、细胞亚群命名</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== 细胞类型鉴定</span></span><br><span class="line">new.cluster.ids &lt;- <span class="built_in">c</span>(<span class="string">&quot;Cancer cell&quot;</span>, <span class="comment">#&quot;KRT5&quot;</span></span><br><span class="line">                     <span class="string">&quot;Immune cell&quot;</span>, <span class="comment">#&quot;PTPRC&quot;</span></span><br><span class="line">                     <span class="string">&quot;Epithelial cells&quot;</span>, <span class="comment">#&quot;CLU&quot;</span></span><br><span class="line">                     <span class="string">&quot;Endothelial cells&quot;</span>, <span class="comment">#&quot;AQP1&quot;</span></span><br><span class="line">                     <span class="string">&quot;Fibroblasts&quot;</span>) <span class="comment">#&quot;COL3A1&quot;, &quot;COL1A2&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">names</span>(new.cluster.ids) &lt;- <span class="built_in">c</span>(<span class="number">0</span>,<span class="number">1</span>,<span class="number">2</span>,<span class="number">5</span>,<span class="number">4</span>)</span><br><span class="line">tissu1 &lt;- RenameIdents(tissu1, new.cluster.ids)</span><br><span class="line">DimPlot(tissu1, reduction = <span class="string">&quot;umap&quot;</span>, label = <span class="literal">TRUE</span>, pt.size = <span class="number">0.5</span>) + NoLegend()</span><br></pre></td></tr></table></figure>
<p><strong>我们的聚类结果</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/UMAP_cluster5_lable_new.png"/>

<p><strong>原文的聚类结果</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/paper_total_cluster.JPG"/>


<h1 id="Step-9、细胞亚群提取及再分群"><a href="#Step-9、细胞亚群提取及再分群" class="headerlink" title="Step 9、细胞亚群提取及再分群"></a>Step 9、细胞亚群提取及再分群</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## =============== </span></span><br><span class="line"><span class="comment">#肿瘤细胞的提取，下面两种方式都可以实现特定亚群的提取</span></span><br><span class="line">Cancer_cell&lt;-tissu1[,tissu1@meta.data$seurat_clusters%in%<span class="built_in">c</span>(<span class="number">0</span>)]</span><br><span class="line"><span class="comment">#Cancer_cell&lt;-tissu1[,Idents(tissu1)%in%c(&quot;Cancer cell&quot;)]</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#标准化之前</span></span><br><span class="line">Cancer_cell &lt;- NormalizeData(Cancer_cell, normalization.method = <span class="string">&quot;LogNormalize&quot;</span>, scale.factor = <span class="number">10000</span>) </span><br><span class="line">Cancer_cell &lt;- FindVariableFeatures(Cancer_cell, selection.method = <span class="string">&#x27;vst&#x27;</span>, nfeatures = <span class="number">2000</span>)</span><br><span class="line"><span class="comment"># Identify the 10 most highly variable genes</span></span><br><span class="line">top10 &lt;- head(VariableFeatures(Cancer_cell), <span class="number">10</span>)</span><br><span class="line">top10</span><br><span class="line"></span><br><span class="line">Cancer_cell &lt;- ScaleData(Cancer_cell, vars.to.regress = <span class="string">&quot;percent.mt&quot;</span>)</span><br><span class="line">Cancer_cell &lt;- RunPCA(Cancer_cell, features = VariableFeatures(object = Cancer_cell)) </span><br><span class="line">DimHeatmap(Cancer_cell, dims = <span class="number">1</span>:<span class="number">20</span>, cells = <span class="number">500</span>, balanced = <span class="literal">TRUE</span>)</span><br><span class="line">Cancer_cell &lt;- FindNeighbors(Cancer_cell, dims = <span class="number">1</span>:<span class="number">10</span>)</span><br><span class="line">Cancer_cell &lt;- FindClusters(Cancer_cell, resolution = <span class="number">0.3</span>)</span><br><span class="line">Cancer_cell &lt;- RunUMAP(Cancer_cell, dims = <span class="number">1</span>:<span class="number">10</span>)</span><br><span class="line">DimPlot(Cancer_cell, reduction = <span class="string">&#x27;umap&#x27;</span>,label = <span class="built_in">T</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#cluster Keratinocyte Cancer</span></span><br><span class="line">FeaturePlot(Cancer_cell, features = <span class="built_in">c</span>(<span class="string">&quot;SPRR3&quot;</span>,<span class="string">&quot;SPRR1A&quot;</span>,<span class="string">&quot;NCCRP1&quot;</span>,<span class="string">&quot;TMPRSS11E&quot;</span>,<span class="string">&quot;APOBEC3A&quot;</span>))</span><br></pre></td></tr></table></figure>
<p><strong>绞尽脑汁，一顿操作，最后还是没有办法很好的复现原文的数据，但是其中一个亚群有较好的聚类出来。</strong></p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Keratinocyte_cancer_cluster.png"/>


<h2 id="总结"><a href="#总结" class="headerlink" title="总结"></a>总结</h2><blockquote>
<p>1、单细胞数据复现可以提升自己对此类数据的理解和掌握。<br>2、不同人处理单细胞数据时设置的参数会有所不同，作者并没有在文章中给出太多相关信息，导致最终的结果有所不同。<br>3、单细胞亚群个数的确定以及类群注释需要非常强的生物学背景。<br>4、 要学的东西太多了，一点点的来，不求快，但求稳，加油💪</p>
</blockquote>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="reward-container">
  <div>If you like this post, you can buy me a coffee ^_^</div>
  <button onclick="document.querySelector('.post-reward').classList.toggle('active');">
    Donate
  </button>
  <div class="post-reward">
      <div>
        <img src="/images/alipay.jpg" alt="gongyuqiqi Alipay">
        <span>Alipay</span>
      </div>

  </div>
</div>

          <div class="post-tags">
              <a href="/tags/tag1/" rel="tag"><i class="fa fa-tag"></i> tag1</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2021/05/09/10X-GENOMICS-pipeline/" rel="prev" title="10X GENOMICS pipeline">
                  <i class="fa fa-chevron-left"></i> 10X GENOMICS pipeline
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2021/05/14/WES/" rel="next" title="WES">
                  WES <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






    <div class="comments utterances-container"></div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>
</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">gongyuqiqi</span>
</div>
<div class="wordcount">
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-line"></i>
    </span>
      <span>Symbols count total: </span>
    <span title="Symbols count total">168k</span>
  </span>
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
      <span>Reading time total &asymp;</span>
    <span title="Reading time total">2:33</span>
  </span>
</div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@3.5.7/dist/jquery.fancybox.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  
<script src="/js/local-search.js"></script>






  





<script>
NexT.utils.loadComments('.utterances-container', () => {
  const script = document.createElement('script');
  script.src = 'https://utteranc.es/client.js';
  script.setAttribute('repo', "gongyuqiqi/utterances");
  script.setAttribute('issue-term', "pathname");
  script.setAttribute('theme', "github-light");
  script.crossOrigin = 'anonymous';
  script.async = true;
  document.querySelector('.utterances-container').appendChild(script);
});
</script>

</body>
</html>
