<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.3.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","images":"/images","scheme":"Muse","version":"8.2.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}};
  </script>
<meta name="description" content="ATAC-seq旧版处理流程">
<meta property="og:type" content="article">
<meta property="og:title" content="ATAC-seq Analysis">
<meta property="og:url" content="http://example.com/2020/02/08/ATAC-seq%20Analysis/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="ATAC-seq旧版处理流程">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rename_rawdata.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ubuntu-atac-seq-image.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-ls.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ATAC-environment.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/clean-data.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawqc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trimqc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw_multiqc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim_multiqc.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-raw-heatmap.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-trim-heatmap.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-sequence-display.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim-sequence-display.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/genome.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/gtf.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/mm10_index.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/align_result.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.bai.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.stat.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-bam-bai.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-stat.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-last-all.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/last-bed.png">
<meta property="og:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/peaks.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/feature-distribution.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/loci-to-TSS.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-over-chromsomes.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-overlap.JPG">
<meta property="article:published_time" content="2020-02-08T08:30:23.000Z">
<meta property="article:modified_time" content="2021-02-14T07:39:16.661Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png">


<link rel="canonical" href="http://example.com/2020/02/08/ATAC-seq%20Analysis/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>
<title>ATAC-seq Analysis | gyq's blog</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">gyq's blog</h1>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>







</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#ATAC-seq-Analysis"><span class="nav-number">1.</span> <span class="nav-text">ATAC-seq Analysis</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%80-%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="nav-number">1.1.</span> <span class="nav-text">(一)数据下载</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E4%BA%8C%EF%BC%89%E8%87%AA%E5%AE%9A%E4%B9%89docker%E5%AE%B9%E5%99%A8%E7%9A%84%E5%90%AF%E5%8A%A8%E4%B8%8E%E6%95%B0%E6%8D%AE%E5%85%B1%E4%BA%AB"><span class="nav-number">1.2.</span> <span class="nav-text">（二）自定义docker容器的启动与数据共享</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E4%B8%89%EF%BC%89%E3%80%81%E6%95%B0%E6%8D%AE%E8%BF%87%E6%BB%A4"><span class="nav-number">1.3.</span> <span class="nav-text">（三）、数据过滤</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9B%9B-%E3%80%81%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6"><span class="nav-number">1.4.</span> <span class="nav-text">(四)、质量控制</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%94-%E6%AF%94%E5%AF%B9%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">1.5.</span> <span class="nav-text">(五)比对参考基因组</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%EF%BC%88%E5%85%AD%EF%BC%89%E4%BD%BF%E7%94%A8macs2%E6%89%BEpeaks"><span class="nav-number">1.6.</span> <span class="nav-text">（六）使用macs2找peaks</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#peaks%E6%B3%A8%E9%87%8A"><span class="nav-number">1.7.</span> <span class="nav-text">peaks注释</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">gongyuqiqi</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives">
          <span class="site-state-item-count">28</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/02/08/ATAC-seq%20Analysis/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="gongyuqiqi">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="gyq's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          ATAC-seq Analysis
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-02-08 16:30:23" itemprop="dateCreated datePublished" datetime="2020-02-08T16:30:23+08:00">2020-02-08</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2021-02-14 15:39:16" itemprop="dateModified" datetime="2021-02-14T15:39:16+08:00">2021-02-14</time>
      </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <p>ATAC-seq旧版处理流程</p>
<a id="more"></a>

<h1 id="ATAC-seq-Analysis"><a href="#ATAC-seq-Analysis" class="headerlink" title="ATAC-seq Analysis"></a>ATAC-seq Analysis</h1><p>表观测序数据中比较重要的就是ATAC-seq、Chip-seq。这两者的原理和分析过程类似。<br>具体的异同详见此链接：<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/87bc2002e82c">https://www.jianshu.com/p/87bc2002e82c</a></p>
<h2 id="一-数据下载"><a href="#一-数据下载" class="headerlink" title="(一)数据下载"></a>(一)数据下载</h2><p><font color=red>数据下载是我走过的最曲折的路</font></p>
<p>强烈不推荐wget、curl——需要向上天再借500年才能等的起。</p>
<p>不推荐prefetch——不知道为什么prefetch在我这里也不起作用！按理说有ascp环境后，prefetch会自动调用ascp进行数据下载呀！可是prefetch依然显示通过https进行数据下载，这就没法下载呀！直接timeout！</p>
<p><strong>正解——调用ascp下载数据</strong></p>
<p>（1）ascp下载NCBI的SRA文件</p>
<p>此方法亲测不行，但是就在2个月前还是可以的。官网给出的解释大致意思就是——不好意思，这段时间出现无法通过ascp下载NCBI数据属于正常现象…blablabla…</p>
<p>（2）ascp下载EBI的fastq文件（强烈推荐，省去了SRA转成fastq的麻烦，很耗资源，很耗时）</p>
<p>在此次数据分析中，下载SRR2927015、SRR2927016、SRR2927018、SRR3545580对应的fastq文件进行ATAC-seq数据分析的实战。</p>
<p>EBI官网输入对应SRR号（以SRR3545580为例），找到相应的fastq文件位置，复制连接地址。</p>
<p><a target="_blank" rel="noopener" href="http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz">http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz</a></p>
<p><a target="_blank" rel="noopener" href="http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz">http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz</a></p>
<p>下载方式如下（下载速递可达到每秒M级别，快的时候能到60M+，亲测此方法为目前最可取的方法）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ascp -QT -l 300m -P33001 -i /home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_1.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i /home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR354/000/SRR3545580/SRR3545580_2.fastq.gz .</span><br></pre></td></tr></table></figure>
<p>最终下载结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawdata.png"></p>
<p>为了后续分析，将其改名为下</p>
<p>2-cell-1  SRR2927015<br>2-cell-2  SRR2927016<br>2-cell-4  SRR2927018<br>2-cell-5  SRR3545580</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rename_rawdata.png"></p>
<p>附上测序数据下载方式的相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/df34b99cbd43">https://www.jianshu.com/p/df34b99cbd43</a></p>
<p><a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/171/">https://www.bioinfo-scrounger.com/archives/171/</a></p>
<h2 id="（二）自定义docker容器的启动与数据共享"><a href="#（二）自定义docker容器的启动与数据共享" class="headerlink" title="（二）自定义docker容器的启动与数据共享"></a>（二）自定义docker容器的启动与数据共享</h2><p>之前创建了具备ATAC-seq数据分析所需环境的docker容器，并push到阿里云。在此只需要pull该镜像，run即可使用。</p>
<p>但是！！！创建容器时并没有装ascp，因为只有普通用户才可以运行ascp。所以数据下载是在ubuntu主机进行的。为了能够在容器中使用下载的测试数据，运用数据卷，实现主机和容器间的数据共享。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 查看镜像</span><br><span class="line">docker images</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ubuntu-atac-seq-image.png"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"># 启动容器实现数据共享</span><br><span class="line">docker run -it -v &#x2F;home&#x2F;gongyuqi&#x2F;Downloads&#x2F;ATACdata&#x2F;:&#x2F;root&#x2F;project&#x2F;atac&#x2F;raw 608eafa789c8</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># 查看容器内是否已经存在fastq数据</span><br><span class="line">cd root&#x2F;project&#x2F;atac&#x2F;raw</span><br><span class="line">ls</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-ls.png"></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"># 激活ATAC环境（该环境配置有处理ATAC-seq数据的所需软件）</span><br><span class="line">conda activate ATAC</span><br><span class="line"># (base)变成了(ATAC)，即由最初的conda环境变成了ATAC环境</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/ATAC-environment.png"></p>
<h2 id="（三）、数据过滤"><a href="#（三）、数据过滤" class="headerlink" title="（三）、数据过滤"></a>（三）、数据过滤</h2><p><strong>运用软件 Trim Galore</strong></p>
<p><strong>去除接头和质量不合格的reads</strong></p>
<blockquote>
<p>Trim Galore是对FastQC和Cutadapt的包装。适用于所有高通量测序，包括RRBS(Reduced Representation Bisulfite-Seq ), Illumina、Nextera 和smallRNA测序平台的双端和单端数据。</p>
</blockquote>
<blockquote>
<p>Trimmomatic是针对Illumina高通量测序平台设计的接头去除和低质量reads清洗软件。</p>
</blockquote>
<p>附上trim-galore相关参考资料: <a target="_blank" rel="noopener" href="https://www.jianshu.com/p/7a3de6b8e503">https://www.jianshu.com/p/7a3de6b8e503</a></p>
<p>以下列数据为例</p>
<p>2-cell-1  SRR2927015</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">trim_galore -q 7 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o &#x2F;root&#x2F;project&#x2F;atac&#x2F;clean&#x2F; 2-cell-1_1.fastq.gz  2-cell-1_2.fastq.gz</span><br></pre></td></tr></table></figure>
<blockquote>
<p><strong>trim_galore</strong></p>
<p>–q：设置线程</p>
<p>–phred33：使用ASCII+33质量值作为Phred的分</p>
<p>–length：去除长度小于参数值的reads</p>
<p>-e：允许的最大误差</p>
<p>–stringency：设置与接头重叠的序列</p>
<p>–paired：对于双段测序文件，正反链质控通过才保留</p>
<p>-o：输出目录</p>
</blockquote>
<p>结果如下</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/clean-data.png"></p>
<p>因为本次数据是用本机（8线程，16G）跑的，要给机子休息的时间，同时实验样本少，所以一个一个样本手动跑。<br>但是！当样本量多了，在服务器上跑的时候，循环就要省事很多，比如跑的时候去睡个觉…</p>
<p><strong>拿本次样本写个循环，举个例</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ls &#x2F;root&#x2F;project&#x2F;raw&#x2F;*1.fq.gz &gt;1</span><br><span class="line">ls &#x2F;root&#x2F;project&#x2F;raw&#x2F;*2.fq.gz &gt;2</span><br><span class="line">paste 1 2 &gt; sample</span><br><span class="line">path&#x3D;&#x2F;root&#x2F;project&#x2F;atac&#x2F;clean</span><br><span class="line">cat sample | while read id; do echo $id arr&#x3D;($id) fq1&#x3D;$&#123;arr[0]&#125; fq2&#x3D;$&#123;arr[1]&#125; nohup trim_galore -q 30 --phred33 -e 0.1 --stringency 4 --paired -o $path $fq1 $fq2 &amp; done</span><br></pre></td></tr></table></figure>
<h2 id="四-、质量控制"><a href="#四-、质量控制" class="headerlink" title="(四)、质量控制"></a>(四)、质量控制</h2><p><font color=red>测试数据的质控真的非常重要！！！如果测序质量本身就很好，没有质空这一步也影响不大。如果测序质量不好，不进行质控的化，直接影响后续的比对率。一般情况下，90%+的比对率才算比较正常，太低的比对率往往是质控步骤有问题，会非常严重的影响最终结果。亲测过！！！！</font></p>
<p><strong>运用软件fastqc、multiqc</strong></p>
<p><strong>进行批处理</strong></p>
<ul>
<li>数据质控</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">cd &#x2F;root&#x2F;project&#x2F;atac&#x2F;qc &amp;&amp; mkdir rawqc trimqc</span><br><span class="line"># 对原始数据进行质控</span><br><span class="line">cd rawqc</span><br><span class="line">fastqc -t 7 &#x2F;root&#x2F;project&#x2F;atac&#x2F;raw&#x2F;*gz -o .&#x2F;</span><br><span class="line"># 对过滤数据进行质控</span><br><span class="line">cd .. &amp; cd trimqc</span><br><span class="line">fastqc -t 7 &#x2F;root&#x2F;project&#x2F;atac&#x2F;clean&#x2F;*gz -o .&#x2F;</span><br></pre></td></tr></table></figure>
<p>（1）原始数据质控结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rawqc.png"></p>
<p>（2）过滤后数据质量控制结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trimqc.png"></p>
<ul>
<li>质控结果整合</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"># 原始数据qc结果整合</span><br><span class="line">cd &#x2F;root&#x2F;project&#x2F;atac&#x2F;qc&#x2F;rawqc &amp;&amp; mkdir multiqc_results</span><br><span class="line">multiqc . -o multiqc_results&#x2F;</span><br><span class="line"># 过滤数据qc结果整合</span><br><span class="line">cd &#x2F;root&#x2F;project&#x2F;atac&#x2F;qc&#x2F;trimqc &amp;&amp; mkdir multiqc_results</span><br><span class="line">multiqc . -o multiqc_results&#x2F;</span><br></pre></td></tr></table></figure>
<p>（1）原始数据整合结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw_multiqc.png"></p>
<p>（2）过滤数据整合结果</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim_multiqc.png"></p>
<p>附上multiqc相关参考资料<br><a target="_blank" rel="noopener" href="https://blog.csdn.net/ada0915/article/details/77201871">https://blog.csdn.net/ada0915/article/details/77201871</a></p>
<ul>
<li>mulitiqc整合结果解读</li>
</ul>
<p>(1)过滤前</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-raw-heatmap.png"></p>
<p>（2）过滤后</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/multiqc-trim-heatmap.png"></p>
<p>（3）解读（随便一搜，各种解读教程实在太多了～）</p>
<p>trim-galore完美的去除了adaptor;去除了质量不合格的序列使得2-cell-1_2和2-cell-2_2的质量过关；2-cell-1_2的N含量超标的序列被去除了。</p>
<p>因为去除了adptor和质量不合格的序列，所以sequence length自然是不一致的了。另外，还没有PCR去重，所以sequence duplication是不过关的，这一点会在PCR去重步骤中得到解决。至于Per Base Sequence Content,前10个碱基总是异常的波动，比对序列长度远大于10，处理时也可以设定不比对前10个，所以这个关系不大。</p>
<p>（4）以Sequence Length和Per Base N Content展示过滤前后的区别</p>
<p>过滤前</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/raw-sequence-display.png"></p>
<p>过滤后</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/trim-sequence-display.png"></p>
<h2 id="五-比对参考基因组"><a href="#五-比对参考基因组" class="headerlink" title="(五)比对参考基因组"></a>(五)比对参考基因组</h2><p><strong>运用软件bowtie2、bedtools</strong></p>
<p>一般对于DNA数据，比对软件一般使用：<strong>bowtie2、bwa</strong></p>
<p>对于RNA数据，比对一般使用：<strong>hisat2、star</strong></p>
<p>使用<strong>bedtools</strong>将最终的bam文件转成bed文件</p>
<p><strong>（1）参考基因组及其注释文件的下载</strong></p>
<p>参考基因组（解压后）<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/genome.png"></p>
<p>参考基因组注释文件（解压后）<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/gtf.png"></p>
<p>本次分析用到的是小鼠的参考基因组及其注释文件</p>
<p>mm10.fa、gencode.vM24.annotation.gtf</p>
<p><strong>（2）参考基因组索引构建</strong>（将构建的参考基因组索引文件cp到容器中的/root/project/atac/reference/mm10_index目录下）</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-build &#x2F;home&#x2F;gongyuqi&#x2F;biosoftware&#x2F;reference&#x2F;genome&#x2F;mm10.fa mm10_index</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/mm10_index.png"></p>
<p><strong>（3）参考基因组的比对，直接生成bam文件</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 -p 7 -x &#x2F;root&#x2F;project&#x2F;atac&#x2F;reference&#x2F;mm10_index -1 ..&#x2F;clean&#x2F;2-cell-1_1_val_1.fq.gz -2 ..&#x2F;clean&#x2F;2-cell-1_2_val_2.fq.gz | samtools sort -@ 7 -O bam -o 2-cell-1.bam -</span><br><span class="line"># 此samtools sort默认按照reads比对到染色体上的坐标排序</span><br></pre></td></tr></table></figure>
<p>bowtie2直接比对的结果会生成sam文件，sam文件很大，比较占空间。在比对这一步就直接利用samtools sort将其转成bam文件。</p>
<p>另外，samtool排序过程中并不会丢弃一部分reads，但是排序后的bam文件相比排序前的体积会更小。</p>
<p>比对相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/6ed1bfbb7b72">https://www.jianshu.com/p/6ed1bfbb7b72</a></p>
<p><strong>比对结果</strong></p>
<p>以2-cell-1_1_val为例进行结果展示</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/align_result.png"></p>
<p><strong>（4）构建原始bam文件的索引文件、状态文件</strong></p>
<p>构建索引文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do samtools index -@ 7 $id; done</span><br></pre></td></tr></table></figure>
<p>.bai文件即位生成的索引文件</p>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.bai.png"></p>
<p>构建状态文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/bam.stat.png"></p>
<p><strong>（5）去除PCR重复、构建去重后的索引文件、状态文件、</strong></p>
<p>sambamba软件去重，会同时生成了去重后的bam和bai文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | while read id; do sambamba markdup -r -t 7 $id rmdup.$id; done</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-bam-bai.png"></p>
<p>构建状态文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls rmdup*.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-stat.png"></p>
<p><strong>（6）去除同一样本的两条reads无法落在同一条染色体及测序质量低于30的reads、去除线粒体基因、构建索引文件、构建状态文件</strong></p>
<p>去除不符合要求的reads及线粒体基因</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls rm*.bam | while read id; do samtools view -@ 7 -h -f 2 -q 30 $id | grep -v chrM |samtools sort -O bam -@7 -o last.$id; done</span><br></pre></td></tr></table></figure>
<p>构建索引文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do samtools index -@ 7 $id; done</span><br></pre></td></tr></table></figure>
<p>构建状态文件</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do samtools flagstat -@ 7 $id &gt; $id.stat; done</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/rmdup-last-all.png"></p>
<p><strong>（7）生成bed文件</strong></p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls last*.bam | while read id; do bedtools bamtobed -i $id &gt; $id.bed; done</span><br></pre></td></tr></table></figure>
<p><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/last-bed.png"></p>
<h2 id="（六）使用macs2找peaks"><a href="#（六）使用macs2找peaks" class="headerlink" title="（六）使用macs2找peaks"></a>（六）使用macs2找peaks</h2><p><strong>使用软件macs2</strong></p>
<p>以last.rmdup.2-cell-1.bam.bed进行演示</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">macs2 callpeak -t last.rmdup.2-cell-1.bam.bed -g mm --nomodel --shift -100 --extsize 200 -n 2-cell-1 --outdir ..&#x2F;peaks&#x2F;</span><br></pre></td></tr></table></figure>
<p>最终生成文件<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/peaks.png"></p>
<h2 id="peaks注释"><a href="#peaks注释" class="headerlink" title="peaks注释"></a>peaks注释</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">&quot;path\\to\\.bed&quot;</span>)</span><br><span class="line">library(ChIPpeakAnno)</span><br><span class="line">library(TxDb.Mmusculus.UCSC.mm10.knownGene)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(BiocInstaller)</span><br><span class="line">library(ChIPseeker)</span><br><span class="line">txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter &lt;- getPromoters(TxDb=txdb, upstream=<span class="number">3000</span>, downstream=<span class="number">3000</span>)</span><br><span class="line">files = <span class="built_in">list</span>(cell_1_summits = <span class="string">&quot;2-cell-1_summits.bed&quot;</span>, cell_2_summits = <span class="string">&quot;2-cell-2_summits.bed&quot;</span>,</span><br><span class="line">             cell_4_summits = <span class="string">&quot;2-cell-4_summits.bed&quot;</span>, cell_5_summits = <span class="string">&quot;2-cell-5_summits.bed&quot;</span>)</span><br><span class="line">peakAnno &lt;- annotatePeak(files[[<span class="number">1</span>]], <span class="comment"># 分别改成2或者3或者4即可，分别对应四个文件</span></span><br><span class="line">                         tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                         TxDb=txdb, annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br></pre></td></tr></table></figure>
<p><strong>（1）拿到每个样本中peaks对应得基因名</strong></p>
<p>这一步非常重要，拿到基因名就可以根据课题需要进行差异分析等</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">genelist&lt;-peakAnno@anno@elementMetadata@listData[[<span class="string">&quot;SYMBOL&quot;</span>]]</span><br></pre></td></tr></table></figure>
<p><strong>（2）单个文件作图</strong></p>
<p>图片就不做展示了</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">plotAnnoPie(peakAnno)</span><br><span class="line">plotAnnoBar(peakAnno)</span><br><span class="line">vennpie(peakAnno)</span><br><span class="line">upsetplot(peakAnno)</span><br><span class="line">upsetplot(peakAnno, vennpie=<span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><strong>（3）plotAnnoBar和plotDistToTSS这两个柱状图都支持多个数据同时展示，方便比较</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">&quot;path\\to\\.bed&quot;</span>)</span><br><span class="line">library(ChIPpeakAnno)</span><br><span class="line">library(TxDb.Mmusculus.UCSC.mm10.knownGene)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(BiocInstaller)</span><br><span class="line">library(ChIPseeker)</span><br><span class="line">txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter &lt;- getPromoters(TxDb=txdb, upstream=<span class="number">3000</span>, downstream=<span class="number">3000</span>)</span><br><span class="line">files = <span class="built_in">list</span>(cell_1_summits = <span class="string">&quot;2-cell-1_summits.bed&quot;</span>, cell_2_summits = <span class="string">&quot;2-cell-2_summits.bed&quot;</span>,</span><br><span class="line">             cell_4_summits = <span class="string">&quot;2-cell-4_summits.bed&quot;</span>, cell_5_summits = <span class="string">&quot;2-cell-5_summits.bed&quot;</span>)</span><br><span class="line">peakAnnoList &lt;- lapply(files, annotatePeak, </span><br><span class="line">                       TxDb=txdb,tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>))</span><br><span class="line">plotAnnoBar(peakAnnoList)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/feature-distribution.JPG"/>


<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">plotDistToTSS(peakAnnoList, title = <span class="string">&quot;Distribution of transcription factor-binding loci\nrelative to TSS&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/loci-to-TSS.JPG"/>

<p><strong>（4）查看peaks在全基因组的位置（以2-cell-1样本为例）</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chropeaks&lt;-readPeakFile(<span class="string">&quot;2-cell-1_peaks.narrowPeak&quot;</span>)</span><br><span class="line">covplot(chropeaks, weightCol=<span class="number">5</span>, title = <span class="string">&quot;Peaks over chromosomes&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-over-chromsomes.JPG"/>


<p><strong>（4）重复样本间共同的peaks和注释基因</strong></p>
<p>也可以用拿到的各个样本的基因名，做韦恩图，会比这个图更美观。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">genes&lt;-lapply(peakAnnoList, <span class="keyword">function</span>(i) as.data.frame(i)$geneId)</span><br><span class="line">vennplot(genes)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/peaks-overlap.JPG"/>

<p>peaks注释相关参考资料</p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c83a38915afc">https://www.jianshu.com/p/c83a38915afc</a></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/NGS/" rel="tag"># NGS</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/02/08/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%BA%8C%EF%BC%89/" rel="prev" title="scRNA-seq数据分析实战（二）">
                  <i class="fa fa-chevron-left"></i> scRNA-seq数据分析实战（二）
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2020/02/08/LVM%E9%80%BB%E8%BE%91%E5%8D%B7%E7%AE%A1%E7%90%86%E5%99%A8/" rel="next" title="LVM逻辑卷管理器">
                  LVM逻辑卷管理器 <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>







<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>
</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">gongyuqiqi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/muse/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/schemes/muse.js"></script><script src="/js/next-boot.js"></script>

  






  





</body>
</html>
