<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>gyq&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta property="og:type" content="website">
<meta property="og:title" content="gyq&#39;s blog">
<meta property="og:url" content="http://example.com/page/2/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="gongyuqiqi">
<meta name="twitter:card" content="summary">
  
    <link rel="alternate" href="/atom.xml" title="gyq's blog" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 5.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">gyq&#39;s blog</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main">
  
    <article id="post-MySQL-transaction-isolation" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/03/03/MySQL-transaction-isolation/" class="article-date">
  <time class="dt-published" datetime="2020-03-02T18:38:39.000Z" itemprop="datePublished">2020-03-03</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/03/03/MySQL-transaction-isolation/">MySQL中事物的隔离级别</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="MySQL中事物的隔离级别"><a href="#MySQL中事物的隔离级别" class="headerlink" title="MySQL中事物的隔离级别"></a>MySQL中事物的隔离级别</h1><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">事物的隔离界别</span><br><span class="line">					脏读		不可重复读		幻读</span><br><span class="line">read uncommited 	√			√				√</span><br><span class="line">read commited		×			√				√</span><br><span class="line">repeatable read		×			×				√</span><br><span class="line">serialization		×			×				×</span><br><span class="line"></span><br><span class="line">mysql中默认 repeatable read</span><br><span class="line">oracle中默认 read commited</span><br><span class="line"></span><br><span class="line">查看隔离级别：select @@tx_isolation;</span><br><span class="line">设置隔离级别：set session|global transaction isolation level 隔离级别;</span><br></pre></td></tr></table></figure>
<h2 id="read-uncommitted"><a href="#read-uncommitted" class="headerlink" title="read uncommitted"></a>read uncommitted</h2><p><strong>无隔离</strong><br>脏读（演示）、不可重复读（演示）、幻读（这里不演示）  </p>
<ul>
<li>终端1中显示当前隔离级别  </li>
<li>设置当前隔离级别为read committed  </li>
<li>在此查看当前隔离级别  </li>
<li>进入test数据库  </li>
<li>查看accout表信息  </li>
<li>开启事物  </li>
<li>在此事物中进行列的修改  </li>
<li>回退  </li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/read_uncommitted1.png"/>

<ul>
<li>终端2显示当前隔离级别</li>
<li>设置当前隔离级别为read committed</li>
<li>在此查看当前隔离级别</li>
<li>进入test数据库</li>
<li>查看account表信息（在终端1做出修改前）</li>
<li>查看account表信息（在终端1做出修改后）</li>
<li>查看account表信息（在终端1回退后）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/read_uncommited2.png"/>


<h2 id="read-committed"><a href="#read-committed" class="headerlink" title="read committed"></a>read committed</h2><p>不脏读（演示）、不可重复读（不演示）、幻读（这里不演示）  </p>
<ul>
<li>设置终端1类型read committed</li>
<li>开启事物</li>
<li>修改列</li>
<li>结束事务</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/read_commited1.png"/>

<ul>
<li>设置终端2类型read committed   </li>
<li>开启事务  </li>
<li>查看account表信息（终端做修改后。但未结束事务，不脏读）  </li>
<li>查看account表信息（终端1做修改后，结束事务，终端2还处于同一事务中，此隔离级别下不可重复读）  </li>
<li>结束事务</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/read_commited2.png"/>

<h2 id="repeatable-read"><a href="#repeatable-read" class="headerlink" title="repeatable read"></a>repeatable read</h2><p>不脏读（演示）、不可重复读（演示）、幻读（这里不演示）  </p>
<ul>
<li>设置终端1类型repeatable read</li>
<li>开启事物</li>
<li>修改列</li>
<li>结束事务</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/repeatable_read1.png"/>

<ul>
<li>设置终端2类型repeatable read</li>
<li>开启事务</li>
<li>查看account表信息（终端1做修改后，未结束事务）</li>
<li>查看account表信息（终端1做修改后，结束事务，终端2未结束事务，此隔离级别下可重复读）</li>
<li>结束事务</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/repeatable_read2.png"/>

<ul>
<li>终端1开启事务</li>
<li>查看account表信息</li>
<li>插入一行信息</li>
<li>查看account表信息</li>
<li>结束事务</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/repeatable_read_illusion1.png"/>

<ul>
<li>终端2开启事务  </li>
<li>查看account表信息（终端1做出修改后）  </li>
<li>修改表中的列信息（终端1做出修改，但未结束事务，时间长了终端2会出现来纳瑟方框报错）  </li>
<li>在此尝试修改表中列信息（终端1做出修改并结束事务，出现红色下划线提示，4个改变，但在终端2事务中最初只有3行信息，此处却出现4行，说明出现幻读）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/repeatable_read_illusion2.png"/>


<h2 id="serialization"><a href="#serialization" class="headerlink" title="serialization"></a>serialization</h2><p>不脏读、可重复度、不幻读</p>
<ul>
<li>设置终端1类型serialization</li>
<li>查看当前隔离级别</li>
<li>开启事务</li>
<li>查看account表信息</li>
<li>修改表的列信息（此处并未提交执行此语句）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/serializable1.png"/>

<ul>
<li>设置终端1类型serialization</li>
<li>开启事务</li>
<li>修改表的列信息（终端1的表修改操作在终端2前，此时终端2列修改无法操作，红色方框未报错信息，等待时间过长，只有端事务1事务结束才能进行此操作）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/serializable2.png"/>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/03/03/MySQL-transaction-isolation/" data-id="ckl4ra1lo0004agvfhw4r1zgb" data-title="MySQL中事物的隔离级别" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-MySQL简单介绍" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/25/MySQL%E7%AE%80%E5%8D%95%E4%BB%8B%E7%BB%8D/" class="article-date">
  <time class="dt-published" datetime="2020-02-24T16:00:00.000Z" itemprop="datePublished">2020-02-25</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/25/MySQL%E7%AE%80%E5%8D%95%E4%BB%8B%E7%BB%8D/">MySQL简单介绍</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>MySQL的下载安装<br>简单命令介绍<br>SQLyog的下载安装</p>
        
          <p class="article-more-link">
            <a href="/2020/02/25/MySQL%E7%AE%80%E5%8D%95%E4%BB%8B%E7%BB%8D/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/25/MySQL%E7%AE%80%E5%8D%95%E4%BB%8B%E7%BB%8D/" data-id="ckl4ra1lp0005agvf131vgubw" data-title="MySQL简单介绍" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-SSH服务" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/24/SSH%E6%9C%8D%E5%8A%A1/" class="article-date">
  <time class="dt-published" datetime="2020-02-23T16:00:00.000Z" itemprop="datePublished">2020-02-24</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/24/SSH%E6%9C%8D%E5%8A%A1/">SSH服务</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>远程控制服务：配置sshd服务、 安全密钥验证、远程传输命令<br>不间断会话服务：管理远程会话、会话共享功能</p>
        
          <p class="article-more-link">
            <a href="/2020/02/24/SSH%E6%9C%8D%E5%8A%A1/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/24/SSH%E6%9C%8D%E5%8A%A1/" data-id="ckl4ra1lu000bagvf7t2i02uj" data-title="SSH服务" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/linux/" rel="tag">linux</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-RNA-seq analysis" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/23/RNA-seq%20analysis/" class="article-date">
  <time class="dt-published" datetime="2020-02-23T08:30:23.000Z" itemprop="datePublished">2020-02-23</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/23/RNA-seq%20analysis/">RNA-seq数据分析流程</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <a id="more"></a>

<h1 id="RNA-seq数据分析流程"><a href="#RNA-seq数据分析流程" class="headerlink" title="RNA-seq数据分析流程"></a>RNA-seq数据分析流程</h1><h2 id="（一）原始数据的质控"><a href="#（一）原始数据的质控" class="headerlink" title="（一）原始数据的质控"></a>（一）原始数据的质控</h2><p>以其中一个样本为例（其他样本指控结果基本类似）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/raw-data</span><br><span class="line"><span class="comment">#对每个样本进行质控</span></span><br><span class="line">fastqc -t 7 /root/project/atac/raw/*gz -o ./</span><br><span class="line"><span class="comment">#统计合并所有指控结果</span></span><br><span class="line">multiqc *fastqc.zip --ignore *.html</span><br></pre></td></tr></table></figure>
<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/1qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/2qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/3qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/4qc.png"/>

<img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/5qc.png"/>


<ul>
<li>1、帮忙分析的数据，从公司给的结果的样本名中包含“clean”，我猜想公司大概已经进行过一次过滤，而且也完全没有接头。</li>
<li>2、另外，没有过关的这些数据也很难通过过滤得到改善（因为我使用trim_galore过滤，质控结果更差了）</li>
<li><strong>3、但是我还是不死心呀，我分别取了一个原始的和自己过滤的样本的前10000条reads,进行比对。结果发现原始的比对率可以达到95%+，自己过滤后的样本比对率比较低。</strong></li>
<li>4、所以，这批数据，公司应该已经进行过滤操作了。</li>
</ul>
<h2 id="（二）比对"><a href="#（二）比对" class="headerlink" title="（二）比对"></a>（二）比对</h2><p><strong>比对软件：hisat2</strong></p>
<p><font color=blue>不同数据类型所需要使用的比对软件也是有讲究的呢，可千万不能乱使用，会出问题的，刚开始接触数据分析时，知道有很多比对软件，但一直用bowtie2进行比对，后面就翻车啦～～～～～～</font></p>
<p>比对软件的选择：<br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/849f8ada0ab7">https://www.jianshu.com/p/849f8ada0ab7</a><br><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/681e02e7f9af">https://www.jianshu.com/p/681e02e7f9af</a><br><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/26506787">https://zhuanlan.zhihu.com/p/26506787</a> </p>
<p><font color=blue>NGS—-bwa<br>Chip（也包括ATAC）—-bowtie2<br>RNA—-hisat2（推荐）、star、Tophat、subjunc</font></p>
<h3 id="第一步：序列比对"><a href="#第一步：序列比对" class="headerlink" title="第一步：序列比对"></a>第一步：序列比对</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#cat sample</span></span><br><span class="line">KO1455_FRAS190065880-1a_1.clean.fq.gz	KO1455_FRAS190065880-1a_2.clean.fq.gz</span><br><span class="line">KO1462_FRAS190065881-1a_1.clean.fq.gz	KO1462_FRAS190065881-1a_2.clean.fq.gz</span><br><span class="line">KO1471_FRAS190065882-1a_1.clean.fq.gz	KO1471_FRAS190065882-1a_2.clean.fq.gz</span><br><span class="line">WT1441_FRAS190065878-1a_1.clean.fq.gz	WT1441_FRAS190065878-1a_2.clean.fq.gz</span><br><span class="line">WT1465_FRAS190065879-1a_1.clean.fq.gz	WT1465_FRAS190065879-1a_2.clean.fq.gz</span><br><span class="line">WT1474_FRAS190065877-1a_1.clean.fq.gz	WT1474_FRAS190065877-1a_2.clean.fq.gz</span><br></pre></td></tr></table></figure>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#因为不想让本机太辛苦，要让它歇息一会儿,所以将KO和WT分批次跑</span></span><br><span class="line">cat sample | head -n 3 &gt; KO_sample</span><br><span class="line">cat sample | tail -n 3 &gt; WT_sample</span><br><span class="line">ref=/home/reference/mm10/genome</span><br><span class="line"></span><br><span class="line">cat KO_sample|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">hisat2 -p 7 -x <span class="variable">$ref</span> -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span> | samtools sort -@ 7 -o /home/align/<span class="variable">$sample</span> -</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">cat WT_sample|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">hisat2 -p 7 -x <span class="variable">$ref</span> -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span> | samtools sort -@ 7 -o /home/align/<span class="variable">$sample</span> -</span><br><span class="line"><span class="keyword">done</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#各样本比对率都在95%以上，这样的比对结果才可以比较有信心的进行后续分析了。</span></span><br></pre></td></tr></table></figure>
<h3 id="第二步：构建索引"><a href="#第二步：构建索引" class="headerlink" title="第二步：构建索引"></a>第二步：构建索引</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|xargs -i samtools index &#123;&#125;</span><br></pre></td></tr></table></figure>
<h3 id="第三部：生成状态文件-统计比对情况"><a href="#第三部：生成状态文件-统计比对情况" class="headerlink" title="第三部：生成状态文件(统计比对情况)"></a>第三部：生成状态文件(统计比对情况)</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|xargs -i samtools flagstat -@ 7 &#123;&#125;</span><br></pre></td></tr></table></figure>
<h2 id="（三）制作表达矩阵"><a href="#（三）制作表达矩阵" class="headerlink" title="（三）制作表达矩阵"></a>（三）制作表达矩阵</h2><p>参考资料：<a target="_blank" rel="noopener" href="https://www.bioinfo-scrounger.com/archives/407/">https://www.bioinfo-scrounger.com/archives/407/</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ref=/home/reference/gtf/gencode.v33.annotation.gtf  </span><br><span class="line">featureCounts -T 8 -p -t exon -g gene_id -a <span class="variable">$gtf</span> -o RNA-matrix.txt *.bam 1&gt;counts.id.log 2&gt;&amp;1</span><br></pre></td></tr></table></figure>
<p>查看featureCounts的情况</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat counts.id.log</span><br></pre></td></tr></table></figure>

<p>在此，要记录一个错误。第一次操作时误使用了人类的gtf文件，结果日志文件显示比对率在10%左右……使用小鼠gtf文件后，比对结果正常多了（因为这批数据是小鼠组织细胞测序的结果）。如下所示。<br><img src="https://gyq-images.oss-cn-hangzhou.aliyuncs.com/img/featureCounts.png"/></p>
<h1 id="（四）差异分析"><a href="#（四）差异分析" class="headerlink" title="（四）差异分析"></a>（四）差异分析</h1><h3 id="第一步：将reads矩阵导入R中"><a href="#第一步：将reads矩阵导入R中" class="headerlink" title="第一步：将reads矩阵导入R中"></a>第一步：将reads矩阵导入R中</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">rm(<span class="built_in">list</span> = ls())</span><br><span class="line">options(stringsAsFactors = <span class="built_in">F</span>)</span><br><span class="line">setwd(<span class="string">&quot;G:\\seq-analysis-data\\RNA-seq&quot;</span>)</span><br><span class="line">a=read.table(<span class="string">&#x27;RNA-matrix.txt&#x27;</span>,header = <span class="built_in">T</span>)</span><br><span class="line">ckeck_a=a[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">12</span>]</span><br></pre></td></tr></table></figure>
<h3 id="第二步：获取正确的Geneid"><a href="#第二步：获取正确的Geneid" class="headerlink" title="第二步：获取正确的Geneid"></a>第二步：获取正确的Geneid</h3><ul>
<li>首先看看矩阵中的Geneid长什么样</li>
</ul>
<img width="20%" height="20%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_geneid.JPG"/>


<ul>
<li>所以要去掉小数点及后面的数字，获取正确的Geneid<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(stringr)</span><br><span class="line"><span class="built_in">class</span>(str_split(a$Geneid,<span class="string">&#x27;[.]&#x27;</span>,simplify = <span class="built_in">T</span>))</span><br><span class="line">a$ensembl_id=str_split(a$Geneid,<span class="string">&#x27;[.]&#x27;</span>,simplify = <span class="built_in">T</span>)[,<span class="number">1</span>]</span><br><span class="line">rownames(a)&lt;-a$ensembl_id</span><br><span class="line">check_a_again=a[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">12</span>]</span><br><span class="line">exprSet&lt;-a[,<span class="number">7</span>:<span class="number">12</span>]</span><br><span class="line">ckeck_exprSet&lt;-head(exprSet)</span><br></pre></td></tr></table></figure>
表达矩阵现在的样子（Geneid名去掉了多余的部分）</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/raw_exprSet.JPG"/>


<h2 id="第三步：id转换"><a href="#第三步：id转换" class="headerlink" title="第三步：id转换"></a>第三步：id转换</h2><p>Geneid和symbol并不是完全一一对应的，存在部分多个Geneid对应同一个基因名的情况。</p>
<p>因为rownames必须唯一，所以就要去掉重复的Geneid或者只保留其中一个（这个保留就比较的人为了~）</p>
<ul>
<li><p>方法一：使用biomaRt包</p>
<p>此方法几乎可以转换全部的id(<font color=orange>推荐</font>)</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br></pre></td><td class="code"><pre><span class="line">library(biomaRt)</span><br><span class="line">library(curl)</span><br><span class="line"><span class="comment">#此步骤需要网速，网速不好会报错（curl相关的），多试几遍，会有好运气的</span></span><br><span class="line">mart &lt;- useDataset(<span class="string">&quot;mmusculus_gene_ensembl&quot;</span>, useMart(<span class="string">&quot;ensembl&quot;</span>))<span class="comment">##人类选择hsapiens_gene_ensembl</span></span><br><span class="line">gene&lt;-row.names(exprSet)<span class="comment">#要转换成SYMBOL的Geneid</span></span><br><span class="line">gene_name&lt;-getBM(<span class="built_in">attributes</span>=<span class="built_in">c</span>(<span class="string">&quot;description&quot;</span>,<span class="string">&quot;external_gene_name&quot;</span>,<span class="string">&quot;ensembl_gene_id&quot;</span>),</span><br><span class="line">                 filters = <span class="string">&quot;ensembl_gene_id&quot;</span>,</span><br><span class="line">                 values = gene, mart = mart)</span><br><span class="line">head(gene_name)</span><br><span class="line">exprSet&lt;-a[,<span class="number">7</span>:<span class="number">12</span>]</span><br><span class="line">head(exprSet)</span><br><span class="line">table(rownames(exprSet)%in%gene_name$ensembl_gene_id)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%gene_name$ensembl_gene_id,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">symbol&lt;-gene_name[match(rownames(exprSet),gene_name$ensembl_gene_id),]</span><br><span class="line">head(symbol)</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">6</span>,<span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">tmp&lt;-by(exprSet,symbol$external_gene_name,<span class="keyword">function</span>(x)rownames(x)[which.max(rowMeans(x))])</span><br><span class="line">probes&lt;-<span class="built_in">as.character</span>(tmp)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%probes,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">rownames(exprSet)&lt;-symbol[match(rownames(exprSet),symbol$ensembl_gene_id),<span class="number">2</span>]</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br><span class="line">write.csv(exprSet,<span class="string">&#x27;raw_exprSet.csv&#x27;</span>)</span><br><span class="line"><span class="comment">#exprSet&lt;-read.csv(&#x27;raw_exprSet.csv&#x27;)</span></span><br><span class="line"><span class="comment">#rownames(exprSet)&lt;-exprSet$X</span></span><br><span class="line"><span class="comment">#exprSet&lt;-exprSet[,-1]</span></span><br></pre></td></tr></table></figure></li>
<li><p>方法二：使用org.Mm.eg.db包<br>可能因为此包不怎么更新，id转换率只有50%左右，这样回丢掉很多信息的，可能会导致后续分析不可靠（不推荐）</p>
</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">library(clusterProfiler)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">symbol&lt;-bitr(rownames(exprSet),fromType = <span class="string">&quot;ENSEMBL&quot;</span>,</span><br><span class="line">             toType=<span class="built_in">c</span>(<span class="string">&quot;SYMBOL&quot;</span>,<span class="string">&quot;ENTREZID&quot;</span>),OrgDb = org.Mm.eg.db)</span><br><span class="line">colnames(symbol)&lt;-<span class="built_in">c</span>(<span class="string">&quot;ensembl_id&quot;</span>,<span class="string">&quot;SYMBOL&quot;</span>,<span class="string">&quot;ENTREZID&quot;</span>)</span><br><span class="line">check_symbol&lt;-symbol[<span class="number">1</span>:<span class="number">10</span>,<span class="number">1</span>:<span class="number">3</span>]</span><br><span class="line">table(rownames(exprSet)%in%symbol$ensembl_id)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%symbol$ensembl_id,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">symbol&lt;-symbol[match(rownames(exprSet),symbol$ensembl_id),]</span><br><span class="line">head(symbol)</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">6</span>,<span class="number">1</span>:<span class="number">6</span>]</span><br><span class="line">tmp&lt;-by(exprSet,symbol$SYMBOL,<span class="keyword">function</span>(x)rownames(x)[which.max(rowMeans(x))])</span><br><span class="line">probes&lt;-<span class="built_in">as.character</span>(tmp)</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">exprSet&lt;-exprSet[rownames(exprSet)%in%probes,]</span><br><span class="line"><span class="built_in">dim</span>(exprSet)</span><br><span class="line">rownames(exprSet)&lt;-symbol[match(rownames(exprSet),symbol$ensembl_id),<span class="number">2</span>]</span><br><span class="line">exprSet[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>]</span><br></pre></td></tr></table></figure>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/3a0e1e3e41d0">https://www.jianshu.com/p/3a0e1e3e41d0</a></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/c06fea33b60f">https://www.jianshu.com/p/c06fea33b60f</a></p>
<ul>
<li>在进行后续分析之前，检验目的基因的敲除效果<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Ang表达量探索</span></span><br><span class="line">Ang&lt;-raw_exprSet[rownames(raw_exprSet)==<span class="string">&quot;Ang&quot;</span>,]</span><br><span class="line">colnames(Ang)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">ang&lt;-as.data.frame(t(Ang))</span><br><span class="line">ang$Group&lt;-<span class="built_in">c</span>(<span class="built_in">rep</span>(<span class="string">&quot;KO&quot;</span>,<span class="number">3</span>),<span class="built_in">rep</span>(<span class="string">&quot;WT&quot;</span>,<span class="number">3</span>))</span><br><span class="line">ang$Sample&lt;-rownames(ang)</span><br><span class="line">colnames(ang)&lt;-<span class="built_in">c</span>(<span class="string">&quot;Counts&quot;</span>,<span class="string">&quot;Group&quot;</span>,<span class="string">&quot;Sample&quot;</span>)</span><br><span class="line">library(ggplot2)</span><br><span class="line">p&lt;-ggplot(ang,aes(Sample,Counts,group=Group))+</span><br><span class="line">  geom_bar(aes(fill=Group),stat=<span class="string">&#x27;identity&#x27;</span>)+</span><br><span class="line">  scale_fill_manual(values = <span class="built_in">c</span>(<span class="string">&quot;#FFA500&quot;</span>,<span class="string">&quot;#228B22&quot;</span>))+</span><br><span class="line">  ylab(<span class="string">&quot;The expression of Ang&quot;</span>)+</span><br><span class="line">  xlab(<span class="string">&quot;&quot;</span>)</span><br><span class="line">p+theme_bw()+theme(panel.border = element_blank(),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),axis.line = element_line(colour = <span class="string">&quot;black&quot;</span>))</span><br></pre></td></tr></table></figure>
由图可看出：目的基因敲除效果显著；另外，也可以一定程度上反应上游数据分析的可靠性（因为在做测序之前，一定是先实验验证过敲低效率的）。</li>
</ul>
<img width="50%" height="50%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ggplot2_ang_counts.jpeg"/>


<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/92473504">https://zhuanlan.zhihu.com/p/92473504</a></p>
<p><a target="_blank" rel="noopener" href="https://www.sohu.com/a/319083452_785442">https://www.sohu.com/a/319083452_785442</a></p>
<h2 id="第四步：DESeq2进行差异分析"><a href="#第四步：DESeq2进行差异分析" class="headerlink" title="第四步：DESeq2进行差异分析"></a>第四步：DESeq2进行差异分析</h2><p>1、DESeq2差异分析的主要步骤</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">library(DESeq2)</span><br><span class="line"><span class="comment">#DEseq2要求输入数据是由整数组成的矩阵。</span></span><br><span class="line"><span class="comment">#DESeq2要求矩阵是没有标准化的。</span></span><br><span class="line"><span class="comment">##构建一个dds对象</span></span><br><span class="line">exprSet_integer=<span class="built_in">round</span>(exprSet)  </span><br><span class="line">(colData &lt;- data.frame(row.names=colnames(exprSet_integer), </span><br><span class="line">                       group_list=group_list) )</span><br><span class="line">dds &lt;- DESeqDataSetFromMatrix(countData = exprSet_integer,</span><br><span class="line">                              colData = colData,</span><br><span class="line">                              design = ~ group_list)</span><br><span class="line">dds2 &lt;- DESeq(dds)<span class="comment">##使用DESeq函数对dds进行normalize</span></span><br><span class="line">resultsNames(dds2)<span class="comment">##查看结果的名称 </span></span><br><span class="line"></span><br><span class="line">res &lt;- results(dds2, contrast=<span class="built_in">c</span>(<span class="string">&quot;group_list&quot;</span>,<span class="string">&quot;KO&quot;</span>,<span class="string">&quot;WT&quot;</span>))</span><br><span class="line"><span class="comment">##提取想要的差异分析结果，这里是&quot;KO&quot;组对&quot;WT&quot;组进行比较。</span></span><br><span class="line">resOrdered &lt;- res[order(res$padj),]<span class="comment">##按padj从小到大排序</span></span><br><span class="line">summary(res)</span><br><span class="line">DEG&lt;-data.frame(resOrdered)<span class="comment">##得到差异分析的data.frame</span></span><br><span class="line">head(DEG)</span><br><span class="line">DEG = na.omit(DEG)</span><br><span class="line">write.csv(DEG,<span class="string">&#x27;DESeq2_KO_VS_WT DEG.csv&#x27;</span>)</span><br><span class="line"><span class="comment">#DEG&lt;-read.csv(&quot;DESeq2_KO_VS_WT DEG.csv&quot;)</span></span><br><span class="line"><span class="comment">#rownames(DEG)&lt;-DEG$X</span></span><br><span class="line"><span class="comment">#DEG&lt;-DEG[,-1]</span></span><br></pre></td></tr></table></figure>
<p>2、DESeq2标准化前后的对比</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#数据离散度非常大的RNA-seq的基因的reads的counts矩阵经过normalization后</span></span><br><span class="line"><span class="comment">#变成了类似于芯片表达矩阵的表达矩阵。生成的2行2列的图可以看出normalization前后的变化</span></span><br><span class="line">rld&lt;-rlogTransformation(dds2)<span class="comment">##得到经过DESeq2软件normalization的表达矩阵。</span></span><br><span class="line">exprSet_normalized&lt;-assay(rld)</span><br><span class="line">par(cex = <span class="number">0.7</span>)</span><br><span class="line">n.sample=ncol(exprSet)</span><br><span class="line"><span class="keyword">if</span>(n.sample&gt;<span class="number">40</span>) par(cex = <span class="number">0.5</span>)</span><br><span class="line">cols &lt;- rainbow(n.sample*<span class="number">1.2</span>)</span><br><span class="line">par(mfrow=<span class="built_in">c</span>(<span class="number">2</span>,<span class="number">2</span>))</span><br><span class="line">colnames(exprSet_integer)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">colnames(exprSet_normalized)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">boxplot(exprSet_integer, col = cols,main=<span class="string">&quot;expression value&quot;</span>,las=<span class="number">2</span>)</span><br><span class="line">boxplot(exprSet_normalized, col = cols,main=<span class="string">&quot;expression value&quot;</span>,las=<span class="number">2</span>)</span><br><span class="line">hist(as.matrix(exprSet))</span><br><span class="line">hist(exprSet_normalized)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/normalization.jpeg"/>

<p><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/30350531">https://zhuanlan.zhihu.com/p/30350531</a></p>
<h2 id="第五步：可视化-……"><a href="#第五步：可视化-……" class="headerlink" title="第五步：可视化 ……"></a>第五步：可视化 ……</h2><ul>
<li>富集分析（可以使用KEGG、GO，这里使用GO）<br>大致看一下受目的基因影响的信号通路<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#1、差异基因的提取</span></span><br><span class="line">topT &lt;- as.data.frame(DEG)</span><br><span class="line">logFC_Cutoff&lt;-with(topT,mean(<span class="built_in">abs</span>(log2FoldChange))+<span class="number">2</span>*sd(<span class="built_in">abs</span>(log2FoldChange)))</span><br><span class="line">logFC_cutoff&lt;-logFC_Cutoff</span><br><span class="line">select_gene=(<span class="built_in">abs</span>(DEG$log2FoldChange)&gt;=logFC_cutoff)&amp;(DEG$pvalue&lt;<span class="number">0.05</span>)</span><br><span class="line">select_gene.sig=DEG[select_gene,]</span><br><span class="line">write.csv(select_gene.sig,<span class="string">&#x27;DESeq2_select_gene.sig.csv&#x27;</span>)</span><br><span class="line">gene=rownames(select_gene.sig)</span><br><span class="line"><span class="comment">#2、差异基因的通路富集</span></span><br><span class="line">library(clusterProfiler)</span><br><span class="line">library(topGO)</span><br><span class="line">library(Rgraphviz)</span><br><span class="line">library(pathview)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">DEG.entrez_id = mapIds(x = org.Mm.eg.db,</span><br><span class="line">                       keys = gene,</span><br><span class="line">                       keytype = <span class="string">&quot;SYMBOL&quot;</span>,</span><br><span class="line">                       column = <span class="string">&quot;ENTREZID&quot;</span>)</span><br><span class="line">DEG.entrez_id = na.omit(DEG.entrez_id)</span><br><span class="line">erich.go.BP = enrichGO(gene = DEG.entrez_id,</span><br><span class="line">                       OrgDb = org.Mm.eg.db,</span><br><span class="line">                       keyType = <span class="string">&quot;ENTREZID&quot;</span>,</span><br><span class="line">                       ont = <span class="string">&quot;BP&quot;</span>,</span><br><span class="line">                       pvalueCutoff = <span class="number">0.5</span>,</span><br><span class="line">                       qvalueCutoff = <span class="number">0.5</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">##分析完成后，作图</span></span><br><span class="line">dotplot(erich.go.BP)</span><br></pre></td></tr></table></figure>
<img width="80%" height="80%" src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GO.jpeg"/>

</li>
</ul>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/47b5ea646932?utm_source=desktop&amp;utm_medium=timeline">https://www.jianshu.com/p/47b5ea646932?utm_source=desktop&amp;utm_medium=timeline</a></p>
<ul>
<li>热图绘制<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">library(pheatmap)</span><br><span class="line">colnames(exprSet)&lt;-<span class="built_in">c</span>(<span class="string">&quot;KO1455&quot;</span>,<span class="string">&quot;KO1462&quot;</span>,<span class="string">&quot;KO1471&quot;</span>,<span class="string">&quot;WT1441&quot;</span>,<span class="string">&quot;WT1465&quot;</span>,<span class="string">&quot;WT1474&quot;</span>)</span><br><span class="line">choose_upgene=<span class="built_in">names</span>(sort(apply(up_gene, <span class="number">1</span>, mad),decreasing = <span class="built_in">T</span>)[<span class="number">1</span>:<span class="number">40</span>])</span><br><span class="line">choose_upgene_matrix=exprSet[choose_upgene,]</span><br><span class="line">choose_downgene=<span class="built_in">names</span>(sort(apply(down_gene, <span class="number">1</span>, mad),decreasing = <span class="built_in">T</span>)[<span class="number">1</span>:<span class="number">40</span>])</span><br><span class="line">choose_downgene_matrix=exprSet[choose_downgene,]</span><br><span class="line">choose_matrix=rbind(choose_upgene_matrix,choose_downgene_matrix)</span><br><span class="line">choose_matrix=t(scale(t(choose_matrix)))</span><br><span class="line">annotation_col = data.frame(CellType = factor(<span class="built_in">rep</span>(<span class="built_in">c</span>(<span class="string">&quot;KO&quot;</span>, <span class="string">&quot;WT&quot;</span>), each=<span class="number">3</span>)),Time = <span class="number">1</span>:<span class="number">3</span>)</span><br><span class="line">rownames(annotation_col)&lt;-colnames(exprSet)</span><br><span class="line">pheatmap(choose_matrix,fontsize = <span class="number">6</span>,cexCol=<span class="number">1</span>,cellwidth=<span class="number">10</span>,cellheigh=<span class="number">5</span>,color = colorRampPalette(<span class="built_in">c</span>(<span class="string">&#x27;navy&#x27;</span>,<span class="string">&#x27;white&#x27;</span>,<span class="string">&#x27;firebrick3&#x27;</span>))(<span class="number">50</span>),annotation_col=annotation_col)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/heatmap.jpeg"/>

</li>
</ul>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/23/RNA-seq%20analysis/" data-id="ckl4ra1lq0006agvf2uoj5s64" data-title="RNA-seq数据分析流程" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-R语言apply、lapply、strsplit、%in%用法介绍" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/23/R%EF%BF%BD%00apply%01lapply%01strsplit%01%25in%25(%EF%BF%BD%EF%BF%BD%EF%BF%BD/" class="article-date">
  <time class="dt-published" datetime="2020-02-23T08:30:23.000Z" itemprop="datePublished">2020-02-23</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/23/R%EF%BF%BD%00apply%01lapply%01strsplit%01%25in%25(%EF%BF%BD%EF%BF%BD%EF%BF%BD/">R语言apply、lapply、strsplit、%in%用法介绍</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>生信分析下游过程中，为了构建好可用于个性化分析的数据，往往要对原有数据进行整理，这个时候<code>apply</code>、<code>lapply</code>、<code>strsplit</code>、<code>%in%</code>就会在不同的地方发光✨发热🔥啦</p>
        
          <p class="article-more-link">
            <a href="/2020/02/23/R%EF%BF%BD%00apply%01lapply%01strsplit%01%25in%25(%EF%BF%BD%EF%BF%BD%EF%BF%BD/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/23/R%E8%AF%AD%E8%A8%80apply%E3%80%81lapply%E3%80%81strsplit%E3%80%81%in%%E7%94%A8%E6%B3%95%E4%BB%8B%E7%BB%8D/" data-id="ckl4ra1lt000aagvfa0111p2b" data-title="R语言apply、lapply、strsplit、%in%用法介绍" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R/" rel="tag">R</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-scRNA-seq数据分析实战（三）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/22/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%89%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-22T08:30:23.000Z" itemprop="datePublished">2020-02-22</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/22/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%89%EF%BC%89/">scRNA-seq数据分析实战（三）</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>single cell RNA-seq  &amp;&amp; 10×<br>利用Seurate 2.3.3完成单细胞转录组下游分析，重现文章《Acquired cancer resistance to combination immunotherapy from transcriptional loss of class I HLA》的部分图片。</p>
        
          <p class="article-more-link">
            <a href="/2020/02/22/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%89%EF%BC%89/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/22/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AE%9E%E6%88%98%EF%BC%88%E4%B8%89%EF%BC%89/" data-id="ckl4ra1m7000oagvfdwjc5g6p" data-title="scRNA-seq数据分析实战（三）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-特定版本R包下载及其报错" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/16/%E7%89%B9%E5%AE%9A%E7%89%88%E6%9C%ACR%E5%8C%85%E4%B8%8B%E8%BD%BD%E5%8F%8A%E5%85%B6%E6%8A%A5%E9%94%99/" class="article-date">
  <time class="dt-published" datetime="2020-02-16T08:30:23.000Z" itemprop="datePublished">2020-02-16</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/16/%E7%89%B9%E5%AE%9A%E7%89%88%E6%9C%ACR%E5%8C%85%E4%B8%8B%E8%BD%BD%E5%8F%8A%E5%85%B6%E6%8A%A5%E9%94%99/">特定版本R包下载及其报错</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>Error: Failed to install ‘unknown package’ from URL:<br>(converted from warning) installation of package ‘dtw’ had non-zero exit status   </p>
        
          <p class="article-more-link">
            <a href="/2020/02/16/%E7%89%B9%E5%AE%9A%E7%89%88%E6%9C%ACR%E5%8C%85%E4%B8%8B%E8%BD%BD%E5%8F%8A%E5%85%B6%E6%8A%A5%E9%94%99/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/16/%E7%89%B9%E5%AE%9A%E7%89%88%E6%9C%ACR%E5%8C%85%E4%B8%8B%E8%BD%BD%E5%8F%8A%E5%85%B6%E6%8A%A5%E9%94%99/" data-id="ckl4ra1mi0014agvf0jjj63to" data-title="特定版本R包下载及其报错" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/R/" rel="tag">R</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-scRNA-seq数据分析学习（三）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-15T08:30:23.000Z" itemprop="datePublished">2020-02-15</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/">scRNA-seq数据分析学习（三）</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>学习R包seurat对scRNA-seq数据进行分析</p>
        
          <p class="article-more-link">
            <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/" data-id="ckl4ra1m2000kagvf0atka0m0" data-title="scRNA-seq数据分析学习（三）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-scRNA-seq数据分析学习（四）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-15T08:30:23.000Z" itemprop="datePublished">2020-02-15</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/">scRNA-seq数据分析学习（四）</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>学习R包monocle对scRNA-seq数据进行分析</p>
        
          <p class="article-more-link">
            <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/" data-id="ckl4ra1m5000nagvf6ksr541k" data-title="scRNA-seq数据分析学习（四）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  
    <article id="post-scRNA-seq数据分析学习（二）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/14/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%BA%8C%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-14T08:30:23.000Z" itemprop="datePublished">2020-02-14</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="p-name article-title" href="/2020/02/14/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%BA%8C%EF%BC%89/">scRNA-seq数据分析学习（二）</a>
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>学习R包scater对scRNA-seq数据进行分析</p>
        
          <p class="article-more-link">
            <a href="/2020/02/14/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%BA%8C%EF%BC%89/#more">Read More</a>
          </p>
        
      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/14/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%BA%8C%EF%BC%89/" data-id="ckl4ra1m4000magvfbww142ix" data-title="scRNA-seq数据分析学习（二）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
</article>



  


  <nav id="page-nav">
    
    <a class="extend prev" rel="prev" href="/">&laquo; Prev</a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><a class="extend next" rel="next" href="/page/3/">Next &raquo;</a>
  </nav>

</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/System/" rel="tag">System</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/notes/" rel="tag">notes</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Linux/" style="font-size: 16.67px;">Linux</a> <a href="/tags/MySQL/" style="font-size: 13.33px;">MySQL</a> <a href="/tags/NGS/" style="font-size: 20px;">NGS</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/System/" style="font-size: 10px;">System</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/notes/" style="font-size: 10px;">notes</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/01/">January 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">December 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/05/">May 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/04/">April 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/03/">March 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/02/">February 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
          </li>
        
          <li>
            <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
          </li>
        
          <li>
            <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
          </li>
        
          <li>
            <a href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
          </li>
        
          <li>
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGS数据分析之表观</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2021 gongyuqiqi<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>