<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.3.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","images":"/images","scheme":"Pisces","version":"8.2.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"/search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>
<meta name="description" content="RNA-seq高级分析之可变剪切">
<meta property="og:type" content="article">
<meta property="og:title" content="可变剪切分析之MISO">
<meta property="og:url" content="http://example.com/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="RNA-seq高级分析之可变剪切">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align2.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align4.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/unzipgff.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ESgff3index.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/miso_run_out.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filter.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sashimi_plot1.JPG">
<meta property="article:published_time" content="2021-01-09T06:05:52.000Z">
<meta property="article:modified_time" content="2021-02-14T07:37:43.609Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG">


<link rel="canonical" href="http://example.com/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>
<title>可变剪切分析之MISO | gyq's blog</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">gyq's blog</h1>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li>
        <li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li>
        <li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E5%88%86%E6%9E%90%E4%B9%8BMISO"><span class="nav-number">1.</span> <span class="nav-text">可变剪切分析之MISO</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="nav-number">1.1.</span> <span class="nav-text">数据下载</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9"><span class="nav-number">1.2.</span> <span class="nav-text">比对</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9%E6%96%B9%E6%B3%95%E4%B8%80"><span class="nav-number">1.2.0.0.1.</span> <span class="nav-text">比对方法一</span></a></li><li class="nav-item nav-level-5"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9%E6%96%B9%E6%B3%95%E4%BA%8C"><span class="nav-number">1.2.0.0.2.</span> <span class="nav-text">比对方法二</span></a></li></ol></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E8%BF%90%E8%A1%8CMISO"><span class="nav-number">1.3.</span> <span class="nav-text">运行MISO</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="gongyuqiqi"
      src="/assets/img/ava.jpg">
  <p class="site-author-name" itemprop="name">gongyuqiqi</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">28</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/assets/img/ava.jpg">
      <meta itemprop="name" content="gongyuqiqi">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="gyq's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          可变剪切分析之MISO
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2021-01-09 14:05:52" itemprop="dateCreated datePublished" datetime="2021-01-09T14:05:52+08:00">2021-01-09</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2021-02-14 15:37:43" itemprop="dateModified" datetime="2021-02-14T15:37:43+08:00">2021-02-14</time>
      </span>

  
      </div>
      <div class="post-meta">
    <span class="post-meta-item" title="Symbols count in article">
      <span class="post-meta-item-icon">
        <i class="far fa-file-word"></i>
      </span>
      <span class="post-meta-item-text">Symbols count in article: </span>
      <span>5.6k</span>
    </span>
    <span class="post-meta-item" title="Reading time">
      <span class="post-meta-item-icon">
        <i class="far fa-clock"></i>
      </span>
      <span class="post-meta-item-text">Reading time &asymp;</span>
      <span>5 mins.</span>
    </span>
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <p>RNA-seq高级分析之可变剪切</p>
<a id="more"></a>

<h1 id="可变剪切分析之MISO"><a href="#可变剪切分析之MISO" class="headerlink" title="可变剪切分析之MISO"></a>可变剪切分析之MISO</h1><p>MISO软件原文档：<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso">https://miso.readthedocs.io/en/fastmiso/index.html#installing-fastmiso</a></p>
<h2 id="数据下载"><a href="#数据下载" class="headerlink" title="数据下载"></a>数据下载</h2><ul>
<li>写一个bash脚本下载数据<br>一次性下载所有的.fastq.gz文件，本次实验数据是单端测序数据<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#！/bin/bash</span></span><br><span class="line">dir=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line"><span class="keyword">for</span> id <span class="keyword">in</span>  &#123;84,78,76,72,82,80,74,68&#125; </span><br><span class="line"><span class="keyword">do</span> </span><br><span class="line">       nohup ascp -QT -l 300m -P33001 -i <span class="variable">$dsa</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR974/SRR9749<span class="variable">$id</span>/SRR9749<span class="variable">$id</span>.fastq.gz . &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>执行脚本</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">chmod +x download.sh</span><br><span class="line">nohup ./download.sh &amp;</span><br></pre></td></tr></table></figure>
<h2 id="比对"><a href="#比对" class="headerlink" title="比对"></a>比对</h2><p>参考资料：<a target="_blank" rel="noopener" href="http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml">http://www.360doc.com/content/18/0714/20/19913717_770401548.shtml</a></p>
<blockquote>
<ul>
<li>tophat进行转录组的比对时，输出的是bam（二进制的sam）文件。</li>
<li>tophat依赖bowtie2（bowtie1也可以），参考基因组索引用bowtie2-build进行构建。</li>
<li>如果FASTA格式的基因组与索引文件不在同一个文件夹下面，tophat会在运行中自动生成，所以记得把参考基因组和索引文件放在同一个文件夹下面，以减少系统运行时的高消耗。</li>
<li>如果GTF或者GFF格式的基因组注释文件存在， tophat会优先比对到注释文件的转录组上。最好事先准备好转录组索引以节约后续的比对时间。</li>
<li>如果没有事先准备好转录组索引，tophat会在运行过程中生成转录组索引，这样就比较耗时耗力了。</li>
</ul>
</blockquote>
<p><strong><table><tr><td bgcolor=GreenYellow>**参考基因组索引构建、转录组索引构建</strong></td></tr></table>**</p>
<p>hg19参考基因组gtf文件下载链接：<a target="_blank" rel="noopener" href="https://www.gencodegenes.org/human/release_36lift37.html">https://www.gencodegenes.org/human/release_36lift37.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#建立hg19参考基因组索引文件</span></span><br><span class="line">nohup time bowtie2-build path/to/hg19.fa ./hg19 &gt; hg19.bowtie_index.log 2&gt;&amp;1 &amp;</span><br><span class="line"><span class="comment">#建立hg19转录组索引文件</span></span><br><span class="line"><span class="comment">#../index/bowtie/hg19为上一步构建的hg19参考基因组索引文件的相对路径及前缀</span></span><br><span class="line">nohup tophat -G ../gencode.v36lift37.annotation.gtf --transcriptome-index=hg19.tr ../index/bowtie/hg19 &amp;</span><br></pre></td></tr></table></figure>

<p><strong><table><tr><td bgcolor=GreenYellow>**参考基因组的比对</strong></td></tr></table>** </p>
<h5 id="比对方法一"><a href="#比对方法一" class="headerlink" title="比对方法一"></a>比对方法一</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#以其中一个样本为例，走一波比对转录组流程。</span></span><br><span class="line"><span class="comment">#理论上可以节约时间，但是实际跑起来非常耗时！</span></span><br><span class="line"><span class="comment">#1G左右得样本跑了4个小时都没有跑完！！！不知道问题出在那儿，但是还是果断kill掉</span></span><br><span class="line">tr_index=/home/gongyuqi/ref/hg19/transcriptome-index/hg19.tr</span><br><span class="line">gene_index=~/ref/hg19/index/bowtie/hg19</span><br><span class="line">tophat2 -o ../aligndata/SRR974980.out --transcriptome-index=<span class="variable">$tr</span> -p 30 --phred64-quals <span class="variable">$gene_index</span> SRR974980.fastq.gz</span><br></pre></td></tr></table></figure>
<h5 id="比对方法二"><a href="#比对方法二" class="headerlink" title="比对方法二"></a>比对方法二</h5><ul>
<li>写一个比对的bash脚本文件（align.txt），如下<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line">ls *.gz|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup tophat -G ~/ref/hg19/gencode.v36lift37.annotation.gtf -p 30 -o ../aligndata/<span class="variable">$id</span>.out ~/ref/hg19/index/bowtie/hg19 <span class="variable">$id</span> &amp;</span><br><span class="line"><span class="keyword">done</span> </span><br></pre></td></tr></table></figure></li>
<li>执行这个脚本，然后去睡觉，早上醒来看看比对得结果如何~<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup bash align.txt &amp;</span><br></pre></td></tr></table></figure>
运行结果<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align1.JPG"/>


</li>
</ul>
<p>每个文件夹打开结果如下<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align2.JPG"/></p>
<p>大概了解一下样本的运行日志</p>
<blockquote>
<p>注意reads的长度——36。miso后续分析中需要知道这个值<br>注意其中有一个error,至于为什么有这个报错，先不追究<br>注意比对结果的那个统计文件align_summary.txt,查看这个文件发现各个样本比对情况都挺好的。这就是为什么这里暂时不追究那个报错<br>最后会统计样本比对时长<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/align4.JPG"/></p>
</blockquote>
<p>3、bam文件排序及索引文件建立</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools sort -@ 30 -m 8G -O bam <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.accepted_hits.sorted.bam;<span class="keyword">done</span></span><br><span class="line">ls *sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> nohup samtools index -@ 30 <span class="variable">$id</span> &amp; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h2 id="运行MISO"><a href="#运行MISO" class="headerlink" title="运行MISO"></a>运行MISO</h2><p><strong><table><tr><td bgcolor=GreenYellow>**下载hg19的GFF文件并构建索引</strong></td></tr></table>**</p>
<p>具体下载网页及详情：<a target="_blank" rel="noopener" href="https://miso.readthedocs.io/en/fastmiso/annotation.html">https://miso.readthedocs.io/en/fastmiso/annotation.html</a></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/annotation/human/hg19</span><br><span class="line">wget hollywood.mit.edu/burgelab/miso/annotations/ver2/miso_annotations_hg19_v2.zip</span><br><span class="line">unzip miso_annotations_hg19_v2.zip</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/unzipgff.JPG"/>

<p>这里我们以外显子的gff3为例构建索引</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">index_gff --index SE.hg19.gff3 ./index</span><br></pre></td></tr></table></figure>
<p>结果如下<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/ESgff3index.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow>**以单端测序样本为例运行miso</strong></td></tr></table>**</p>
<p>这一步生成的结果是很关键的，会计算alternative splicing events的PSI值。而这个在最后可视化的阶段会以posterior distributions over Ψ(PSI)的形式呈现——即sashimi_plot图最右边一列的柱状图。</p>
<ul>
<li>写一个miso_run的脚本，内容如下<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">#!/bin/bash</span></span><br><span class="line"><span class="built_in">cd</span> /home/gongyuqi/project/AS/aligndata/accpted_bam</span><br><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line">out_dir=/home/gongyuqi/project/AS/miso_step1</span><br><span class="line">ls *.sorted.bam|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup miso --run <span class="variable">$index_db</span> <span class="variable">$id</span> --output-dir <span class="variable">$out_dir</span>/<span class="variable">$&#123;id%%.*&#125;</span> --read-len 36 &amp;</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure></li>
<li>运行脚本<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup ./miso_run.sh &amp;</span><br></pre></td></tr></table></figure>
举例查看其中一个样本的运行结果</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/miso_run_out.JPG"/>


<table><tr><td bgcolor=GreenYellow>**样本间比较**</td></tr></table>

<p>这个软件最大的问题在于只能进行两个样本的比较，这一点就很气！！！<br>气归气，还是先运行一下，至少先把流程走完，先学起来。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">compare_miso --compare-samples SRR974978/ SRR974982/ comparison/</span><br></pre></td></tr></table></figure>

<p><strong><table><tr><td bgcolor=GreenYellow>**过滤</strong></td></tr></table>**</p>
<p>设置各种参数的阈值，目的是筛选差异显著的alternative splicing events。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">filter_events \</span><br><span class="line">--filter  SRR974978_vs_SRR974982.miso_bf \</span><br><span class="line">--num-inc 1 \</span><br><span class="line">--num-exc 1 \</span><br><span class="line">--num-sum-inc-exc 10 \</span><br><span class="line">--delta-psi 0.20 \</span><br><span class="line">--bayes-factor 10 \</span><br><span class="line">--output-dir ./filter_out</span><br></pre></td></tr></table></figure>
<p>过滤前后.misoz_bf文件中的alternative splicing events有明显减少<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/filter.JPG"/></p>
<p><strong><table><tr><td bgcolor=GreenYellow>**可视化</strong></td></tr></table>**</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">index_db=/home/gongyuqi/project/AS/annotation/human/hg19/index/</span><br><span class="line"><span class="comment">#下面做演示的剪切事件是从上一步过滤后的.miso_bf.filtered文件中提取的一个</span></span><br><span class="line">sashimi_plot \</span><br><span class="line">--plot-event <span class="string">&quot;chr20:1093906:1094052:+@chr20:1099396:1099545:+@chr20:1106141:1106293:+&quot;</span> \</span><br><span class="line"><span class="variable">$index_db</span> \</span><br><span class="line">sashimi_plot_settings.txt  \</span><br><span class="line">--output-dir ./sashimi_plot</span><br></pre></td></tr></table></figure>

<p>其中sashimi_plot_settings.txt是配置文件，内容如下。当然，也可以根据自己的作图需要修改相应的参数</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br></pre></td><td class="code"><pre><span class="line">[data]</span><br><span class="line"><span class="comment"># directory where BAM files are</span></span><br><span class="line">bam_prefix = /home/gongyuqi/project/AS/aligndata/accpted_bam/</span><br><span class="line"><span class="comment"># directory where MISO output is</span></span><br><span class="line">miso_prefix = /home/gongyuqi/project/AS/miso_step1</span><br><span class="line"></span><br><span class="line">bam_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974.accepted_hits.sorted.bam&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976.accepted_hits.sorted.bam&quot;</span>]</span><br><span class="line"></span><br><span class="line">miso_files = [</span><br><span class="line">    <span class="string">&quot;SRR974978&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974984&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974968&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974972&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974982&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974980&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974974&quot;</span>,</span><br><span class="line">    <span class="string">&quot;SRR974976&quot;</span>]</span><br><span class="line"></span><br><span class="line">[plotting]</span><br><span class="line"><span class="comment"># Dimensions of figure to be plotted (in inches)</span></span><br><span class="line">fig_width = 4</span><br><span class="line">fig_height = 10</span><br><span class="line"><span class="comment"># Factor to scale down introns and exons by</span></span><br><span class="line">intron_scale = 30</span><br><span class="line">exon_scale = 4</span><br><span class="line"><span class="comment"># Whether to use a log scale or not when plotting</span></span><br><span class="line">logged = False</span><br><span class="line">font_size = 6</span><br><span class="line"></span><br><span class="line"><span class="comment"># Max y-axis</span></span><br><span class="line">ymax = 150</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot posterior distributions inferred by MISO</span></span><br><span class="line">show_posteriors = True</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to show posterior distributions as bar summaries</span></span><br><span class="line">bar_posteriors = False</span><br><span class="line"></span><br><span class="line"><span class="comment"># Whether to plot the number of reads in each junction</span></span><br><span class="line">number_junctions = True</span><br><span class="line"></span><br><span class="line">resolution = .5</span><br><span class="line">posterior_bins = 40</span><br><span class="line">gene_posterior_ratio = 5</span><br><span class="line"></span><br><span class="line"><span class="comment"># List of colors for read denisites of each sample</span></span><br><span class="line">colors = [<span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#CC0011&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>,</span><br><span class="line">          <span class="string">&quot;#FF8800&quot;</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Number of mapped reads in each sample</span></span><br><span class="line"><span class="comment"># (Used to normalize the read density for RPKM calculation)</span></span><br><span class="line">coverages = [27232306,</span><br><span class="line">             16799279,</span><br><span class="line">             31298963,</span><br><span class="line">             26308550,</span><br><span class="line">             27067722,</span><br><span class="line">             17372484,</span><br><span class="line">             26095030,</span><br><span class="line">             26980460]</span><br><span class="line"></span><br><span class="line"><span class="comment"># Bar color for Bayes factor distribution</span></span><br><span class="line"><span class="comment"># plots (--plot-bf-dist)</span></span><br><span class="line"><span class="comment"># Paint them blue</span></span><br><span class="line">bar_color = <span class="string">&quot;b&quot;</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Bayes factors thresholds to use for --plot-bf-dist</span></span><br><span class="line">bf_thresholds = [0, 1, 2, 5, 10, 20]</span><br></pre></td></tr></table></figure>
<p>可视化结果如下<br>因为我们选取的剪切事件是SRR974978和SRR974982之间的差异剪切事件，所以在多个样本间趋势不是很明显。但这里我们先学个方法。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/sashimi_plot1.JPG"/></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/NGS/" rel="tag"># NGS</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/12/08/2020%20ATAC-seq%20analysis/" rel="prev" title="2020 ATAC-seq analysis">
                  <i class="fa fa-chevron-left"></i> 2020 ATAC-seq analysis
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/" rel="next" title="可变剪切之rMART">
                  可变剪切之rMART <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>







<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>
</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">gongyuqiqi</span>
</div>
<div class="wordcount">
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-line"></i>
    </span>
      <span>Symbols count total: </span>
    <span title="Symbols count total">136k</span>
  </span>
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
      <span>Reading time total &asymp;</span>
    <span title="Reading time total">2:03</span>
  </span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a>
  </div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  
<script src="/js/local-search.js"></script>






  





</body>
</html>
