<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  
  
  <title>scRNA-seq数据分析学习（三） | gyq&#39;s blog</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="学习R包seurat对scRNA-seq数据进行分析">
<meta property="og:type" content="article">
<meta property="og:title" content="scRNA-seq数据分析学习（三）">
<meta property="og:url" content="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="学习R包seurat对scRNA-seq数据进行分析">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/identicalmetacounts.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/seu1.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mitoinfo.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/VlnPlot.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GenePlotnUMInGene.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tailsortgene.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GenePlotnUMInGene.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CellPlot.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/normalizetruefalse.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/fvg.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/lengthfvg.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcareduction.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcainfo.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pca1pca2.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pca1pca2BC.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PCA1heatmap.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcaresulution.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tsneafterpca.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/grouppcaBC.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/headmarker_df.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/VlnPlotcluster1marker.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FeaturePlotcluster1marker.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/clustersheatmap.JPG">
<meta property="article:published_time" content="2020-02-15T08:30:23.000Z">
<meta property="article:modified_time" content="2021-02-14T06:34:09.032Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/identicalmetacounts.JPG">
  
    <link rel="alternate" href="/atom.xml" title="gyq's blog" type="application/atom+xml">
  
  
    <link rel="shortcut icon" href="/favicon.png">
  
  
    
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/typeface-source-code-pro@0.0.71/index.min.css">

  
  
<link rel="stylesheet" href="/css/style.css">

  
    
<link rel="stylesheet" href="/fancybox/jquery.fancybox.min.css">

  
<meta name="generator" content="Hexo 5.3.0"></head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">gyq&#39;s blog</a>
      </h1>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://example.com"></form>
      </div>
    </div>
  </div>
</header>

      <div class="outer">
        <section id="main"><article id="post-scRNA-seq数据分析学习（三）" class="h-entry article article-type-post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  <div class="article-meta">
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/" class="article-date">
  <time class="dt-published" datetime="2020-02-15T08:30:23.000Z" itemprop="datePublished">2020-02-15</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 class="p-name article-title" itemprop="headline name">
      scRNA-seq数据分析学习（三）
    </h1>
  

      </header>
    
    <div class="e-content article-entry" itemprop="articleBody">
      
        <p>学习R包seurat对scRNA-seq数据进行分析</p>
<a id="more"></a>

<h1 id="scRNA-seq数据分析（三）"><a href="#scRNA-seq数据分析（三）" class="headerlink" title="scRNA-seq数据分析（三）"></a>scRNA-seq数据分析（三）</h1><p>😊seurat不是一个R包，seurat是一个优秀的R包<br>😊seurat不是提供服务的，seurat是提供一条龙服务的   </p>
<ul>
<li>seurat版本有2.×.×和3.×.×。同一般的R包升级不太一样的是：2.×和3.×之间区别还是蛮多的，各种函数也都有变化，虽然升级带来了更多的优点，但是函数名称的变化就会给学习者带来不小的麻烦呀！</li>
<li>这里先学习2.×。为什么呢？因为我还不会3.×呀😵😏😜</li>
</ul>
<p><strong>载入R包</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">rm(<span class="built_in">list</span> = ls())<span class="comment">#clear the environment</span></span><br><span class="line">options(warn=-<span class="number">1</span>)<span class="comment">#turn off warning message globally</span></span><br><span class="line">library(Seurat)</span><br></pre></td></tr></table></figure>
<p><strong>同样使用scRNAseq内置数据集</strong>  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">library(scRNAseq)</span><br><span class="line">data(fluidigm)<span class="comment">#加载测试数据</span></span><br><span class="line">assay(fluidigm)&lt;-assays(fluidigm)$rsem_counts</span><br><span class="line">ct&lt;-<span class="built_in">floor</span>(assays(fluidigm)$rsem_counts)</span><br><span class="line">ct[<span class="number">1</span>:<span class="number">4</span>,<span class="number">1</span>:<span class="number">4</span>]</span><br><span class="line">counts&lt;-ct</span><br></pre></td></tr></table></figure>
<h2 id="创建Seurat要求的对象"><a href="#创建Seurat要求的对象" class="headerlink" title="创建Seurat要求的对象"></a>创建Seurat要求的对象</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">names</span>(metadata(fluidigm))</span><br><span class="line">meta&lt;-as.data.frame(colData(fluidigm))</span><br><span class="line">identical(rownames(meta),colnames(counts))<span class="comment">#检测meta和counts这两个对象，后面有需要</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/identicalmetacounts.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">seu&lt;-CreateSeuratObject(raw.data = counts,</span><br><span class="line">                            meta.data = meta,</span><br><span class="line">                            min.cells = <span class="number">3</span>,</span><br><span class="line">                            min.genes = <span class="number">200</span>,</span><br><span class="line">                            project = <span class="string">&quot;seu&quot;</span>)</span><br><span class="line">seu</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/seu1.JPG"/>


<p><strong>增加相关属性信息</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#增加线粒体基因信息，如果线粒体所占基因比例过高，意味着可能是死细胞</span></span><br><span class="line">mito.gene&lt;-grep(pattern = <span class="string">&quot;^MT-&quot;</span>,x=rownames(x=seu@data),value = <span class="literal">TRUE</span>)<span class="comment">#但是我们知道这个数据集里面并没有线粒体基因</span></span><br><span class="line">percent.mito&lt;-Matrix::colSums(seu@raw.data[mito.gene,])/Matrix::colSums(seu@raw.data)</span><br><span class="line">seu&lt;-AddMetaData(object = seu,metadata = percent.mito,col.name = <span class="string">&quot;percent.mito&quot;</span>)<span class="comment">#加入了线粒体基因的信息</span></span><br><span class="line"><span class="comment">#这里也可以加入ERCC等其他属性</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mitoinfo.JPG"/>


<h2 id="可视化（初步）"><a href="#可视化（初步）" class="headerlink" title="可视化（初步）"></a>可视化（初步）</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#可视化，meta信息里面存在分组变量，可以指定分组，scRNAseq数据集里的分组信息是&quot;Biological_Condition&quot;</span></span><br><span class="line">VlnPlot(object = seu,features.plot = <span class="built_in">c</span>(<span class="string">&quot;nGene&quot;</span>,<span class="string">&quot;nUMI&quot;</span>,<span class="string">&quot;percent.mito&quot;</span>),group.by = <span class="string">&#x27;Biological_Condition&#x27;</span>,nCol = <span class="number">3</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/VlnPlot.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#查看一下某两个属性间的相关性</span></span><br><span class="line">GenePlot(object=seu,gene1=<span class="string">&quot;nUMI&quot;</span>,gene2=<span class="string">&quot;nGene&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GenePlotnUMInGene.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#查看基因间的相关性</span></span><br><span class="line">tail(sort(Matrix::rowSums(seu@raw.data)))</span><br><span class="line">GenePlot(object = seu,gene1 = <span class="string">&quot;SOX11&quot;</span>,gene2 = <span class="string">&quot;EEF1A1&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tailsortgene.JPG"/>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/GenePlotnUMInGene.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#细胞间的相关性，更好的方式是用cor()+heatmap展示</span></span><br><span class="line">CellPlot(seu,seu@cell.names[<span class="number">3</span>],seu@cell.names[<span class="number">5</span>],do.identify=<span class="literal">FALSE</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/CellPlot.JPG"/>


<h2 id="表达矩阵归一化"><a href="#表达矩阵归一化" class="headerlink" title="表达矩阵归一化"></a>表达矩阵归一化</h2><p>只有进行归一化后，样本之间的比较才更能说明问题 </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">identical(seu@raw.data,seu@data)</span><br><span class="line">seu&lt;-NormalizeData(object = seu,normalization.method = <span class="string">&quot;LogNormalize&quot;</span>,</span><br><span class="line">                   scale.factor = <span class="number">10000</span>,display.progress = <span class="built_in">F</span>)</span><br><span class="line"><span class="comment">#将每个细胞中总UMI设定为10000，计算方法为loge（每个细胞中基因的nUMI/该细胞内总UMI*10000+1）</span></span><br><span class="line"><span class="comment">#经过归一化后，seu对象里面的data被改变了</span></span><br><span class="line">identical(seu@raw.data,seu@data)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/normalizetruefalse.JPG"/>


<h2 id="寻找波动比较明显的基因，后续使用这些差异基因进行分析，主要为了降低计算量"><a href="#寻找波动比较明显的基因，后续使用这些差异基因进行分析，主要为了降低计算量" class="headerlink" title="寻找波动比较明显的基因，后续使用这些差异基因进行分析，主要为了降低计算量"></a>寻找波动比较明显的基因，后续使用这些差异基因进行分析，主要为了降低计算量</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">seu&lt;-FindVariableGenes(object = seu,mean.function = ExpMean,dispersion.function = LogVMR,</span><br><span class="line">                       x.low.cutoff = <span class="number">0.0125</span>,</span><br><span class="line">                       y.high.cutoff = <span class="number">3</span>,</span><br><span class="line">                       y.cutoff = <span class="number">0.5</span>)</span><br><span class="line"><span class="comment">#选择不同的阈值，得到的基因取决于实际情况</span></span><br><span class="line"><span class="built_in">length</span>(seu@var.genes)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/fvg.JPG"/>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/lengthfvg.JPG"/>


<h2 id="对归一化后的矩阵进行分群"><a href="#对归一化后的矩阵进行分群" class="headerlink" title="对归一化后的矩阵进行分群"></a>对归一化后的矩阵进行分群</h2><ul>
<li>对矩阵进行回归建模，以及scale</li>
<li>center=T：每个细胞中基因表达量-该基因在所有细胞中的表达量</li>
<li>scale=T：每个细胞中基因中心化后的表达值/该基因所在所有细胞中表达值的标准差</li>
<li>注意：执行ScaleData之前需要先执行NormalizeData</li>
</ul>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">seu&lt;-ScaleData(object = seu,vars.to.regress = <span class="built_in">c</span>(<span class="string">&quot;nUMI&quot;</span>),display.progress = <span class="built_in">F</span>)<span class="comment">#在这里只去除了文库大小的影响</span></span><br></pre></td></tr></table></figure>
<h2 id="PCA降维"><a href="#PCA降维" class="headerlink" title="PCA降维"></a>PCA降维</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#采用上述的差异基因进行降维</span></span><br><span class="line">seu&lt;-RunPCA(object = seu,</span><br><span class="line">            pc.genes = seu@var.genes,</span><br><span class="line">            do.print = <span class="number">1</span>:<span class="number">5</span>,</span><br><span class="line">            genes.print = <span class="number">5</span>)</span><br><span class="line">seu@dr</span><br><span class="line"><span class="comment">#这样就能拿到PC的基因的重要性占比情况</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcareduction.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tmp&lt;-seu@dr$pca@gene.loadings</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcainfo.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#看一下前两个主成分的情况</span></span><br><span class="line">VizPCA(seu,pcs.use = <span class="number">1</span>:<span class="number">2</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pca1pca2.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">PCAPlot(seu,dim.1=<span class="number">1</span>,dim.2=<span class="number">2</span>,group.by=<span class="string">&#x27;Biological_Condition&#x27;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pca1pca2BC.JPG"/>


<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#热图查看PC1的情况</span></span><br><span class="line">PCHeatmap(object = seu,pc.use = <span class="number">1</span>,cells.use = ncol(seu@data),do.balanced = <span class="literal">TRUE</span>,label.columns = <span class="literal">FALSE</span>)</span><br><span class="line"><span class="comment">#一次性展示前10个主成分在各样本间的体现情况</span></span><br><span class="line"><span class="comment">#PCHeatmap(object = seu,pc.use = 1:10,cells.use = ncol(seu@data),do.balanced = TRUE,label.columns = FALSE)</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/PCA1heatmap.JPG"/>


<p>基于PCA情况看看细胞如何分群<br>重点：需要搞清楚<strong>resolution</strong>参数</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">seu&lt;-FindClusters(object = seu,</span><br><span class="line">                  reduction.type = <span class="string">&quot;pca&quot;</span>,</span><br><span class="line">                  dims.use = <span class="number">1</span>:<span class="number">10</span>,force.recalc = <span class="built_in">T</span>,</span><br><span class="line">                  resolution = <span class="number">0.9</span>,print.output = <span class="number">0</span>,</span><br><span class="line">                  save.SNN = <span class="literal">TRUE</span>)</span><br><span class="line">PrintFindClustersParams(seu)</span><br><span class="line">table(seu@meta.data$res.0.9)</span><br></pre></td></tr></table></figure>
<p><strong>resolution</strong>的值调得不一样，最后table出来得细胞亚群数也会不一样</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/pcaresulution.JPG"/>


<h2 id="细胞分群后的tSNE图"><a href="#细胞分群后的tSNE图" class="headerlink" title="细胞分群后的tSNE图"></a>细胞分群后的tSNE图</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">seu&lt;-RunTSNE(object = seu,</span><br><span class="line">             dims.use = <span class="number">1</span>:<span class="number">10</span>,</span><br><span class="line">             do.fast=<span class="literal">TRUE</span>,</span><br><span class="line">             perplexity=<span class="number">10</span>)</span><br><span class="line">TSNEPlot(object = seu)<span class="comment">#由图看出，tSNE分出了3群，为什么呢？因为上面PCA降维出3各亚群呀！</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/tsneafterpca.JPG"/>

<p><strong>同样是PCA降维算法得到得3各细胞亚群，tSNE明显展示出更好的试图效果</strong>  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">table(meta$Biological_Condition)</span><br><span class="line">table(meta$Biological_Condition,seu@meta.data$res.0.9)</span><br><span class="line"><span class="comment">#这里就有一个问题了：这里明明在探讨tSNR降维，为什么这里又用到了上面pca降维中所生成的参数res.0.9呢？</span></span><br><span class="line"><span class="comment">#有一个办法，就是先不进行pca降维，tSNE降维看看是否依然得到当前的数据</span></span><br><span class="line"><span class="comment">#问题来了，我验证了上一条猜想，结果呢：Error in GetDimReduction(object = object, reduction.type = reduction.use,  : </span></span><br><span class="line">                                        <span class="comment">#pca  dimensional reduction has not been computed</span></span><br><span class="line"><span class="comment">#所以呢，在这里的tSNE降维居然要用到PCA降维，嗯......表示不是很理解呀！两者不是独立的吗！？！？</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/grouppcaBC.JPG"/>


<h2 id="对每个亚群寻找marker基因"><a href="#对每个亚群寻找marker基因" class="headerlink" title="对每个亚群寻找marker基因"></a>对每个亚群寻找marker基因</h2><p>下面的代码需要适时修改，因为每次分组都不一样，本次是3组，因为pca降维成3组<br><strong>以第一群细胞为例<code>ident.1 = 1</code></strong>  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">markers_df&lt;-FindMarkers(object = seu,ident.1 = <span class="number">1</span>,min.pct = <span class="number">0.25</span>)</span><br><span class="line">print(x=head(markers_df))</span><br><span class="line">markers_genes=rownames(head(x=markers_df,n=<span class="number">5</span>))</span><br><span class="line">vlnPlot(object=seu,features.plot=markers_genes,use.raw=<span class="literal">TRUE</span>,y.log=<span class="literal">TRUE</span>)</span><br><span class="line">FeaturePlot(object = seu,</span><br><span class="line">            features.plot = markers_genes,</span><br><span class="line">            cols.use = <span class="built_in">c</span>(<span class="string">&quot;grey&quot;</span>,<span class="string">&quot;blue&quot;</span>),</span><br><span class="line">            reduction.use = <span class="string">&quot;tsne&quot;</span>)</span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/headmarker_df.JPG"/>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/VlnPlotcluster1marker.JPG"/>


<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FeaturePlotcluster1marker.JPG"/>


<h2 id="展示各个分类的marker基因的表达情况"><a href="#展示各个分类的marker基因的表达情况" class="headerlink" title="展示各个分类的marker基因的表达情况"></a>展示各个分类的marker基因的表达情况</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">seu.markers&lt;-FindAllMarkers(object = seu,only.pos = <span class="literal">TRUE</span>,min.pct = <span class="number">0.25</span>,thresh.use=<span class="number">0.25</span>)</span><br><span class="line">DT::datatable(seu.markers)<span class="comment">#这个操作有点优秀呀，可以在Rstudio里面试一试</span></span><br></pre></td></tr></table></figure>
<h2 id="热图展示各个亚群的marker基因"><a href="#热图展示各个亚群的marker基因" class="headerlink" title="热图展示各个亚群的marker基因"></a>热图展示各个亚群的marker基因</h2><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(dplyr)</span><br><span class="line">seu.markers%&gt;%group_by(cluster)%&gt;%top_n(<span class="number">2</span>,avg_logFC)</span><br><span class="line"><span class="comment">#每个亚群挑10个marker基因进行展示</span></span><br><span class="line">top10&lt;-seu.markers%&gt;%group_by(cluster)%&gt;%top_n(<span class="number">10</span>,avg_logFC)</span><br><span class="line">DoHeatmap(object = seu,genes.use = top10$gene,slim.col.label = <span class="literal">TRUE</span>,remove.key = <span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment">#FeaturePlot批量产图</span></span><br><span class="line"><span class="comment">#FeaturePlot(object = seu,features.plot = top10$gene,cols.use = c(&quot;grey&quot;,&quot;blue&quot;),reduction.use = &quot;tsne&quot;)</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/clustersheatmap.JPG"/>



      
    </div>
    <footer class="article-footer">
      <a data-url="http://example.com/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E4%B8%89%EF%BC%89/" data-id="ckl4ra1m2000kagvf0atka0m0" data-title="scRNA-seq数据分析学习（三）" class="article-share-link">Share</a>
      
      
      
  <ul class="article-tag-list" itemprop="keywords"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li></ul>

    </footer>
  </div>
  
    
<nav id="article-nav">
  
    <a href="/2020/02/16/%E7%89%B9%E5%AE%9A%E7%89%88%E6%9C%ACR%E5%8C%85%E4%B8%8B%E8%BD%BD%E5%8F%8A%E5%85%B6%E6%8A%A5%E9%94%99/" id="article-nav-newer" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Newer</strong>
      <div class="article-nav-title">
        
          特定版本R包下载及其报错
        
      </div>
    </a>
  
  
    <a href="/2020/02/15/scRNA-seq%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E5%AD%A6%E4%B9%A0%EF%BC%88%E5%9B%9B%EF%BC%89/" id="article-nav-older" class="article-nav-link-wrap">
      <strong class="article-nav-caption">Older</strong>
      <div class="article-nav-title">scRNA-seq数据分析学习（四）</div>
    </a>
  
</nav>

  
</article>


</section>
        
          <aside id="sidebar">
  
    

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tags</h3>
    <div class="widget">
      <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Linux/" rel="tag">Linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/MySQL/" rel="tag">MySQL</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/NGS/" rel="tag">NGS</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/R/" rel="tag">R</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/System/" rel="tag">System</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/linux/" rel="tag">linux</a></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/notes/" rel="tag">notes</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Tag Cloud</h3>
    <div class="widget tagcloud">
      <a href="/tags/Linux/" style="font-size: 16.67px;">Linux</a> <a href="/tags/MySQL/" style="font-size: 13.33px;">MySQL</a> <a href="/tags/NGS/" style="font-size: 20px;">NGS</a> <a href="/tags/R/" style="font-size: 13.33px;">R</a> <a href="/tags/System/" style="font-size: 10px;">System</a> <a href="/tags/linux/" style="font-size: 10px;">linux</a> <a href="/tags/notes/" style="font-size: 10px;">notes</a>
    </div>
  </div>

  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2021/01/">January 2021</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/12/">December 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/05/">May 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/04/">April 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/03/">March 2020</a></li><li class="archive-list-item"><a class="archive-list-link" href="/archives/2020/02/">February 2020</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2021/01/23/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BSUPPA2/">SUPPA2</a>
          </li>
        
          <li>
            <a href="/2021/01/14/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMART/">rMART</a>
          </li>
        
          <li>
            <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/">MISO</a>
          </li>
        
          <li>
            <a href="/2020/12/08/2020%20ATAC-seq%20analysis/">2020 ATAC-seq analysis</a>
          </li>
        
          <li>
            <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/">NGS数据分析之表观</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      
      &copy; 2021 gongyuqiqi<br>
      Powered by <a href="https://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>

    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    


<script src="/js/jquery-3.4.1.min.js"></script>



  
<script src="/fancybox/jquery.fancybox.min.js"></script>




<script src="/js/script.js"></script>





  </div>
</body>
</html>