<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.3.0">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.2/css/all.min.css">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/animate.css@3.1.1/animate.min.css">

<script class="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","images":"/images","scheme":"Pisces","version":"8.2.1","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12},"copycode":false,"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"prism":false,"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"},"path":"/search.xml","localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false}};
  </script>
<meta name="description" content="ATAC-seq最新处理流程">
<meta property="og:type" content="article">
<meta property="og:title" content="2020 ATAC-seq analysis">
<meta property="og:url" content="http://example.com/2020/12/08/2020%20ATAC-seq%20analysis/index.html">
<meta property="og:site_name" content="gyq&#39;s blog">
<meta property="og:description" content="ATAC-seq最新处理流程">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/multiqc.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mapped.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/NRF-PBC1-PBC2.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NRF PBC1 PBC2.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/cc.qc.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/before_after_blacklist-bed.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/narrowPeak for idr.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FRiP.JPG">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution TSS.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotAnnoPie.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/upsetplot(peakAnno, vennpie=TRUE).jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep1.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep2.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep1.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep2.jpeg">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png">
<meta property="og:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png">
<meta property="article:published_time" content="2020-12-08T05:53:55.000Z">
<meta property="article:modified_time" content="2021-02-14T07:36:34.101Z">
<meta property="article:author" content="gongyuqiqi">
<meta property="article:tag" content="NGS">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG">


<link rel="canonical" href="http://example.com/2020/12/08/2020%20ATAC-seq%20analysis/">


<script class="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>
<title>2020 ATAC-seq analysis | gyq's blog</title>
  




  <noscript>
  <style>
  body { margin-top: 2rem; }

  .use-motion .menu-item,
  .use-motion .sidebar,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header {
    visibility: visible;
  }

  .use-motion .header,
  .use-motion .site-brand-container .toggle,
  .use-motion .footer { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle,
  .use-motion .custom-logo-image {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line {
    transform: scaleX(1);
  }

  .search-pop-overlay, .sidebar-nav { display: none; }
  .sidebar-panel { display: block; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <h1 class="site-title">gyq's blog</h1>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu">
        <li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li>
        <li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li>
        <li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>Search
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup"><div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off" maxlength="80"
           placeholder="Searching..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close" role="button">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div class="search-result-container no-result">
  <div class="search-result-icon">
    <i class="fa fa-spinner fa-pulse fa-5x"></i>
  </div>
</div>

    </div>
  </div>

</div>
        
  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>

  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#2020-ATAC-seq-analysis"><span class="nav-number">1.</span> <span class="nav-text">2020 ATAC-seq analysis</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%80%E3%80%81ATAC-seq%E4%B8%AA%E6%80%A7%E5%8C%96%E7%9A%84conda-environment%E7%9A%84%E6%90%AD%E5%BB%BA"><span class="nav-number">1.1.</span> <span class="nav-text">一、ATAC-seq个性化的conda environment的搭建</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1%E3%80%81git-clone"><span class="nav-number">1.1.1.</span> <span class="nav-text">1、git clone</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2%E3%80%81%E7%8E%AF%E5%A2%83%E6%90%AD%E5%BB%BA"><span class="nav-number">1.1.2.</span> <span class="nav-text">2、环境搭建</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%8C%E3%80%81%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD"><span class="nav-number">1.2.</span> <span class="nav-text">二、数据下载</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%89%E3%80%81bowtie2%E6%AF%94%E5%AF%B9"><span class="nav-number">1.3.</span> <span class="nav-text">三、bowtie2比对</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1%E3%80%81-fastqc-multiqc%E6%9F%A5%E7%9C%8B%E5%8E%9F%E5%A7%8B%E6%95%B0%E6%8D%AE%E7%9A%84%E8%B4%A8%E9%87%8F%E6%83%85%E5%86%B5"><span class="nav-number">1.3.1.</span> <span class="nav-text">1、 fastqc+multiqc查看原始数据的质量情况</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2%E3%80%81-%E8%BF%87%E6%BB%A4%E4%BD%8E%E8%B4%A8%E9%87%8F%E7%9A%84reads"><span class="nav-number">1.3.2.</span> <span class="nav-text">2、 过滤低质量的reads</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3%E3%80%81-%E6%AF%94%E5%AF%B9%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="nav-number">1.3.3.</span> <span class="nav-text">3、 比对参考基因组</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%9B%9B%E3%80%81PCR%E5%8E%BB%E9%99%A4%E9%87%8D%E5%8F%8A%E8%B4%A8%E6%8E%A7%E5%88%B6"><span class="nav-number">1.4.</span> <span class="nav-text">四、PCR去除重及质控制</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1%E3%80%81%E5%88%9D%E6%AD%A5%E8%BF%87%E6%BB%A4"><span class="nav-number">1.4.1.</span> <span class="nav-text">1、初步过滤</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2%E3%80%81%E8%BF%9B%E4%B8%80%E6%AD%A5"><span class="nav-number">1.4.2.</span> <span class="nav-text">2、进一步</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3%E3%80%81%E6%A0%87%E8%AE%B0PCR%E9%87%8D%E5%A4%8D"><span class="nav-number">1.4.3.</span> <span class="nav-text">3、标记PCR重复</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4%E3%80%81%E5%8E%BB%E9%99%A4PCR%E9%87%8D%E5%A4%8D%E3%80%81%E5%BB%BA%E7%AB%8B%E7%B4%A2%E5%BC%95%E6%96%87%E4%BB%B6%E3%80%81%E6%8C%89name%E6%8E%92%E5%BA%8F%E7%9A%84bam%E6%96%87%E4%BB%B6%E3%80%81%E8%B4%A8%E6%8E%A7%E6%96%87%E4%BB%B6"><span class="nav-number">1.4.4.</span> <span class="nav-text">4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#5%E3%80%81%E6%B5%8B%E5%BA%8F%E6%96%87%E5%BA%93%E5%A4%8D%E6%9D%82%E5%BA%A6%E7%9A%84%E6%A3%80%E9%AA%8C"><span class="nav-number">1.4.5.</span> <span class="nav-text">5、测序文库复杂度的检验</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#6%E3%80%81%E6%8F%92%E5%85%A5%E7%89%87%E6%AE%B5%E8%B4%A8%E6%8E%A7%E6%A3%80%E6%B5%8B"><span class="nav-number">1.4.6.</span> <span class="nav-text">6、插入片段质控检测</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#7%E3%80%81TSS%E5%AF%8C%E9%9B%86"><span class="nav-number">1.4.7.</span> <span class="nav-text">7、TSS富集</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%94%E3%80%81%E7%94%9F%E6%88%90tagAlign%E6%A0%BC%E5%BC%8F%E6%96%87%E4%BB%B6"><span class="nav-number">1.5.</span> <span class="nav-text">五、生成tagAlign格式文件</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1-Convert-PE-BAM-to-tagAlign"><span class="nav-number">1.5.1.</span> <span class="nav-text">1. Convert PE BAM to tagAlign</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2-Stand-Cross-Correlation-analysis"><span class="nav-number">1.5.2.</span> <span class="nav-text">2. Stand Cross Correlation analysis</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%85%AD%E3%80%81Call-Peaks"><span class="nav-number">1.6.</span> <span class="nav-text">六、Call Peaks</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1%E3%80%81%E5%8E%BB%E9%99%A4%E7%BA%BF%E7%B2%92%E4%BD%93%E5%9F%BA%E5%9B%A0%E7%9A%84TagAlign%E6%A0%BC%E5%BC%8F%E6%96%87%E4%BB%B6%E8%BF%9B%E8%A1%8Cshift%E6%93%8D%E4%BD%9C%EF%BC%8C%E8%BE%93%E5%85%A5macs2%E8%BD%AF%E4%BB%B6%E5%8E%BBcallpeak"><span class="nav-number">1.6.1.</span> <span class="nav-text">1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2%E3%80%81%E5%8E%BB%E9%99%A4ENCODE%E5%88%97%E5%85%A5%E9%BB%91%E5%90%8D%E5%8D%95%E7%9A%84%E5%8C%BA%E5%9F%9F"><span class="nav-number">1.6.2.</span> <span class="nav-text">2、去除ENCODE列入黑名单的区域</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3%E3%80%81Irreproducibility-Discovery-Rate-IDR-%E8%AF%84%E4%BC%B0"><span class="nav-number">1.6.3.</span> <span class="nav-text">3、Irreproducibility Discovery Rate (IDR)评估</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#4%E3%80%81Fraction-of-reads-in-peaks-FRiP-%E8%AF%84%E4%BC%B0"><span class="nav-number">1.6.4.</span> <span class="nav-text">4、Fraction of reads in peaks (FRiP)评估</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%83%E3%80%81Peak-annotation"><span class="nav-number">1.7.</span> <span class="nav-text">七、Peak annotation</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#1%E3%80%81Feature-Distribution"><span class="nav-number">1.7.1.</span> <span class="nav-text">1、Feature Distribution</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#2%E3%80%81%E6%9F%A5%E7%9C%8Bpeaks%E5%9C%A8%E5%85%A8%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%8A%E7%9A%84%E5%88%86%E5%B8%83"><span class="nav-number">1.7.2.</span> <span class="nav-text">2、查看peaks在全基因组上的分布</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#3%E3%80%81%E6%8B%BF%E5%88%B0%E6%AF%8F%E4%B8%AA%E6%A0%B7%E6%9C%AC%E4%B8%ADpeaks%E5%AF%B9%E5%BA%94%E5%BE%97%E5%9F%BA%E5%9B%A0%E5%90%8D"><span class="nav-number">1.7.3.</span> <span class="nav-text">3、拿到每个样本中peaks对应得基因名</span></a></li></ol></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author site-overview-item animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="gongyuqiqi"
      src="/assets/img/ava.jpg">
  <p class="site-author-name" itemprop="name">gongyuqiqi</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap site-overview-item animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">28</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



        </div>
      </div>
    </div>
  </aside>
  <div class="sidebar-dimmer"></div>


    </header>

    
  <div class="back-to-top" role="button">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>


    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2020/12/08/2020%20ATAC-seq%20analysis/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/assets/img/ava.jpg">
      <meta itemprop="name" content="gongyuqiqi">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="gyq's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          2020 ATAC-seq analysis
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2020-12-08 13:53:55" itemprop="dateCreated datePublished" datetime="2020-12-08T13:53:55+08:00">2020-12-08</time>
    </span>
      <span class="post-meta-item">
        <span class="post-meta-item-icon">
          <i class="far fa-calendar-check"></i>
        </span>
        <span class="post-meta-item-text">Edited on</span>
        <time title="Modified: 2021-02-14 15:36:34" itemprop="dateModified" datetime="2021-02-14T15:36:34+08:00">2021-02-14</time>
      </span>

  
      </div>
      <div class="post-meta">
    <span class="post-meta-item" title="Symbols count in article">
      <span class="post-meta-item-icon">
        <i class="far fa-file-word"></i>
      </span>
      <span class="post-meta-item-text">Symbols count in article: </span>
      <span>17k</span>
    </span>
    <span class="post-meta-item" title="Reading time">
      <span class="post-meta-item-icon">
        <i class="far fa-clock"></i>
      </span>
      <span class="post-meta-item-text">Reading time &asymp;</span>
      <span>16 mins.</span>
    </span>
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">
        <p>ATAC-seq最新处理流程</p>
<a id="more"></a>

<h1 id="2020-ATAC-seq-analysis"><a href="#2020-ATAC-seq-analysis" class="headerlink" title="2020 ATAC-seq analysis"></a><strong>2020 ATAC-seq analysis</strong></h1><h2 id="一、ATAC-seq个性化的conda-environment的搭建"><a href="#一、ATAC-seq个性化的conda-environment的搭建" class="headerlink" title="一、ATAC-seq个性化的conda environment的搭建"></a>一、ATAC-seq个性化的conda environment的搭建</h2><h3 id="1、git-clone"><a href="#1、git-clone" class="headerlink" title="1、git clone"></a>1、git clone</h3><p>git clone文章中提到的atac-seq-pipeline (里面有你接下来分析所需要的绝大部分东西。另外一定要去看README.md，这个文档会告诉你分析ATAC-seq需要做些准备，进行些什么工作。)</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ <span class="built_in">cd</span> ~/ATAC</span><br><span class="line">$ git <span class="built_in">clone</span> https://github.com/ENCODE-DCC/atac-seq-pipeline</span><br></pre></td></tr></table></figure>
<h3 id="2、环境搭建"><a href="#2、环境搭建" class="headerlink" title="2、环境搭建"></a>2、环境搭建</h3><p>（1）conda base environment搭建</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#一路yes下去</span></span><br><span class="line">$ wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh</span><br><span class="line">$ bash Miniconda3-4.6.14-Linux-x86_64.sh</span><br><span class="line"><span class="comment">#最后一个yes时你同意了每次自动启动conda的base环境，显示安装完成后，一定要退出当前的终端，再重新打开一次，你就会发现在游标的最左边多了一个（base）,你的base conda environment就激活啦！你的conda环境搭建就成功了一小步(*^__^*) 嘻嘻……</span></span><br></pre></td></tr></table></figure>
<p>（2）失活掉base conda environment自动激活的设置</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda config --<span class="built_in">set</span> auto_activate_base <span class="literal">false</span></span><br><span class="line"><span class="comment">#同样的完成上诉操作后，记得退出当前终端，再次打开一个新的终端。你就会发现游标的最左边没有（base）了</span></span><br></pre></td></tr></table></figure>
<p>（3）安装pipeline’s conda environment</p>
<p>！？你是不是充满了疑问！？下面的两个shell脚本哪里冒出来的，干什么用的。不要忘了，我们第一步git clone了这个atac-seq-pipeline呢！这两个脚本就是就是来源于这个clone下来的pipeline的。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 网不给力的朋友，比如说我，这一部就是整个分析流程的限速步骤！！！网络给力这一步也需要花一些时间，读一下两个shell文件就知道为什么啦。</span></span><br><span class="line">$ bash scripts/uninstall_conda_env.sh  <span class="comment"># uninstall it for clean-install</span></span><br><span class="line"><span class="comment">#下面这个shell脚本会创建两个分析ATAC-seq数据所需要的环境，一个基于python3，一个基于python2的</span></span><br><span class="line">$ bash scripts/install_conda_env.sh</span><br></pre></td></tr></table></figure>
<p>（4）激活安装的conda环境</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ conda activate encode-atac-seq-pipeline</span><br><span class="line"><span class="comment">#你一定会疑问为什么是encode-atac-seq-pipeline,阅读一下scripts/install_conda_env.sh这个脚本就知道啦。因为源文档作者把这个环境名命名为encode-atac-seq-pipeline。</span></span><br></pre></td></tr></table></figure>
<p>（5）查看所有的conda环境</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda env list</span><br></pre></td></tr></table></figure>
<p>我们会看到有一个基于python2的环境，没有标明python版本的那个encode-atac-seq-pipeline是基于python3的。有些软件是要基于特定版本的python运行的。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/conda environment.JPG"/></p>
<h2 id="二、数据下载"><a href="#二、数据下载" class="headerlink" title="二、数据下载"></a>二、数据下载</h2><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#一次性下载所有的......_1.fastq.gz和......_2.fastq.gz样本</span></span><br><span class="line">dir=/home/gongyuqi/.aspera/connect/etc/asperaweb_id_dsa.openssh</span><br><span class="line">x=_1</span><br><span class="line">y=_2</span><br><span class="line"><span class="keyword">for</span> id <span class="keyword">in</span> &#123;93,98,99&#125;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$id<span class="variable">$x</span>.fastq.gz .</span><br><span class="line">ascp -QT -l 300m -P33001 -i <span class="variable">$dir</span> era-fasp@fasp.sra.ebi.ac.uk:/vol1/fastq/SRR126/0<span class="variable">$id</span>/SRR126920<span class="variable">$id</span>/SRR126920$id<span class="variable">$y</span>.fastq.gz . </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<h2 id="三、bowtie2比对"><a href="#三、bowtie2比对" class="headerlink" title="三、bowtie2比对"></a>三、bowtie2比对</h2><h3 id="1、-fastqc-multiqc查看原始数据的质量情况"><a href="#1、-fastqc-multiqc查看原始数据的质量情况" class="headerlink" title="1、 fastqc+multiqc查看原始数据的质量情况"></a>1、 fastqc+multiqc查看原始数据的质量情况</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">conda insatll -y fastqc multiqc</span><br><span class="line"><span class="built_in">cd</span> raw </span><br><span class="line">fastqc -t 8 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></table></figure>
<p>multiqc结果表明原始数据质量已经很好了，完全没有adaptor残留,连PCR重复都是勉强过关的。per base sequence content和kmer content两项指标不合格，但是并不会对后续的比对造成多大的负面影响，同时这两项指标通过过滤软件的处理可能并不会后得到改善。这样的数据其实可以直接比对的。<br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/multiqc.JPG"/></p>
<h3 id="2、-过滤低质量的reads"><a href="#2、-过滤低质量的reads" class="headerlink" title="2、 过滤低质量的reads"></a>2、 过滤低质量的reads</h3><p>（如果原始数据需要进行过滤处理，那么走下面这个过滤流程，我们这里就不过滤了）</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">nohup cat rawdata.txt|<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">trim_galore -j 35 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o ../clean <span class="variable">$fq1</span> <span class="variable">$fq2</span></span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#过滤后的数据质量控制，看数据是否有得到改善，有没有去掉adaptor非常重要</span></span><br><span class="line"><span class="built_in">cd</span> clean</span><br><span class="line">fastqc -t 20 *gz -o ./</span><br><span class="line">multiqc *.gzip -o ./</span><br></pre></td></tr></table></figure>
<h3 id="3、-比对参考基因组"><a href="#3、-比对参考基因组" class="headerlink" title="3、 比对参考基因组"></a>3、 比对参考基因组</h3><ul>
<li><p>比对</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup cat sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">fq1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">fq2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;fq1%%_*&#125;</span>.sort.bam</span><br><span class="line">bowtie2  --very-sensitive -X2000 --mm --threads 50 -x /home/gongyuqi/ref/mm10/index/mm10 -1 <span class="variable">$fq1</span> -2 <span class="variable">$fq2</span>|samtools sort -O bam -@ 15 - -o ../align/<span class="variable">$sample</span> </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure></li>
<li><p>比对情况统计及查看</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">ls *.bam | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">samtools flagstat -@ 4 <span class="variable">$id</span>  &gt; <span class="variable">$&#123;id%%.*&#125;</span>.<span class="built_in">stat</span></span><br><span class="line"><span class="keyword">done</span></span><br><span class="line">ls *.<span class="built_in">stat</span> | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> | grep <span class="string">&quot;mapped (&quot;</span>;<span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/mapped.JPG"/>
</li>
<li><p>生成bam文件的索引文件</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nohup ls *sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools index -@ 10 <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>

</li>
</ul>
<h2 id="四、PCR去除重及质控制"><a href="#四、PCR去除重及质控制" class="headerlink" title="四、PCR去除重及质控制"></a>四、PCR去除重及质控制</h2><h3 id="1、初步过滤"><a href="#1、初步过滤" class="headerlink" title="1、初步过滤"></a>1、初步过滤</h3><p>#=============================<br>#Remove unmapped, mate unmapped<br>#not primary alignment, reads failing platform<br>#Only keep properly paired reads<br>#Obtain name sorted BAM file<br>#=============================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#fixmate要求输入的bam文件按名称排序，所以第一步初步过滤后要按reads名称排序一下</span></span><br><span class="line">nohup ls *.sort.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools view -@ 8 -F 524 -f 2 -u <span class="variable">$id</span> | samtools sort -@ 8 -n /dev/stdin -o ../filter/<span class="variable">$&#123;id%%.*&#125;</span>.filt.nmsrt.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#multimapping=4是bowtie2的默认设置</span></span><br><span class="line"><span class="built_in">cd</span> ../filter</span><br><span class="line">multimapping=4 </span><br><span class="line">nohup ls *filt.nmsrt.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools view -@ 8 -h <span class="variable">$id</span>|assign_multimappers.py -k <span class="variable">$multimapping</span> --paired-end |samtools fixmate -@ 8 -r /dev/stdin <span class="variable">$&#123;id%%.*&#125;</span>.filt.nmsrt.fixmate.bam; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="2、进一步"><a href="#2、进一步" class="headerlink" title="2、进一步"></a>2、进一步</h3><p>#===============================================<br>#Remove orphan reads (pair was removed)<br>#and read pairs mapping to different chromosomes<br>#Obtain position sorted BAM<br>#===============================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line">nohup ls *.filt.nmsrt.fixmate.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> samtools view -@ 8 -F 1804 -f 2 -u <span class="variable">$id</span> | samtools sort -@ 8 /dev/stdin -o <span class="variable">$&#123;id%%.*&#125;</span>.prefilt.bam;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="3、标记PCR重复"><a href="#3、标记PCR重复" class="headerlink" title="3、标记PCR重复"></a>3、标记PCR重复</h3><p>#===============<br>#Mark duplicates<br>#===============</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">MARKDUP=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line">nohup ls *.prefilt.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> java -Xmx16G -jar <span class="variable">$MARKDUP</span> MarkDuplicates INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">$&#123;id%%.*&#125;</span>.prefilt.dupmark.bam METRICS_FILE=<span class="variable">$&#123;id%%.*&#125;</span>.prefilt.dupmark.qc VALIDATION_STRINGENCY=LENIENT ASSUME_SORTED=<span class="literal">true</span> REMOVE_DUPLICATES=<span class="literal">false</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件"><a href="#4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件" class="headerlink" title="4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件"></a>4、去除PCR重复、建立索引文件、按name排序的bam文件、质控文件</h3><p>#============================<br>#Remove duplicates<br>#Index final position sorted BAM<br>#Create final name sorted BAM<br>#============================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line"><span class="comment">#去重</span></span><br><span class="line">nohup ls *.prefilt.dupmark.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools view -@ 16 -F 1804 -f 2 -b <span class="variable">$id</span> &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.bam;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#构建索引文件</span></span><br><span class="line">nohup ls *nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools index <span class="variable">$id</span>;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="5、测序文库复杂度的检验"><a href="#5、测序文库复杂度的检验" class="headerlink" title="5、测序文库复杂度的检验"></a>5、测序文库复杂度的检验</h3><p>#=============================<br>#Compute library complexity<br>#=============================<br>#Sort by name<br>#convert to bedPE and obtain fragment coordinates<br>#sort by position and strand<br>#Obtain unique count statistics<br>#================================================</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本写循环</span></span><br><span class="line">nohup ls *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> samtools sort -@ 16 -n <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.bam;<span class="keyword">done</span> &amp;</span><br><span class="line">nohup ls *.nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools bamtobed -bedpe -i <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1,$2,$4,$6,$9,$10&#125;&#x27;</span> | grep -v <span class="string">&#x27;chrM&#x27;</span> | sort | uniq -c | awk <span class="string">&#x27;BEGIN&#123;mt=0;m0=0;m1=0;m2=0&#125; ($1==1)&#123;m1=m1+1&#125; ($1==2)&#123;m2=m2+1&#125; &#123;m0=m0+1&#125; &#123;mt=mt+$1&#125; END&#123;m1_m2=-1.0; if(m2&gt;0) m1_m2=m1/m2;printf &quot;%d\t%d\t%d\t%d\t%f\t%f\t%f\n&quot;,mt,m0,m1,m2,m0/mt,m1/m0,m1_m2&#125;&#x27;</span> &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.pbc.qc;<span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<p>Library complexity measures计算结果如下，…nodup.pbc.qc文件格式为：<br>TotalReadPairs<br>DistinctReadPairs<br>OneReadPair<br>TwoReadPairs<br>NRF=Distinct/Total<br>PBC1=OnePair/Distinct<br>PBC2=OnePair/TwoPair</p>
<p>针对NRF、PBC1、PBC2这几个指标，ENCODE官网提供的标准如下:</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/NRF-PBC1-PBC2.jpeg"></p>
<p>我们此次4个样本的分析结果显示这几个指标的值如下所示：</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/NRF PBC1 PBC2.JPG"/>

<p>除了第一个样本（SRR12692092），其他样本计算结果显示NRF、PBC1、PBC2的值都非常完美，说明我们进行过滤和PCR去重的bam文件质量上没有问题，可以用于后续的分析。</p>
<h3 id="6、插入片段质控检测"><a href="#6、插入片段质控检测" class="headerlink" title="6、插入片段质控检测"></a>6、插入片段质控检测</h3><table><tr><td bgcolor=GreenYellow>用于评估ATAC/Chip实验质量好坏的一个重要指标</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#多个样本循环</span></span><br><span class="line">picard_jar=/home/gongyuqi/miniconda3/envs/encode-atac-seq-pipeline/share/picard-2.20.7-0/picard.jar</span><br><span class="line">nohup ls *.nodup.srt.name.bam | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> java -Xmx6G -XX:ParallelGCThreads=1 -jar <span class="variable">$&#123;picard_jar&#125;</span> CollectInsertSizeMetrics INPUT=<span class="variable">$id</span> OUTPUT=<span class="variable">$&#123;id%%.*&#125;</span>.INSERT.DATA H=<span class="variable">$&#123;id%%.*&#125;</span>.INSERT.PLOT.pdf VERBOSITY=ERROR QUIET=TRUE USE_JDK_DEFLATER=TRUE USE_JDK_INFLATER=TRUE W=1000 STOP_AFTER=5000000; <span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<p>成功的ATAC-seq实验会生成具有逐级递减和周期性的峰，对应无核小体区域（Nucleosome free region, NRF,100bp）,单核区域（mono-nucleosome,200bp），双核区域（400bp），三核区域（600bp）。</p>
<p>本次结果显示样本质量OK</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.INSERT.PLOT.jpg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.INSERT.PLOT.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.INSERT.PLOT.jpg" width="50%"></p>
<h3 id="7、TSS富集"><a href="#7、TSS富集" class="headerlink" title="7、TSS富集"></a>7、TSS富集</h3><p><strong>以BRM014-10uM_24h_wt样本为例，可视化peaks在基因组上的分布特征(TSS区域；TSS、gene-body、TSE区域)</strong></p>
<ul>
<li>both -R and -S can accept multiple files</li>
<li>UCSC官网下载小鼠参考基因组注释文件，详情参考：<a target="_blank" rel="noopener" href="http://www.bio-info-trainee.com/2494.html">http://www.bio-info-trainee.com/2494.html</a></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#生成bw文件用于后续的computeMatrix</span></span><br><span class="line">nohup ls *.nodup.bam |<span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">nohup bamCoverage --normalizeUsing CPM -b <span class="variable">$id</span> -o <span class="variable">$&#123;id%%.*&#125;</span>.nodup.bw</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#注释文件绝对路劲</span></span><br><span class="line">refseq=/home/gongyuqi/ref/mm10/mouse.ucsc.refseq.bed</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS</span></span><br><span class="line"><span class="comment">##生成plotHeatmap需要的gz文件</span></span><br><span class="line">nohup computeMatrix reference-point  --referencePoint TSS  -p 30 \</span><br><span class="line">-b 3000 -a 3000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##可视化文件</span></span><br><span class="line">nohup plotHeatmap -m BRM014-10uM_24h_wt_TSS.gz  -out BRM014-10uM_24h_wt_TSS_rainbow.png --colorMap rainbow &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment">#TSS_gene-body_TSE</span></span><br><span class="line"><span class="comment">##生成plotHeatmap需要的gz文件</span></span><br><span class="line">nohup computeMatrix scale-regions -p 30 \</span><br><span class="line">-a 3000 -b 3000 -m 5000 \</span><br><span class="line">-R <span class="variable">$refseq</span> \</span><br><span class="line">-S SRR12692098.nodup.bw SRR12692099.nodup.bw \</span><br><span class="line">--skipZeros -o BRM014-10uM_24h_wt_TSS_gene0body_TSE.gz \</span><br><span class="line">--outFileSortedRegions BRM014-10uM_24h_wt_TSS.bed &amp;</span><br><span class="line"><span class="comment">##可视化文件</span></span><br><span class="line">nohup plotHeatmap -m BRM014-10uM_24h_wt_TSS_genebody_TSE.gz  -out BRM014-10uM_24h_wt_TSS_genebody_TSE.png --colorMap rainbow &amp;</span><br></pre></td></tr></table></figure>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_rainbow.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014-10uM_24h_wt_TSS_genebody_TSE.png" width="50%"></p>
<h2 id="五、生成tagAlign格式文件"><a href="#五、生成tagAlign格式文件" class="headerlink" title="五、生成tagAlign格式文件"></a>五、生成tagAlign格式文件</h2><h3 id="1-Convert-PE-BAM-to-tagAlign"><a href="#1-Convert-PE-BAM-to-tagAlign" class="headerlink" title="1. Convert PE BAM to tagAlign"></a>1. Convert PE BAM to tagAlign</h3><ul>
<li>对于单端序列。直接用bed格式就可以；对于双端序列，推荐用bedpe格式。这两种格式都可以称之为tagAlign，可以作为macs的输入文件。</li>
<li>tagAligen格式相比bam，文件大小会小很多，更加方便文件的读取。在转换得到tagAlign格式之后，我们就可以很容易的将坐标进行偏移</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">nohup ls *nodup.srt.name.bam|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools bamtobed -bedpe -mate1 -i <span class="variable">$id</span>  | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.bedpe.gz;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#含有chrM的染色体的TagAlign文件</span></span><br><span class="line">nohup ls *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;printf &quot;%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n&quot;,$1,$2,$3,$9,$4,$5,$6,$10&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#去除chrM的染色体的TagAlign文件</span></span><br><span class="line">nohup ls *nodup.srt.name.bedpe.gz|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -P -v <span class="string">&quot;^chrM&quot;</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;printf &quot;%s\t%s\t%s\tN\t1000\t%s\n%s\t%s\t%s\tN\t1000\t%s\n&quot;,$1,$2,$3,$9,$4,$5,$6,$10&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.nomit.srt.name.tagAlign.gz; <span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<h3 id="2-Stand-Cross-Correlation-analysis"><a href="#2-Stand-Cross-Correlation-analysis" class="headerlink" title="2. Stand Cross Correlation analysis"></a>2. Stand Cross Correlation analysis</h3><table><tr><td bgcolor=GreenYellow>用于评估ATAC/Chip实验质量好坏的一个重要指标</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">NREADS=25000000</span><br><span class="line">nohup ls *.nodup.srt.name.bedpe.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> zcat <span class="variable">$id</span> | grep -v “chrM” | shuf -n <span class="variable">$&#123;NREADS&#125;</span> --random-source=&lt;(openssl enc -aes-256-ctr -pass pass:$(zcat -f <span class="variable">$&#123;id%%.*&#125;</span>.nodup.srt.name.tagAlign.gz | wc -c) -nosalt &lt;/dev/zero 2&gt;/dev/null) | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;print $1,$2,$3,&quot;N&quot;,&quot;1000&quot;,$9&#125;&#x27;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%.*&#125;</span>.nodup.nomit.srt.name.$((NREADS / <span class="number">1000000</span>)).tagAlign.gz; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#命令最终会生成交叉相关质量评估文件，*.cc.qc文件中会输出包含11列的信息，重点关注9-11列的信息，cc.plot.pdf文件相当于*.cc.qc文件的可视化</span></span><br><span class="line">nohup ls *$((NREADS / <span class="number">1000000</span>)).tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> Rscript $(<span class="built_in">which</span> run_spp.R) -c=<span class="variable">$id</span> -p=10 -filtchr=chrM -savp=<span class="variable">$&#123;id%%.*&#125;</span>.cc.plot.pdf -out=<span class="variable">$&#123;id%%.*&#125;</span>.cc.qc; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#质控结果查看，主要看NSC,RSC,Quality tag三个值即输出文件的第9列，第10列，第11列。</span></span><br><span class="line">ls *.cc.qc|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> | awk <span class="string">&#x27;&#123;print $9, &quot;\t&quot;, $10, &quot;\t&quot;, $11&#125;&#x27;</span>;<span class="keyword">done</span> </span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/cc.qc.JPG" width="50%">
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692093.cc.plot.jpg" width="50%">

<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098.cc.plot.jpg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692099.cc.plot.jpg" width="50%"></p>
<ul>
<li>质控结果解读<blockquote>
<p>Normalized strand cross-correlation coefficent (NSC)：NSC是最大交叉相关值除以背景交叉相关的比率(所有可能的链转移的最小交叉相关值)。NSC值越大表明富集效果越好，NSC值低于1.1表明较弱的富集，小于1表示无富集。NSC值稍微低于1.05，有较低的信噪比或很少的峰，这肯能是生物学真实现象，比如有的因子在特定组织类型中只有很少的结合位点；也可能确实是数据质量差。</p>
<p>Relative strand cross-correlation coefficient (RSC)：RSC是片段长度相关值减去背景相关值除以phantom-peak相关值减去背景相关值。RSC的最小值可能是0，表示无信号；富集好的实验RSC值大于1；低于1表示质量低。</p>
<p>QualityTag: Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High,2:veryHigh)</p>
<p>查看交叉相关性质量评估结果，主要看NSC,RSC,Quality tag三个值，这三个值分别对应输出文件的第9列，第10列，第11列。</p>
</blockquote>
</li>
</ul>
<h2 id="六、Call-Peaks"><a href="#六、Call-Peaks" class="headerlink" title="六、Call Peaks"></a>六、Call Peaks</h2><h3 id="1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak"><a href="#1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak" class="headerlink" title="1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak"></a>1、去除线粒体基因的TagAlign格式文件进行shift操作，输入macs2软件去callpeak</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">smooth_window=150 <span class="comment"># default</span></span><br><span class="line">shiftsize=$(( -<span class="variable">$smooth_window</span>/<span class="number">2</span> ))</span><br><span class="line">pval_thresh=0.01</span><br><span class="line">nohup ls *nodup.nomit.srt.name.tagAlign.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; \</span><br><span class="line"><span class="keyword">do</span> macs2 callpeak \</span><br><span class="line">-t <span class="variable">$id</span> -f BED -n <span class="string">&quot;<span class="variable">$&#123;id%%.*&#125;</span>&quot;</span> -g mm -p <span class="variable">$pval_thresh</span> \</span><br><span class="line">--<span class="built_in">shift</span> <span class="variable">$shiftsize</span> --extsize <span class="variable">$smooth_window</span> --nomodel -B --SPMR --keep-dup all --call-summits; \</span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<h3 id="2、去除ENCODE列入黑名单的区域"><a href="#2、去除ENCODE列入黑名单的区域" class="headerlink" title="2、去除ENCODE列入黑名单的区域"></a>2、去除ENCODE列入黑名单的区域</h3><ul>
<li>去除黑名单的bed文件，用于后续的peaks注释<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">BLACKLIST=/home/gongyuqi/project/ATAC/mm10.blacklist.bed.gz</span><br><span class="line"><span class="comment">#*_summits.bed为macs2软件callpeak的结果文件之一</span></span><br><span class="line">nohup ls *_summits.bed | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools intersect -a <span class="variable">$id</span> -b <span class="variable">$BLACKLIST</span> -v &gt; <span class="variable">$&#123;id%%.*&#125;</span>_filt_blacklist.bed; <span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#查看过滤黑名单的区域前后的bed文件的peaks数</span></span><br><span class="line">ls *summits.bed|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> |wc -l &gt;&gt;summits.bed.txt;<span class="keyword">done</span></span><br><span class="line">ls *summits_filt_blacklist.bed|<span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> cat <span class="variable">$id</span> |wc -l &gt;&gt;summits_filt_blacklist.bed.txt;<span class="keyword">done</span></span><br><span class="line">past summits.bed.txt summits_filt_blacklist.bed.txt </span><br></pre></td></tr></table></figure>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/before_after_blacklist-bed.JPG"/>





</li>
</ul>
<ul>
<li>去除黑名单的narrowPeaks文件，用于后续的IDR评估<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#使用IDR需要先对MACS2的结果文件narrowPeak根据-log10(p-value)进行排序,-log10(p-value)在第八列。</span></span><br><span class="line"><span class="comment"># Sort by Col8 in descending order and replace long peak names in Column 4 with Peak_&lt;peakRank&gt;</span></span><br><span class="line"><span class="comment">#*_peaks.narrowPeak为macs2软件callpeak的结果文件之一</span></span><br><span class="line">NPEAKS=300000 </span><br><span class="line">ls *_peaks.narrowPeak | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> sort -k 8gr,8gr <span class="variable">$id</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125;&#123;$4=&quot;Peak_&quot;NR ; print $0&#125;&#x27;</span> | head -n <span class="variable">$&#123;NPEAKS&#125;</span> | gzip -nc &gt; <span class="variable">$&#123;id%%_*&#125;</span>.narrowPeak.gz; <span class="keyword">done</span></span><br><span class="line"></span><br><span class="line">BLACKLIST=../BLACKLIST/mm10.blacklist.bed.gz</span><br><span class="line"></span><br><span class="line"><span class="comment">#生成不压缩文件</span></span><br><span class="line">ls *.narrowPeak.gz | <span class="keyword">while</span> <span class="built_in">read</span> id; <span class="keyword">do</span> bedtools intersect -v -a <span class="variable">$id</span> -b <span class="variable">$&#123;BLACKLIST&#125;</span> | awk <span class="string">&#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; &#123;if ($5&gt;1000) $5=1000; print $0&#125;&#x27;</span> | grep -P <span class="string">&#x27;chr[\dXY]+[ \t]&#x27;</span>  &gt; <span class="variable">$&#123;id%%.*&#125;</span>.narrowPeak.filt_blacklist; <span class="keyword">done</span></span><br><span class="line"><span class="comment">#生成压缩文件</span></span><br><span class="line"><span class="comment">#ls *.narrowPeak.gz | while read id; do bedtools intersect -v -a $id -b $&#123;BLACKLIST&#125; | awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; &#123;if ($5&gt;1000) $5=1000; print $0&#125;&#x27; | grep -P &#x27;chr[\dXY]+[ \t]&#x27; | gzip -nc &gt; $&#123;id%%.*&#125;.narrowPeak.filt_blacklist.gz; done</span></span><br></pre></td></tr></table></figure>
<h3 id="3、Irreproducibility-Discovery-Rate-IDR-评估"><a href="#3、Irreproducibility-Discovery-Rate-IDR-评估" class="headerlink" title="3、Irreproducibility Discovery Rate (IDR)评估"></a>3、Irreproducibility Discovery Rate (IDR)评估</h3><table><tr><td bgcolor=GreenYellow>用于评估重复样本间peaks一致性的重要指标</td></tr></table>
首先生成narrowPeak_sample.txt文件，方便后续循环处理，生成文件内容如下：
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/narrowPeak for idr.JPG"/>

</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">nohup cat narrowPeak_sample.txt | <span class="keyword">while</span> <span class="built_in">read</span> id</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">arr=(<span class="variable">$&#123;id&#125;</span>)</span><br><span class="line">Rep1=<span class="variable">$&#123;arr[0]&#125;</span></span><br><span class="line">Rep2=<span class="variable">$&#123;arr[1]&#125;</span></span><br><span class="line">sample=<span class="variable">$&#123;Rep1%%.*&#125;</span>_<span class="variable">$&#123;Rep2%%.*&#125;</span>_idr</span><br><span class="line">idr --samples <span class="variable">$Rep1</span> <span class="variable">$Rep2</span> --input-file-type narrowPeak -o <span class="variable">$sample</span> --plot  </span><br><span class="line"><span class="keyword">done</span> &amp;</span><br></pre></td></tr></table></figure>
<ul>
<li>DMSO_24h_wt （样本处理情况）</li>
<li>SRR12692092.filt_blacklist.narrowPeak SRR12692093.filt_blacklist.narrowPeak</li>
<li>没有通过IDR阈值的显示为红色</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr_command.png"/>

<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692092_SRR12692093_idr.png"/>

<ul>
<li>BRM014-10uM_24h_wt （样本处理情况）</li>
<li>SRR12692098.filt_blacklist.narrowPeak SRR12692099.filt_blacklist.narrowPeak</li>
<li>没有通过IDR阈值的显示为红色</li>
</ul>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr_command.png"/>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/SRR12692098_SRR12692099_idr.png"/>

<blockquote>
<p>IDR评估会同时考虑peaks间的overlap和富集倍数的一致性。通过IDR阈值（0.05）的占比越大，说明重复样本间peaks一致性越好。从idr的分析结果看，我们的测试数据还可以的呢。</p>
</blockquote>
<p><strong>IDR评估相关参考资料：</strong>  </p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/d8a7056b4294?utm_campaign=maleskine&utm_content=note&utm_medium=seo_notes&utm_source=recommendation">重复样本的处理——IDR</a></p>
<h3 id="4、Fraction-of-reads-in-peaks-FRiP-评估"><a href="#4、Fraction-of-reads-in-peaks-FRiP-评估" class="headerlink" title="4、Fraction of reads in peaks (FRiP)评估"></a>4、Fraction of reads in peaks (FRiP)评估</h3><table><tr><td bgcolor=GreenYellow>反映样本富集效果好坏的评价指标</td></tr></table>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#生成bed文件</span></span><br><span class="line">nohup ls *.nodup.bam|<span class="keyword">while</span> <span class="built_in">read</span> id;<span class="keyword">do</span> (bedtools bamtobed -i <span class="variable">$id</span> &gt;<span class="variable">$&#123;id%%.*&#125;</span>.nodup.bed) ;<span class="keyword">done</span> &amp;</span><br><span class="line"><span class="comment">#批量计算FRiP</span></span><br><span class="line">ls *_summits_filt_blacklist.bed|<span class="keyword">while</span>  <span class="built_in">read</span> id;</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$id</span></span><br><span class="line">bed=<span class="variable">$&#123;id%%_*&#125;</span>.nodup.bed </span><br><span class="line">Reads=$(bedtools intersect -a <span class="variable">$bed</span> -b <span class="variable">$id</span> |wc -l|awk <span class="string">&#x27;&#123;print $1&#125;&#x27;</span>)</span><br><span class="line">totalReads=$(wc -l <span class="variable">$bed</span>|awk <span class="string">&#x27;&#123;print $1&#125;&#x27;</span>)</span><br><span class="line"><span class="built_in">echo</span> <span class="variable">$Reads</span>  <span class="variable">$totalReads</span> </span><br><span class="line"><span class="built_in">echo</span> <span class="string">&#x27;==&gt; FRiP value:&#x27;</span> $(bc &lt;&lt;&lt; <span class="string">&quot;scale=2;100*<span class="variable">$Reads</span>/<span class="variable">$totalReads</span>&quot;</span>)<span class="string">&#x27;%&#x27;</span></span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<p>FRiP值在5%以上算比较好的。但也不绝对，这是个软阈值，可以作为一个参考。</p>
<img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/FRiP.JPG"/>

<p><strong>FRiP评估相关参考资料：</strong></p>
<p><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation">https://www.jianshu.com/p/09e05bcd6981?utm_campaign=maleskine&amp;utm_content=note&amp;utm_medium=seo_notes&amp;utm_source=recommendation</a></p>
<h2 id="七、Peak-annotation"><a href="#七、Peak-annotation" class="headerlink" title="七、Peak annotation"></a>七、Peak annotation</h2><h3 id="1、Feature-Distribution"><a href="#1、Feature-Distribution" class="headerlink" title="1、Feature Distribution"></a>1、Feature Distribution</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">setwd(<span class="string">&quot;path to bed file&quot;</span>)</span><br><span class="line">library(ChIPpeakAnno)</span><br><span class="line">library(TxDb.Mmusculus.UCSC.mm10.knownGene)</span><br><span class="line">library(org.Mm.eg.db)</span><br><span class="line">library(BiocInstaller)</span><br><span class="line">library(ChIPseeker)</span><br><span class="line">txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene</span><br><span class="line">promoter &lt;- getPromoters(TxDb=txdb, upstream=<span class="number">3000</span>, downstream=<span class="number">3000</span>)</span><br><span class="line">files = <span class="built_in">list</span>(DMSO_24h_wt_rep1 = <span class="string">&quot;SRR12692092_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             DMSO_24h_wt_rep2 = <span class="string">&quot;SRR12692093_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             BRM014_10uM_24h_wt_rep1 = <span class="string">&quot;SRR12692098_summits_filt_blacklist.bed&quot;</span>,</span><br><span class="line">             BRM014_10uM_24h_wt_rep2 = <span class="string">&quot;SRR12692099_summits_filt_blacklist.bed&quot;</span>)</span><br><span class="line"><span class="comment">#汇总所有样本</span></span><br><span class="line"><span class="comment">#plotAnnoBar和plotDistToTSS这两个柱状图都支持多个数据同时展示</span></span><br><span class="line">peakAnnoList &lt;- lapply(files, annotatePeak, </span><br><span class="line">                       TxDb=txdb,</span><br><span class="line">                       tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>))</span><br><span class="line">plotAnnoBar(peakAnnoList,title = <span class="string">&quot;                        Feature Distribution&quot;</span>)</span><br><span class="line">plotDistToTSS(peakAnnoList,title = <span class="string">&quot;                 Feature Distribution relative to TSS&quot;</span>)</span><br><span class="line"><span class="comment">#例举单个样本</span></span><br><span class="line">peakAnno &lt;- annotatePeak(files[[<span class="number">1</span>]],<span class="comment"># 分别改成2或者3或者4即可，分别对应四个文件</span></span><br><span class="line">                        tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                        TxDb=txdb,</span><br><span class="line">                        annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">plotAnnoPie(peakAnnoLipeakAnnost)</span><br><span class="line">upsetplot(peakAnno, vennpie=<span class="literal">TRUE</span>)</span><br></pre></td></tr></table></figure>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/Feature Distribution TSS.jpeg" width="50%"></p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/plotAnnoPie.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/upsetplot(peakAnno, vennpie=TRUE).jpeg" width="50%"></p>
<h3 id="2、查看peaks在全基因组上的分布"><a href="#2、查看peaks在全基因组上的分布" class="headerlink" title="2、查看peaks在全基因组上的分布"></a>2、查看peaks在全基因组上的分布</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#输入文件的准备</span></span><br><span class="line">DMSO_24h_wt_rep1&lt;-read.csv(<span class="string">&quot;SRR12692092_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">DMSO_24h_wt_rep1&lt;-DMSO_24h_wt_rep1[,-<span class="number">4</span>]</span><br><span class="line">DMSO_24h_wt_rep2&lt;-read.csv(<span class="string">&quot;SRR12692093_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">DMSO_24h_wt_rep2&lt;-DMSO_24h_wt_rep2[,-<span class="number">4</span>]</span><br><span class="line">BRM014_10uM_24h_wt_rep1&lt;-read.csv(<span class="string">&quot;SRR12692098_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">BRM014_10uM_24h_wt_rep1&lt;-BRM014_10uM_24h_wt_rep1[,-<span class="number">4</span>]</span><br><span class="line">BRM014_10uM_24h_wt_rep2&lt;-read.csv(<span class="string">&quot;SRR12692099_summits_filt_blacklist.csv&quot;</span>)</span><br><span class="line">BRM014_10uM_24h_wt_rep2&lt;-BRM014_10uM_24h_wt_rep2[,-<span class="number">4</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment">#以DMSO_24h_wt_rep1为例</span></span><br><span class="line">set.seed(<span class="number">123</span>)</span><br><span class="line">circos.initializeWithIdeogram(plotType = <span class="built_in">c</span>(<span class="string">&quot;axis&quot;</span>, <span class="string">&quot;labels&quot;</span>))</span><br><span class="line">circos.track(ylim = <span class="built_in">c</span>(<span class="number">0</span>, <span class="number">1</span>), panel.fun = <span class="keyword">function</span>(x, y) &#123;</span><br><span class="line">  chr = CELL_META$sector.index</span><br><span class="line">  xlim = CELL_META$xlim</span><br><span class="line">  ylim = CELL_META$ylim</span><br><span class="line">  circos.rect(xlim[<span class="number">1</span>], <span class="number">0</span>, xlim[<span class="number">2</span>], <span class="number">1</span>)</span><br><span class="line">&#125;, track.height = <span class="number">0.15</span>, bg.border = <span class="literal">NA</span>, bg.col=rainbow(<span class="number">24</span>))</span><br><span class="line">text(<span class="number">0</span>, <span class="number">0</span>, <span class="string">&quot;DMSO_24h_wt_rep1&quot;</span>, cex = <span class="number">1.5</span>)</span><br><span class="line">circos.genomicDensity(DMSO_24h_wt_rep1, col = <span class="built_in">c</span>(<span class="string">&quot;#000080&quot;</span>), track.height = <span class="number">0.2</span>)</span><br><span class="line">circos.clear()</span><br></pre></td></tr></table></figure>
<p>看到这样的结果，第一反应就是————为什么两种处理情况下染色体开放程度那么像！？难道我代码有问题！？经过反复检查验证（将一个样本chr1上的peaks都删掉，再次运行上述代码，就会发现显著的改变），可以确定分析上是没有问题的。这两种处理导致的差异可能不是很显著。再加上20万+的peaks放在这个小小的circos图上展示，有些差异会被掩盖掉。就如在做TSS富集分析的时候，单独看TSS前后3Kb区域，可以看到有两个峰，但在看TSS-genebody-TSE区域是，TSS处相对微弱的那个峰就被掩盖掉了。</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt_rep2.jpeg" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep1.jpeg" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt_rep2.jpeg" width="50%"></p>
<h3 id="3、拿到每个样本中peaks对应得基因名"><a href="#3、拿到每个样本中peaks对应得基因名" class="headerlink" title="3、拿到每个样本中peaks对应得基因名"></a>3、拿到每个样本中peaks对应得基因名</h3><p>这一步非常重要，拿到基因名就可以根据课题需要进行差异分析等</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#以DMSO_24h_wt样本为例</span></span><br><span class="line"><span class="comment">#replicate 1</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep1 &lt;- annotatePeak(files[[<span class="number">1</span>]],</span><br><span class="line">                                          tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                                          TxDb=txdb,</span><br><span class="line">                                          annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">genelist_DMSO_24h_wt_rep1_uniqe&lt;-as.data.frame(unique(peakAnno_DMSO_24h_wt_rep1@anno@elementMetadata@listData[[<span class="string">&quot;SYMBOL&quot;</span>]]))</span><br><span class="line">colnames(genelist_DMSO_24h_wt_rep1_uniqe)&lt;-<span class="string">&quot;symbol&quot;</span></span><br><span class="line"><span class="comment">#replicate 2</span></span><br><span class="line">peakAnno_DMSO_24h_wt_rep2 &lt;- annotatePeak(files[[<span class="number">2</span>]],</span><br><span class="line">                                          tssRegion=<span class="built_in">c</span>(-<span class="number">3000</span>, <span class="number">3000</span>),</span><br><span class="line">                                          TxDb=txdb,</span><br><span class="line">                                          annoDb=<span class="string">&quot;org.Mm.eg.db&quot;</span>)</span><br><span class="line">genelist_DMSO_24h_wt_rep2_uniqe&lt;-as.data.frame(unique(peakAnno_DMSO_24h_wt_rep2@anno@elementMetadata@listData[[<span class="string">&quot;SYMBOL&quot;</span>]]))</span><br><span class="line">colnames(genelist_DMSO_24h_wt_rep2_uniqe)&lt;-<span class="string">&quot;symbol&quot;</span></span><br><span class="line"><span class="comment">#重复样本间共同的开放基因</span></span><br><span class="line">venn.diagram(</span><br><span class="line">  x=<span class="built_in">list</span>(</span><br><span class="line">    DMSO_24h_wt_rep1=genelist_DMSO_24h_wt_rep1_uniqe$symbol,</span><br><span class="line">    DMSO_24h_wt_rep2=genelist_DMSO_24h_wt_rep2_uniqe$symbol</span><br><span class="line">  ),</span><br><span class="line">  filename = <span class="string">&quot;DMSO_24h_wt.png&quot;</span>,</span><br><span class="line">  lty=<span class="string">&quot;dotted&quot;</span>,</span><br><span class="line">  lwd=<span class="number">3</span>,</span><br><span class="line">  col=<span class="string">&quot;transparent&quot;</span>,</span><br><span class="line">  fill=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;cornflowerblue&quot;</span>),</span><br><span class="line">  alpha=<span class="number">0.5</span>,</span><br><span class="line">  </span><br><span class="line">  label.col=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;white&quot;</span>,<span class="string">&quot;darkblue&quot;</span>) ,</span><br><span class="line">  cex=<span class="number">1</span>,</span><br><span class="line">  fontfamily=<span class="string">&quot;serif&quot;</span>,</span><br><span class="line">  fontface=<span class="string">&quot;bold&quot;</span>,</span><br><span class="line">  cat.default.pos=<span class="string">&quot;text&quot;</span>,</span><br><span class="line">  cat.col=<span class="built_in">c</span>(<span class="string">&quot;darkorchid1&quot;</span>,<span class="string">&quot;darkblue&quot;</span>),</span><br><span class="line">  cat.cex=<span class="number">0.6</span>,</span><br><span class="line">  cat.fontfamily=<span class="string">&quot;serif&quot;</span>,</span><br><span class="line">  cat.dist=<span class="built_in">c</span>(<span class="number">0.3</span>,<span class="number">0.3</span>),</span><br><span class="line">  cat.pos=<span class="number">0</span></span><br><span class="line">)</span><br><span class="line"></span><br><span class="line"><span class="comment">#查看各组内样本间的overlapping reads：DMSO_24h_wt， BRM014_10uM_24h_wt；</span></span><br><span class="line"><span class="comment">#以及组间peaks的异同情况：DMSO_24h_wt vs. BRM014_10uM_24h_wt</span></span><br><span class="line"><span class="comment">#代码类似上面的，就不一一展示了</span></span><br></pre></td></tr></table></figure>
<p>从下图可以看出，不管是组间还是组内，差异的peaks数目都不是很多了，这一点也验证了我们上面做的再全基因组范围内查看peaks的分布结果。</p>
<p><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_24h_wt.png" width="50%"><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/BRM014_10uM_24h_wt.png" width="50%"><br><img src="https://blog-image-host.oss-cn-shanghai.aliyuncs.com/gyqblog/DMSO_BRM014.png" width="50%"></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/NGS/" rel="tag"># NGS</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2020/05/14/NGS%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90%E4%B9%8B%E8%A1%A8%E8%A7%82/" rel="prev" title="NGS数据分析之表观">
                  <i class="fa fa-chevron-left"></i> NGS数据分析之表观
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2021/01/09/%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E4%B9%8BMISO/" rel="next" title="可变剪切分析之MISO">
                  可变剪切分析之MISO <i class="fa fa-chevron-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>







<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      const activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      const commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>
</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">


<div class="copyright">
  &copy; 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">gongyuqiqi</span>
</div>
<div class="wordcount">
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-line"></i>
    </span>
      <span>Symbols count total: </span>
    <span title="Symbols count total">136k</span>
  </span>
  <span class="post-meta-item">
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
      <span>Reading time total &asymp;</span>
    <span title="Reading time total">2:03</span>
  </span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" class="theme-link" rel="noopener" target="_blank">NexT.Pisces</a>
  </div>

    </div>
  </footer>

  
  <script src="https://cdn.jsdelivr.net/npm/animejs@3.2.1/lib/anime.min.js"></script>
<script src="/js/utils.js"></script><script src="/js/motion.js"></script><script src="/js/next-boot.js"></script>

  
<script src="/js/local-search.js"></script>






  





</body>
</html>
